{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "dc0ff5ff-54e3-4977-befc-750988e9a61f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2'"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a004536f-d2b4-432c-afbf-8210b1eb4fcc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata\n"
     ]
    }
   ],
   "source": [
    "cd Rainforestdata/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9781b2e5-37de-4034-8dcf-860f6c428288",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bc8e8b95-81db-4fb3-b9de-9636123963c0",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SRR18218015.1 1 length=106\n",
      "NTATTTCCTCAGTCACCGTGGATGCGACTGACAGTCTNNNNNNNTGAGGCCNAGNNNNTGAGTCANNNCNNNGAGGCTGNNNNNNNTGTTGACNNNGGNCTTGCTA\n",
      "+SRR18218015.1 1 length=106\n",
      "#8ACCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGF#######::CFDFG#:C####::CFDDF###:###::DFGGG#######::CCDGG###::#:CFFFGG\n",
      "@SRR18218015.2 2 length=300\n",
      "NTCCTGTAGCTGGTATCGTTGAGCCTTACGCATGGCTNNNNNNNGCAGACANCANNNNCAATGAGNNTGNNNAACACAANNNNNNNCTGACATNNNACNTGAGCAACCCTAAGGTCCGTAAGATGATTNNNGNCTTACANANNNNTNNNGNCGANGCGTATGCGAANNCACTCGCAGANNNNNNGAAGNNNCCACCTCAGNTTCAGCNTGGCNNNNNNNCTATTGAACCACGAGAAGNCGACATGCCNNGGANTGAGAATNGTGACGGTACTNTTACATTGAAGTTTAAATGCTTCNCNT\n",
      "+SRR18218015.2 2 length=300\n",
      "#8ACCGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGG#######::CDFGG#:C####::DGFFE##::###::CDEGG#######::CFFGG###::#:CDFGGGGGGGGGGGGGGGGGGDFGBFGG###:#:BFGGG#8####:###:#::B#::AFFE8FFDE##::DFGFGGGG######38@6###66DFGGGGG#6@FGFG#6=:F#######44=FGGFFGCFGGGGGGC#4:;EGGGGG##228#2;8FGGC#2:CFGGGGGGG#2;EGGGGF?FCAFFGGGGGGGGC#1#2\n",
      "@SRR18218015.3 3 length=301\n",
      "NCTCAGGAGTCATCTTAGGGTTCTCAGCCATTAATGCNNNNNNNAGTCTCTNCNNNNNGTGTGGANNNNNNNCATATCCNNNNNNNCATGACTNNNTGNTAGCATTCTGGAAACCTTCAGCCCCACCANNNANCTCAGTNTNNNNCNNNTNCTNNTTANTGCTGTANNCGATAGGTACNNNNNNCCCTNNNAAGCGACTGNCTGGGANGATGNNNNNNNCCTTAGGGTTCCCGAACANTGCAGGGTTNNCCANCATCTCNNGCTTAGAGTCCNGAACTTGCTTCATAATGTCGTATNCNTT\n",
      "+SRR18218015.3 3 length=301\n",
      "#8ACCGGGGGGGGFGGGEGGGGGGFGGFCFGGFGFGG#######::CFFFG#:#####::DFGGF#######::CCFEE#\u001b[7mSRR18218015_1.fastq.gz\u001b[m\u001b[K\u0007\u001b[H\u001b[2J\u001b[H\u001b[H\u001b[2J\u001b[H@SRR18218015.1 1 length=106\n",
      "NTATTTCCTCAGTCACCGTGGATGCGACTGACAGTCTNNNNNNNTGAGGCCNAGNNNNTGAGTCANNNCNNNGAGGCTGNNNNNNNTGTTGACNNNGGNCTTGCTA\n",
      "+SRR18218015.1 1 length=106\n",
      "#8ACCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGF#######::CFDFG#:C####::CFDDF###:###::DFGGG#######::CCDGG###::#:CFFFGG\n",
      "@SRR18218015.2 2 length=300\n",
      "NTCCTGTAGCTGGTATCGTTGAGCCTTACGCATGGCTNNNNNNNGCAGACANCANNNNCAATGAGNNTGNNNAACACAANNNNNNNCTGACATNNNACNTGAGCAACCCTAAGGTCCGTAAGATGATTNNNGNCTTACANANNNNTNNNGNCGANGCGTATGCGAANNCACTCGCAGANNNNNNGAAGNNNCCACCTCAGNTTCAGCNTGGCNNNNNNNCTATTGAACCACGAGAAGNCGACATGCCNNGGANTGAGAATNGTGACGGTACTNTTACATTGAAGTTTAAATGCTTCNCNT\n",
      "+SRR18218015.2 2 length=300\n",
      "#8ACCGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGG#######::CDFGG#:C####::DGFFE##::###::CDEGG#######::CFFGG###::#:CDFGGGGGGGGGGGGGGGGGGDFGBFGG###:#:BFGGG#8####:###:#::B#::AFFE8FFDE##::DFGFGGGG######38@6###66DFGGGGG#6@FGFG#6=:F#######44=FGGFFGCFGGGGGGC#4:;EGGGGG##228#2;8FGGC#2:CFGGGGGGG#2;EGGGGF?FCAFFGGGGGGGGC#1#2\n",
      "@SRR18218015.3 3 length=301\n",
      "NCTCAGGAGTCATCTTAGGGTTCTCAGCCATTAATGCNNNNNNNAGTCTCTNCNNNNNGTGTGGANNNNNNNCATATCCNNNNNNNCATGACTNNNTGNTAGCATTCTGGAAACCTTCAGCCCCACCANNNANCTCAGTNTNNNNCNNNTNCTNNTTANTGCTGTANNCGATAGGTACNNNNNNCCCTNNNAAGCGACTGNCTGGGANGATGNNNNNNNCCTTAGGGTTCCCGAACANTGCAGGGTTNNCCANCATCTCNNGCTTAGAGTCCNGAACTTGCTTCATAATGTCGTATNCNTT\n",
      "+SRR18218015.3 3 length=301\n",
      "#8ACCGGGGGGGGFGGGEGGGGGGFGGFCFGGFGFGG#######::CFFFG#:#####::DFGGF#######::CCFEE#:\u001b[K"
     ]
    }
   ],
   "source": [
    "!less SRR18218015_1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b058ab45-f1a7-42c5-86fa-e1d5e5901c60",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: fastqc: command not found\n"
     ]
    }
   ],
   "source": [
    "!fastqc -o fastqc SRR18218015_1.fastq.gz SRR18218015_2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "b3298022-f125-42c7-883c-13c2ba5f49ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc'"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "88e55ebb-8568-4c30-a6af-9bcca2db85dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cafb934f-7f47-40d9-9b8a-1bce5fb1adf9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: fastqc: command not found\n"
     ]
    }
   ],
   "source": [
    "!fastqc -o fastqc SRR18218015_1.fastq.gz SRR18218015_2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6eb1425f-e87f-4e9f-bb78-aadfbc4c1382",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (3322819607.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[4], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    fastqc -o fastqc SRR18218015_1.fastq.gz SRR18218015_2.fastq.gz\u001b[0m\n\u001b[0m              ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "fastqc -o fastqc SRR18218015_1.fastq.gz SRR18218015_2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "f2911d14-3ae8-4b4a-9492-2251a070699d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'phageinformatics_UCPH_June_2024/Rainforestdata_2/'\n",
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc\n"
     ]
    }
   ],
   "source": [
    "cd phageinformatics_UCPH_June_2024/Rainforestdata_2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "3610b2eb-0d42-400b-a2be-46f4193f8fda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024'"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "70b26250-fe1a-4702-8a9c-4163f3bfb3e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2\n"
     ]
    }
   ],
   "source": [
    "cd Rainforestdata_2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "89b3ac7d-e6e1-45b3-af56-4164a222de6a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-07-18 15:31:01--  https://sid.erda.dk/share_redirect/BYsn3Xo3IZ/bin/fastx_detect_fq.py\n",
      "Resolving sid.erda.dk (sid.erda.dk)... 130.225.104.13\n",
      "Connecting to sid.erda.dk (sid.erda.dk)|130.225.104.13|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 2195 (2.1K)\n",
      "Saving to: ‘fastx_detect_fq.py’\n",
      "\n",
      "fastx_detect_fq.py  100%[===================>]   2.14K  --.-KB/s    in 0s      \n",
      "\n",
      "2024-07-18 15:31:02 (34.5 MB/s) - ‘fastx_detect_fq.py’ saved [2195/2195]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://sid.erda.dk/share_redirect/BYsn3Xo3IZ/bin/fastx_detect_fq.py\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "d1a46741-9044-43ba-83b6-74be522bf877",
   "metadata": {},
   "outputs": [],
   "source": [
    "!chmod 755 fastx_detect_fq.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "76332c29-4ddd-4bd3-a9ce-2ed8fd9fa12a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sanger\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "!gzip -dc SRR18218015_1.fastq.gz | ./fastx_detect_fq.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "2fc24228-6520-4060-b42a-d48b91607321",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SRR18218015.1 1 length=106\n",
      "NTATTTCCTCAGTCACCGTGGATGCGACTGACAGTCTNNNNNNNTGAGGCCNAGNNNNTGAGTCANNNCNNNGAGGCTGNNNNNNNTGTTGACNNNGGNCTTGCTA\n",
      "+SRR18218015.1 1 length=106\n",
      "#8ACCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGF#######::CFDFG#:C####::CFDDF###:###::DFGGG#######::CCDGG###::#:CFFFGG\n",
      "@SRR18218015.2 2 length=300\n",
      "NTCCTGTAGCTGGTATCGTTGAGCCTTACGCATGGCTNNNNNNNGCAGACANCANNNNCAATGAGNNTGNNNAACACAANNNNNNNCTGACATNNNACNTGAGCAACCCTAAGGTCCGTAAGATGATTNNNGNCTTACANANNNNTNNNGNCGANGCGTATGCGAANNCACTCGCAGANNNNNNGAAGNNNCCACCTCAGNTTCAGCNTGGCNNNNNNNCTATTGAACCACGAGAAGNCGACATGCCNNGGANTGAGAATNGTGACGGTACTNTTACATTGAAGTTTAAATGCTTCNCNT\n",
      "+SRR18218015.2 2 length=300\n",
      "#8ACCGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGG#######::CDFGG#:C####::DGFFE##::###::CDEGG#######::CFFGG###::#:CDFGGGGGGGGGGGGGGGGGGDFGBFGG###:#:BFGGG#8####:###:#::B#::AFFE8FFDE##::DFGFGGGG######38@6###66DFGGGGG#6@FGFG#6=:F#######44=FGGFFGCFGGGGGGC#4:;EGGGGG##228#2;8FGGC#2:CFGGGGGGG#2;EGGGGF?FCAFFGGGGGGGGC#1#2\n",
      "@SRR18218015.3 3 length=301\n",
      "NCTCAGGAGTCATCTTAGGGTTCTCAGCCATTAATGCNNNNNNNAGTCTCTNCNNNNNGTGTGGANNNNNNNCATATCCNNNNNNNCATGACTNNNTGNTAGCATTCTGGAAACCTTCAGCCCCACCANNNANCTCAGTNTNNNNCNNNTNCTNNTTANTGCTGTANNCGATAGGTACNNNNNNCCCTNNNAAGCGACTGNCTGGGANGATGNNNNNNNCCTTAGGGTTCCCGAACANTGCAGGGTTNNCCANCATCTCNNGCTTAGAGTCCNGAACTTGCTTCATAATGTCGTATNCNTT\n"
     ]
    }
   ],
   "source": [
    "!head SRR18218015_1.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "c5c779e1-623d-4919-bf3f-10bf982d10ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp SRR18218015_1.fastq SRR18218015_1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "4ef03575-6cfa-4dd8-a657-f822e9f3831d",
   "metadata": {},
   "outputs": [],
   "source": [
    "!rm SRR18218015_1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "ee78d700-faba-469c-8fc9-92bc4e174d97",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp SRR18218015_1.fastq SRR18218015_1_trial.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "b88c4759-e844-43ff-a446-0aadc1f6e1b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gzip SRR18218015_1.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "f88758c0-002f-4518-b3ad-036545dc6eb5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SRR18218015.1 1 length=106\n",
      "NTATTTCCTCAGTCACCGTGGATGCGACTGACAGTCTNNNNNNNTGAGGCCNAGNNNNTGAGTCANNNCNNNGAGGCTGNNNNNNNTGTTGACNNNGGNCTTGCTA\n",
      "+SRR18218015.1 1 length=106\n",
      "#8ACCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGF#######::CFDFG#:C####::CFDDF###:###::DFGGG#######::CCDGG###::#:CFFFGG\n",
      "@SRR18218015.2 2 length=300\n",
      "NTCCTGTAGCTGGTATCGTTGAGCCTTACGCATGGCTNNNNNNNGCAGACANCANNNNCAATGAGNNTGNNNAACACAANNNNNNNCTGACATNNNACNTGAGCAACCCTAAGGTCCGTAAGATGATTNNNGNCTTACANANNNNTNNNGNCGANGCGTATGCGAANNCACTCGCAGANNNNNNGAAGNNNCCACCTCAGNTTCAGCNTGGCNNNNNNNCTATTGAACCACGAGAAGNCGACATGCCNNGGANTGAGAATNGTGACGGTACTNTTACATTGAAGTTTAAATGCTTCNCNT\n",
      "+SRR18218015.2 2 length=300\n",
      "#8ACCGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGG#######::CDFGG#:C####::DGFFE##::###::CDEGG#######::CFFGG###::#:CDFGGGGGGGGGGGGGGGGGGDFGBFGG###:#:BFGGG#8####:###:#::B#::AFFE8FFDE##::DFGFGGGG######38@6###66DFGGGGG#6@FGFG#6=:F#######44=FGGFFGCFGGGGGGC#4:;EGGGGG##228#2;8FGGC#2:CFGGGGGGG#2;EGGGGF?FCAFFGGGGGGGGC#1#2\n",
      "@SRR18218015.3 3 length=301\n",
      "NCTCAGGAGTCATCTTAGGGTTCTCAGCCATTAATGCNNNNNNNAGTCTCTNCNNNNNGTGTGGANNNNNNNCATATCCNNNNNNNCATGACTNNNTGNTAGCATTCTGGAAACCTTCAGCCCCACCANNNANCTCAGTNTNNNNCNNNTNCTNNTTANTGCTGTANNCGATAGGTACNNNNNNCCCTNNNAAGCGACTGNCTGGGANGATGNNNNNNNCCTTAGGGTTCCCGAACANTGCAGGGTTNNCCANCATCTCNNGCTTAGAGTCCNGAACTTGCTTCATAATGTCGTATNCNTT\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "!gzip -dc SRR18218015_1.fastq.gz | head "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "d88bf263-83fb-4e51-aeb7-075b33d3aedf",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+SRR18218015.705459 705459 length=300\n",
      "CCCCCGGGGGGGGGGGGGGGGGGEEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGFGGGGGGGGGGEEGGGGDFGGGGGGGGGGGGGCGGGGGFFGGGGGGGDGGGGGGGGGGGGGGGDGGGGGG>G>EGDGFCG@CGEGGDGGGDFEGGGGFDDGFFGFGAGGGFCGB7@FD:G?6>FBFFAFA4AAAF<26B2<B????FF?0904>F?FBFFF:>(93(4(<9486>1>(44(,(24((47(5?90707:\n",
      "@SRR18218015.705460 705460 length=125\n",
      "GTCCTATAGTGATAGTTAAGTCAGAATTACTTGCATATCAGTGGGTTAGCGTGAAGATGACTGAAGTCACCGTTGGTCGTGTGCTTGATGGTTGACCGCTGGTCTATCGCCTTCCAGCCTGAGAC\n",
      "+SRR18218015.705460 705460 length=125\n",
      "CCCCCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "@SRR18218015.705461 705461 length=301\n",
      "CTTCACGCTAACCCACTGATATGCAAGTAATTCTGACTTAACTATCACTATAGGACTCAAGGTCTAAGACTCCAAGTGAAAGACCAAAGAGACTTTAAGTGAAAGACTAATAACAAGGACTTTAAGTGTGAGCGTCATCTCAATTGACAAGCACGACTTCTCTGATGTGTCGAATGCAATCGAACCCTTCAACGTACTGGCGGACCACTACGGTCAGGACCTCGCAGTCAAGCAGCTACAACTTGAGCACGAGGCATACACTGAAGGAGAGCGACGTTTCATCAAGAACCTTTAGCGCCAG\n",
      "+SRR18218015.705461 705461 length=301\n",
      "CCCCCGGGGGGGFGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGFFGGGGGGGGGDGGGGGGGGGGCFFGGGGGGGGGGCCGFGGFGGGGFGFGGGGGGGGGGGGGGGGGGGGGGGDAEF<<F9?CFGGGGFGGCGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFFDGGGGGGGGCEEGFGGGGGGGGGGEFFD6ADGFGFGGDFGGFFAFGDGFFAEFEFFFE@D@BBBEFFFBFFF@BEF=239@;@EEEEEFEEF831,7;439993:?ABFFF<7?F:4<)4<4366;(\n"
     ]
    }
   ],
   "source": [
    "!gzip -dc SRR18218015_2.fastq.gz | tail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ae5cf703-56d0-41cd-80ba-0bc133c3e6b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2\n"
     ]
    }
   ],
   "source": [
    "cd /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "714e112d-3788-4b3e-9909-e31e57d5ed5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "19d12c67-f5de-492d-86e3-75547cc89bc1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!gzip -dc SRR18218015_1.fastq.gz > SRR18218015_1.fastq\n",
    "!gzip -dc SRR18218015_2.fastq.gz > SRR18218015_2.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6c1dd698-c2ec-4f4e-ac17-3192f6221ae7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Attempting trial-4 of the adapter removal. Input files are unziped from their gzip files to their raw files format (.fastq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d3160c21-243b-4046-a8cb-1c812e36fbae",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null\n",
      "null\n",
      "Started analysis of SRR18218015_1.fastq\n",
      "Approx 5% complete for SRR18218015_1.fastq\n",
      "Approx 10% complete for SRR18218015_1.fastq\n",
      "Approx 15% complete for SRR18218015_1.fastq\n",
      "Approx 20% complete for SRR18218015_1.fastq\n",
      "Approx 25% complete for SRR18218015_1.fastq\n",
      "Approx 30% complete for SRR18218015_1.fastq\n",
      "Approx 35% complete for SRR18218015_1.fastq\n",
      "Approx 40% complete for SRR18218015_1.fastq\n",
      "Approx 45% complete for SRR18218015_1.fastq\n",
      "Approx 50% complete for SRR18218015_1.fastq\n",
      "Approx 55% complete for SRR18218015_1.fastq\n",
      "Approx 60% complete for SRR18218015_1.fastq\n",
      "Approx 65% complete for SRR18218015_1.fastq\n",
      "Approx 70% complete for SRR18218015_1.fastq\n",
      "Approx 75% complete for SRR18218015_1.fastq\n",
      "Approx 80% complete for SRR18218015_1.fastq\n",
      "Approx 85% complete for SRR18218015_1.fastq\n",
      "Approx 90% complete for SRR18218015_1.fastq\n",
      "Approx 95% complete for SRR18218015_1.fastq\n",
      "Analysis complete for SRR18218015_1.fastq\n",
      "Started analysis of SRR18218015_2.fastq\n",
      "Approx 5% complete for SRR18218015_2.fastq\n",
      "Approx 10% complete for SRR18218015_2.fastq\n",
      "Approx 15% complete for SRR18218015_2.fastq\n",
      "Approx 20% complete for SRR18218015_2.fastq\n",
      "Approx 25% complete for SRR18218015_2.fastq\n",
      "Approx 30% complete for SRR18218015_2.fastq\n",
      "Approx 35% complete for SRR18218015_2.fastq\n",
      "Approx 40% complete for SRR18218015_2.fastq\n",
      "Approx 45% complete for SRR18218015_2.fastq\n",
      "Approx 50% complete for SRR18218015_2.fastq\n",
      "Approx 55% complete for SRR18218015_2.fastq\n",
      "Approx 60% complete for SRR18218015_2.fastq\n",
      "Approx 65% complete for SRR18218015_2.fastq\n",
      "Approx 70% complete for SRR18218015_2.fastq\n",
      "Approx 75% complete for SRR18218015_2.fastq\n",
      "Approx 80% complete for SRR18218015_2.fastq\n",
      "Approx 85% complete for SRR18218015_2.fastq\n",
      "Approx 90% complete for SRR18218015_2.fastq\n",
      "Approx 95% complete for SRR18218015_2.fastq\n",
      "Analysis complete for SRR18218015_2.fastq\n"
     ]
    }
   ],
   "source": [
    "!fastqc -o fastqc/trial4files/ SRR18218015_1.fastq SRR18218015_2.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "94da1c84-8fc9-4648-8e02-945118cf3264",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Trimming single ended reads ...\n",
      "Opening FASTQ file 'SRR18218015_2.fastq', line numbers start at 1\n",
      "Processed a total of 705,461 reads in 2.0s; 359,000 reads per second on average ...\n"
     ]
    }
   ],
   "source": [
    "!AdapterRemoval --file1 SRR18218015_2.fastq --basename SRR18218015_2 --trimqualities --trimns --minquality 20 --minlength 50 --trim3p 11 --adapter1 NNNNNNNNNNN --threads 7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ae0b49ac-a4e5-45b6-b49b-ceb1114cd4a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null\n",
      "Started analysis of SRR18218015_2.truncated\n",
      "Approx 5% complete for SRR18218015_2.truncated\n",
      "Approx 10% complete for SRR18218015_2.truncated\n",
      "Approx 15% complete for SRR18218015_2.truncated\n",
      "Approx 20% complete for SRR18218015_2.truncated\n",
      "Approx 25% complete for SRR18218015_2.truncated\n",
      "Approx 30% complete for SRR18218015_2.truncated\n",
      "Approx 35% complete for SRR18218015_2.truncated\n",
      "Approx 40% complete for SRR18218015_2.truncated\n",
      "Approx 45% complete for SRR18218015_2.truncated\n",
      "Approx 50% complete for SRR18218015_2.truncated\n",
      "Approx 55% complete for SRR18218015_2.truncated\n",
      "Approx 60% complete for SRR18218015_2.truncated\n",
      "Approx 65% complete for SRR18218015_2.truncated\n",
      "Approx 70% complete for SRR18218015_2.truncated\n",
      "Approx 75% complete for SRR18218015_2.truncated\n",
      "Approx 80% complete for SRR18218015_2.truncated\n",
      "Approx 85% complete for SRR18218015_2.truncated\n",
      "Approx 90% complete for SRR18218015_2.truncated\n",
      "Approx 95% complete for SRR18218015_2.truncated\n",
      "Analysis complete for SRR18218015_2.truncated\n"
     ]
    }
   ],
   "source": [
    "!fastqc -o fastqc/trial4files/ SRR18218015_2.truncated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f371397d-988a-4b67-b67d-178289df65df",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AdapterRemoval ver. 2.3.3\n",
      "Trimming of single-end reads\n",
      "\n",
      "\n",
      "[Adapter sequences]\n",
      "Adapter1[1]: NNNNNNNNNNN\n",
      "Adapter2[1]: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "\n",
      "\n",
      "[Adapter trimming]\n",
      "RNG seed: NA\n",
      "Alignment shift value: 2\n",
      "Global mismatch threshold: 0.333333\n",
      "Quality format (input): Phred+33\n",
      "Quality score max (input): 41\n",
      "Quality format (output): Phred+33\n",
      "Quality score max (output): 41\n",
      "Mate-number separator (input): '/'\n",
      "Trimming 5p: 0\n",
      "Trimming 3p: 11\n",
      "Trimming Ns: Yes\n",
      "Trimming Phred scores <= 20: Yes\n",
      "Trimming using sliding windows: No\n",
      "Minimum genomic length: 50\n",
      "Maximum genomic length: 4294967295\n",
      "Collapse overlapping reads: No\n",
      "Deterministic collapse: No\n",
      "Conservative collapse: No\n",
      "Minimum overlap (in case of collapse): 11\n",
      "Minimum adapter overlap: 0\n",
      "\n",
      "\n",
      "[Trimming statistics]\n",
      "Total number of reads: 705461\n",
      "Number of unaligned reads: 705461\n",
      "Number of well aligned reads: 0\n",
      "Number of discarded mate 1 reads: 60604\n",
      "Number of singleton mate 1 reads: 644857\n",
      "Number of reads with adapters[1]: 0\n",
      "Number of retained reads: 644857\n",
      "Number of retained nucleotides: 117559184\n",
      "Average length of retained reads: 182.303\n",
      "\n",
      "\n",
      "[Length distribution]\n",
      "Length\tMate1\tDiscarded\tAll\n",
      "0\t0\t8805\t8805\n",
      "1\t0\t0\t0\n",
      "2\t0\t0\t0\n",
      "3\t0\t0\t0\n",
      "4\t0\t0\t0\n",
      "5\t0\t0\t0\n",
      "6\t0\t0\t0\n",
      "7\t0\t0\t0\n",
      "8\t0\t0\t0\n",
      "9\t0\t0\t0\n",
      "10\t0\t0\t0\n",
      "11\t0\t0\t0\n",
      "12\t0\t0\t0\n",
      "13\t0\t0\t0\n",
      "14\t0\t0\t0\n",
      "15\t0\t0\t0\n",
      "16\t0\t0\t0\n",
      "17\t0\t0\t0\n",
      "18\t0\t0\t0\n",
      "19\t0\t1\t1\n",
      "20\t0\t0\t0\n",
      "21\t0\t0\t0\n",
      "22\t0\t8\t8\n",
      "23\t0\t27\t27\n",
      "24\t0\t2084\t2084\n",
      "25\t0\t1480\t1480\n",
      "26\t0\t952\t952\n",
      "27\t0\t813\t813\n",
      "28\t0\t1110\t1110\n",
      "29\t0\t1536\t1536\n",
      "30\t0\t1948\t1948\n",
      "31\t0\t2452\t2452\n",
      "32\t0\t2895\t2895\n",
      "33\t0\t3207\t3207\n",
      "34\t0\t2991\t2991\n",
      "35\t0\t2191\t2191\n",
      "36\t0\t1327\t1327\n",
      "37\t0\t1065\t1065\n",
      "38\t0\t1252\t1252\n",
      "39\t0\t1804\t1804\n",
      "40\t0\t2326\t2326\n",
      "41\t0\t2807\t2807\n",
      "42\t0\t3030\t3030\n",
      "43\t0\t3201\t3201\n",
      "44\t0\t2822\t2822\n",
      "45\t0\t2258\t2258\n",
      "46\t0\t1631\t1631\n",
      "47\t0\t1299\t1299\n",
      "48\t0\t1405\t1405\n",
      "49\t0\t1877\t1877\n",
      "50\t2417\t0\t2417\n",
      "51\t2838\t0\t2838\n",
      "52\t3269\t0\t3269\n",
      "53\t3144\t0\t3144\n",
      "54\t2967\t0\t2967\n",
      "55\t2766\t0\t2766\n",
      "56\t1993\t0\t1993\n",
      "57\t1559\t0\t1559\n",
      "58\t1520\t0\t1520\n",
      "59\t1873\t0\t1873\n",
      "60\t2391\t0\t2391\n",
      "61\t2769\t0\t2769\n",
      "62\t3421\t0\t3421\n",
      "63\t3381\t0\t3381\n",
      "64\t3435\t0\t3435\n",
      "65\t2907\t0\t2907\n",
      "66\t2382\t0\t2382\n",
      "67\t1916\t0\t1916\n",
      "68\t1566\t0\t1566\n",
      "69\t1833\t0\t1833\n",
      "70\t2320\t0\t2320\n",
      "71\t2918\t0\t2918\n",
      "72\t3177\t0\t3177\n",
      "73\t3554\t0\t3554\n",
      "74\t3596\t0\t3596\n",
      "75\t3148\t0\t3148\n",
      "76\t2769\t0\t2769\n",
      "77\t2103\t0\t2103\n",
      "78\t1648\t0\t1648\n",
      "79\t1803\t0\t1803\n",
      "80\t2125\t0\t2125\n",
      "81\t2910\t0\t2910\n",
      "82\t3131\t0\t3131\n",
      "83\t3543\t0\t3543\n",
      "84\t3397\t0\t3397\n",
      "85\t3390\t0\t3390\n",
      "86\t3051\t0\t3051\n",
      "87\t2406\t0\t2406\n",
      "88\t2128\t0\t2128\n",
      "89\t1774\t0\t1774\n",
      "90\t2041\t0\t2041\n",
      "91\t2369\t0\t2369\n",
      "92\t3096\t0\t3096\n",
      "93\t3503\t0\t3503\n",
      "94\t3412\t0\t3412\n",
      "95\t3330\t0\t3330\n",
      "96\t3180\t0\t3180\n",
      "97\t2622\t0\t2622\n",
      "98\t2245\t0\t2245\n",
      "99\t2075\t0\t2075\n",
      "100\t2144\t0\t2144\n",
      "101\t2190\t0\t2190\n",
      "102\t2608\t0\t2608\n",
      "103\t3266\t0\t3266\n",
      "104\t3486\t0\t3486\n",
      "105\t3441\t0\t3441\n",
      "106\t3271\t0\t3271\n",
      "107\t2539\t0\t2539\n",
      "108\t2368\t0\t2368\n",
      "109\t2058\t0\t2058\n",
      "110\t2024\t0\t2024\n",
      "111\t2078\t0\t2078\n",
      "112\t2595\t0\t2595\n",
      "113\t2974\t0\t2974\n",
      "114\t3158\t0\t3158\n",
      "115\t3265\t0\t3265\n",
      "116\t3132\t0\t3132\n",
      "117\t2927\t0\t2927\n",
      "118\t2640\t0\t2640\n",
      "119\t2006\t0\t2006\n",
      "120\t1957\t0\t1957\n",
      "121\t2050\t0\t2050\n",
      "122\t2335\t0\t2335\n",
      "123\t2702\t0\t2702\n",
      "124\t3207\t0\t3207\n",
      "125\t2987\t0\t2987\n",
      "126\t3236\t0\t3236\n",
      "127\t2945\t0\t2945\n",
      "128\t2582\t0\t2582\n",
      "129\t2366\t0\t2366\n",
      "130\t2030\t0\t2030\n",
      "131\t2113\t0\t2113\n",
      "132\t2294\t0\t2294\n",
      "133\t2517\t0\t2517\n",
      "134\t2888\t0\t2888\n",
      "135\t2992\t0\t2992\n",
      "136\t3054\t0\t3054\n",
      "137\t2837\t0\t2837\n",
      "138\t2729\t0\t2729\n",
      "139\t2439\t0\t2439\n",
      "140\t2042\t0\t2042\n",
      "141\t2135\t0\t2135\n",
      "142\t1915\t0\t1915\n",
      "143\t2401\t0\t2401\n",
      "144\t2533\t0\t2533\n",
      "145\t2879\t0\t2879\n",
      "146\t2870\t0\t2870\n",
      "147\t2913\t0\t2913\n",
      "148\t2636\t0\t2636\n",
      "149\t2329\t0\t2329\n",
      "150\t2019\t0\t2019\n",
      "151\t2150\t0\t2150\n",
      "152\t1927\t0\t1927\n",
      "153\t2215\t0\t2215\n",
      "154\t2235\t0\t2235\n",
      "155\t2490\t0\t2490\n",
      "156\t2856\t0\t2856\n",
      "157\t2795\t0\t2795\n",
      "158\t2576\t0\t2576\n",
      "159\t2404\t0\t2404\n",
      "160\t2131\t0\t2131\n",
      "161\t2089\t0\t2089\n",
      "162\t1932\t0\t1932\n",
      "163\t2018\t0\t2018\n",
      "164\t2052\t0\t2052\n",
      "165\t2351\t0\t2351\n",
      "166\t2516\t0\t2516\n",
      "167\t2451\t0\t2451\n",
      "168\t2445\t0\t2445\n",
      "169\t2385\t0\t2385\n",
      "170\t2134\t0\t2134\n",
      "171\t2043\t0\t2043\n",
      "172\t1865\t0\t1865\n",
      "173\t1858\t0\t1858\n",
      "174\t2035\t0\t2035\n",
      "175\t2080\t0\t2080\n",
      "176\t2326\t0\t2326\n",
      "177\t2382\t0\t2382\n",
      "178\t2455\t0\t2455\n",
      "179\t2219\t0\t2219\n",
      "180\t2025\t0\t2025\n",
      "181\t1887\t0\t1887\n",
      "182\t1814\t0\t1814\n",
      "183\t1841\t0\t1841\n",
      "184\t1893\t0\t1893\n",
      "185\t1978\t0\t1978\n",
      "186\t1970\t0\t1970\n",
      "187\t2169\t0\t2169\n",
      "188\t2408\t0\t2408\n",
      "189\t2177\t0\t2177\n",
      "190\t2139\t0\t2139\n",
      "191\t1917\t0\t1917\n",
      "192\t1817\t0\t1817\n",
      "193\t1799\t0\t1799\n",
      "194\t1764\t0\t1764\n",
      "195\t1870\t0\t1870\n",
      "196\t2025\t0\t2025\n",
      "197\t2018\t0\t2018\n",
      "198\t2059\t0\t2059\n",
      "199\t1972\t0\t1972\n",
      "200\t2037\t0\t2037\n",
      "201\t1855\t0\t1855\n",
      "202\t1674\t0\t1674\n",
      "203\t1676\t0\t1676\n",
      "204\t1645\t0\t1645\n",
      "205\t1798\t0\t1798\n",
      "206\t1775\t0\t1775\n",
      "207\t1843\t0\t1843\n",
      "208\t1939\t0\t1939\n",
      "209\t1897\t0\t1897\n",
      "210\t1875\t0\t1875\n",
      "211\t1825\t0\t1825\n",
      "212\t1504\t0\t1504\n",
      "213\t1617\t0\t1617\n",
      "214\t1467\t0\t1467\n",
      "215\t1613\t0\t1613\n",
      "216\t1672\t0\t1672\n",
      "217\t1657\t0\t1657\n",
      "218\t1748\t0\t1748\n",
      "219\t1859\t0\t1859\n",
      "220\t1756\t0\t1756\n",
      "221\t1639\t0\t1639\n",
      "222\t1606\t0\t1606\n",
      "223\t1412\t0\t1412\n",
      "224\t1400\t0\t1400\n",
      "225\t1482\t0\t1482\n",
      "226\t1498\t0\t1498\n",
      "227\t1636\t0\t1636\n",
      "228\t1623\t0\t1623\n",
      "229\t1598\t0\t1598\n",
      "230\t1615\t0\t1615\n",
      "231\t1539\t0\t1539\n",
      "232\t1467\t0\t1467\n",
      "233\t1329\t0\t1329\n",
      "234\t1310\t0\t1310\n",
      "235\t1448\t0\t1448\n",
      "236\t1334\t0\t1334\n",
      "237\t1307\t0\t1307\n",
      "238\t1389\t0\t1389\n",
      "239\t1467\t0\t1467\n",
      "240\t1541\t0\t1541\n",
      "241\t1524\t0\t1524\n",
      "242\t1375\t0\t1375\n",
      "243\t1228\t0\t1228\n",
      "244\t1215\t0\t1215\n",
      "245\t1280\t0\t1280\n",
      "246\t1263\t0\t1263\n",
      "247\t1311\t0\t1311\n",
      "248\t1177\t0\t1177\n",
      "249\t1374\t0\t1374\n",
      "250\t1351\t0\t1351\n",
      "251\t1371\t0\t1371\n",
      "252\t1317\t0\t1317\n",
      "253\t1314\t0\t1314\n",
      "254\t1137\t0\t1137\n",
      "255\t1264\t0\t1264\n",
      "256\t1223\t0\t1223\n",
      "257\t1262\t0\t1262\n",
      "258\t1200\t0\t1200\n",
      "259\t1200\t0\t1200\n",
      "260\t1212\t0\t1212\n",
      "261\t1266\t0\t1266\n",
      "262\t1316\t0\t1316\n",
      "263\t1121\t0\t1121\n",
      "264\t1169\t0\t1169\n",
      "265\t1231\t0\t1231\n",
      "266\t1362\t0\t1362\n",
      "267\t1331\t0\t1331\n",
      "268\t1350\t0\t1350\n",
      "269\t1356\t0\t1356\n",
      "270\t1348\t0\t1348\n",
      "271\t1314\t0\t1314\n",
      "272\t1585\t0\t1585\n",
      "273\t1601\t0\t1601\n",
      "274\t1967\t0\t1967\n",
      "275\t1919\t0\t1919\n",
      "276\t2100\t0\t2100\n",
      "277\t2221\t0\t2221\n",
      "278\t2388\t0\t2388\n",
      "279\t2555\t0\t2555\n",
      "280\t3100\t0\t3100\n",
      "281\t3222\t0\t3222\n",
      "282\t3670\t0\t3670\n",
      "283\t4142\t0\t4142\n",
      "284\t5356\t0\t5356\n",
      "285\t6842\t0\t6842\n",
      "286\t9332\t0\t9332\n",
      "287\t13005\t0\t13005\n",
      "288\t15461\t0\t15461\n",
      "289\t35812\t0\t35812\n",
      "290\t51275\t0\t51275\n"
     ]
    }
   ],
   "source": [
    "!cat SRR18218015_2.settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "01b19757-799b-49af-8432-5419d1325bdc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AdapterRemoval ver. 2.3.3\n",
      "\n",
      "This program searches for and removes remnant adapter sequences from\n",
      "your read data.  The program can analyze both single end and paired end\n",
      "data.  For detailed explanation of the parameters, please refer to the\n",
      "man page.  For comments, suggestions  and feedback please contact Stinus\n",
      "Lindgreen (stinus@binf.ku.dk) and Mikkel Schubert (MikkelSch@gmail.com).\n",
      "\n",
      "If you use the program, please cite the paper:\n",
      "    Schubert, Lindgreen, and Orlando (2016). AdapterRemoval v2: rapid\n",
      "    adapter trimming, identification, and read merging.\n",
      "    BMC Research Notes, 12;9(1):88.\n",
      "\n",
      "    http://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-016-1900-2\n",
      "\n",
      "\n",
      "OPTIONS:\n",
      "    --help  \n",
      "        Display this message.\n",
      "\n",
      "    --version  \n",
      "        Print the version string.\n",
      "\n",
      "\n",
      "    --file1 FILE [FILE ...]\n",
      "        Input files containing mate 1 reads or single-ended reads; one or\n",
      "        more files may be listed [REQUIRED].\n",
      "\n",
      "    --file2 [FILE ...]\n",
      "        Input files containing mate 2 reads; if used, then the same number of\n",
      "        files as --file1 must be listed [OPTIONAL].\n",
      "\n",
      "    --identify-adapters  \n",
      "        Attempt to identify the adapter pair of PE reads, by searching for\n",
      "        overlapping mate reads [default: off].\n",
      "\n",
      "    --threads THREADS\n",
      "        Maximum number of threads [default: 1]\n",
      "\n",
      "\n",
      "FASTQ OPTIONS:\n",
      "    --qualitybase BASE\n",
      "        Quality base used to encode Phred scores in input; either 33, 64, or\n",
      "        solexa [default: 33].\n",
      "\n",
      "    --qualitybase-output BASE\n",
      "        Quality base used to encode Phred scores in output; either 33, 64, or\n",
      "        solexa. By default, reads will be written in the same format as the\n",
      "        that specified using --qualitybase.\n",
      "\n",
      "    --qualitymax BASE\n",
      "        Specifies the maximum Phred score expected in input files, and used\n",
      "        when writing output. ASCII encoded values are limited to the\n",
      "        characters '!' (ASCII = 33) to '~' (ASCII = 126), meaning that\n",
      "        possible scores are 0 - 93 with offset 33, and 0 - 62 for offset 64\n",
      "        and Solexa scores [default: 41].\n",
      "\n",
      "    --mate-separator CHAR\n",
      "        Character separating the mate number (1 or 2) from the read name in\n",
      "        FASTQ records [default: '/'].\n",
      "\n",
      "    --interleaved  \n",
      "        This option enables both the --interleaved-input option and the\n",
      "        --interleaved-output option [default: off].\n",
      "\n",
      "    --interleaved-input  \n",
      "        The (single) input file provided contains both the mate 1 and mate 2\n",
      "        reads, one pair after the other, with one mate 1 reads followed by\n",
      "        one mate 2 read. This option is implied by the --interleaved option\n",
      "        [default: off].\n",
      "\n",
      "    --interleaved-output  \n",
      "        If set, trimmed paired-end reads are written to a single file\n",
      "        containing mate 1 and mate 2 reads, one pair after the other. This\n",
      "        option is implied by the --interleaved option [default: off].\n",
      "\n",
      "    --combined-output  \n",
      "        If set, all reads are written to the same file(s), specified by\n",
      "        --output1 and --output2 (--output1 only if --interleaved-output is\n",
      "        not set). Discarded reads are replaced with a single 'N' with Phred\n",
      "        score 0 [default: off].\n",
      "\n",
      "\n",
      "OUTPUT FILES:\n",
      "    --basename BASENAME\n",
      "        Default prefix for all output files for which no filename was\n",
      "        explicitly set [default: your_output].\n",
      "\n",
      "    --settings FILE\n",
      "        Output file containing information on the parameters used in the run\n",
      "        as well as overall statistics on the reads after trimming [default:\n",
      "        BASENAME.settings]\n",
      "\n",
      "    --output1 FILE\n",
      "        Output file containing trimmed mate1 reads [default:\n",
      "        BASENAME.pair1.truncated (PE), BASENAME.truncated (SE), or\n",
      "        BASENAME.paired.truncated (interleaved PE)]\n",
      "\n",
      "    --output2 FILE\n",
      "        Output file containing trimmed mate 2 reads [default:\n",
      "        BASENAME.pair2.truncated (only used in PE mode, but not if\n",
      "        --interleaved-output is enabled)]\n",
      "\n",
      "    --singleton FILE\n",
      "        Output file to which containing paired reads for which the mate has\n",
      "        been discarded [default: BASENAME.singleton.truncated]\n",
      "\n",
      "    --outputcollapsed FILE\n",
      "        If --collapsed is set, contains overlapping mate-pairs which have\n",
      "        been merged into a single read (PE mode) or reads for which the\n",
      "        adapter was identified by a minimum overlap, indicating that the\n",
      "        entire template molecule is present. This does not include which have\n",
      "        subsequently been trimmed due to low-quality or ambiguous nucleotides\n",
      "        [default: BASENAME.collapsed]\n",
      "\n",
      "    --outputcollapsedtruncated FILE\n",
      "        Collapsed reads (see --outputcollapsed) which were trimmed due the\n",
      "        presence of low-quality or ambiguous nucleotides [default:\n",
      "        BASENAME.collapsed.truncated]\n",
      "\n",
      "    --discarded FILE\n",
      "        Contains reads discarded due to the --minlength, --maxlength or\n",
      "        --maxns options [default: BASENAME.discarded]\n",
      "\n",
      "\n",
      "OUTPUT COMPRESSION:\n",
      "    --gzip  \n",
      "        Enable gzip compression [default: off]\n",
      "\n",
      "    --gzip-level LEVEL\n",
      "        Compression level, 0 - 9 [default: 6]\n",
      "\n",
      "    --bzip2  \n",
      "        Enable bzip2 compression [default: off]\n",
      "\n",
      "    --bzip2-level LEVEL\n",
      "        Compression level, 0 - 9 [default: 9]\n",
      "\n",
      "\n",
      "TRIMMING SETTINGS:\n",
      "    --adapter1 SEQUENCE\n",
      "        Adapter sequence expected to be found in mate 1 reads [default:\n",
      "        AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG].\n",
      "\n",
      "    --adapter2 SEQUENCE\n",
      "        Adapter sequence expected to be found in mate 2 reads [default:\n",
      "        AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT].\n",
      "\n",
      "    --adapter-list FILENAME\n",
      "        Read table of white-space separated adapters pairs, used as if the\n",
      "        first column was supplied to --adapter1, and the second column was\n",
      "        supplied to --adapter2; only the first adapter in each pair is\n",
      "        required SE trimming mode [default: <not set>].\n",
      "\n",
      "\n",
      "    --minadapteroverlap LENGTH\n",
      "        In single-end mode, reads are only trimmed if the overlap between\n",
      "        read and the adapter is at least X bases long, not counting ambiguous\n",
      "        nucleotides (N); this is independent of the --minalignmentlength when\n",
      "        using --collapse, allowing a conservative selection of putative\n",
      "        complete inserts while ensuring that all possible adapter\n",
      "        contamination is trimmed [default: 0].\n",
      "\n",
      "    --mm MISMATCH_RATE\n",
      "        Max error-rate when aligning reads and/or adapters. If > 1, the max\n",
      "        error-rate is set to 1 / MISMATCH_RATE; if < 0, the defaults are\n",
      "        used, otherwise the user-supplied value is used directly [default:\n",
      "        1/3 for trimming; 1/10 when identifying adapters].\n",
      "\n",
      "    --shift N\n",
      "        Consider alignments where up to N nucleotides are missing from the 5'\n",
      "        termini [default: 2].\n",
      "\n",
      "\n",
      "    --trim5p N [N]\n",
      "        Trim the 5' of reads by a fixed amount after removing adapters, but\n",
      "        before carrying out quality based trimming. Specify one value to trim\n",
      "        mate 1 and mate 2 reads the same amount, or two values separated by a\n",
      "        space to trim each mate different amounts [default: no trimming].\n",
      "\n",
      "    --trim3p N [N]\n",
      "        Trim the 3' of reads by a fixed amount. See --trim5p.\n",
      "\n",
      "    --trimns  \n",
      "        If set, trim ambiguous bases (N) at 5'/3' termini [default: off]\n",
      "\n",
      "    --maxns MAX\n",
      "        Reads containing more ambiguous bases (N) than this number after\n",
      "        trimming are discarded [default: 1000].\n",
      "\n",
      "    --trimqualities  \n",
      "        If set, trim bases at 5'/3' termini with quality scores <= to\n",
      "        --minquality value [default: off]\n",
      "\n",
      "    --trimwindows INT\n",
      "        If set, quality trimming will be carried out using window based\n",
      "        approach, where windows with an average quality less than\n",
      "        --minquality will be trimmed. If >= 1, this value will be used as the\n",
      "        window size. If the value is < 1, the value will be multiplied with\n",
      "        the read length to determine a window size per read. If the resulting\n",
      "        window size is 0 or larger than the read length, the read length is\n",
      "        used as the window size. This option implies --trimqualities\n",
      "        [default: <not set>].\n",
      "\n",
      "    --minquality PHRED\n",
      "        Inclusive minimum; see --trimqualities for details [default: 2]\n",
      "\n",
      "    --preserve5p  \n",
      "        If set, bases at the 5p will not be trimmed by --trimns,\n",
      "        --trimqualities, and --trimwindows. Collapsed reads will not be\n",
      "        quality trimmed when this option is enabled [default: 5p bases are\n",
      "        trimmed]\n",
      "\n",
      "\n",
      "    --minlength LENGTH\n",
      "        Reads shorter than this length are discarded following trimming\n",
      "        [default: 15].\n",
      "\n",
      "    --maxlength LENGTH\n",
      "        Reads longer than this length are discarded following trimming\n",
      "        [default: 4294967295].\n",
      "\n",
      "\n",
      "READ MERGING:\n",
      "    --collapse  \n",
      "        When set, paired ended read alignments of --minalignmentlength or\n",
      "        more bases are combined into a single consensus sequence,\n",
      "        representing the complete insert, and written to either\n",
      "        basename.collapsed or basename.collapsed.truncated (if trimmed due to\n",
      "        low-quality bases following collapse); for single-ended reads,\n",
      "        putative complete inserts are identified as having at least\n",
      "        --minalignmentlength bases overlap with the adapter sequence, and are\n",
      "        written to the the same files [default: off].\n",
      "\n",
      "    --collapse-deterministic  \n",
      "        In standard --collapse mode, AdapterRemoval will randomly select one\n",
      "        of two different overlapping bases if these have the same quality\n",
      "        (otherwise it picks the highest quality base). With\n",
      "        --collapse-deterministic, AdapterRemoval will instead set such bases\n",
      "        to N. Setting this option also sets--collapse [default: off].\n",
      "\n",
      "    --collapse-conservatively  \n",
      "        Enables a more conservative merging algorithm inspired by fastq-join,\n",
      "        in which the higher quality score is picked for matching bases and\n",
      "        the max score minus the min score is picked for mismatching bases.\n",
      "        For more details, see the documentation. --seed and\n",
      "        --collapse-deterministic have no effect when this is enabled. Setting\n",
      "        this option also sets --collapse [default: off].\n",
      "\n",
      "    --minalignmentlength LENGTH\n",
      "        If --collapse is set, paired reads must overlap at least this number\n",
      "        of bases to be collapsed, and single-ended reads must overlap at\n",
      "        least this number of bases with the adapter to be considered complete\n",
      "        template molecules [default: 11].\n",
      "\n",
      "    --seed SEED\n",
      "        Sets the RNG seed used when choosing between bases with equal Phred\n",
      "        scores when --collapse is enabled. This option is not available if\n",
      "        more than one thread is used. If not specified, aseed is generated\n",
      "        using the current time.\n",
      "\n",
      "\n",
      "DEMULTIPLEXING:\n",
      "    --barcode-list FILENAME\n",
      "        List of barcodes or barcode pairs for single or double-indexed\n",
      "        demultiplexing. Note that both indexes should be specified for both\n",
      "        single-end and paired-end trimming, if double-indexed multiplexing\n",
      "        was used, in order to ensure that the demultiplexed reads can be\n",
      "        trimmed correctly [default: <not set>].\n",
      "\n",
      "    --barcode-mm N\n",
      "        Maximum number of mismatches allowed when counting mismatches in both\n",
      "        the mate 1 and the mate 2 barcode for paired reads.\n",
      "\n",
      "    --barcode-mm-r1 N\n",
      "        Maximum number of mismatches allowed for the mate 1 barcode; if not\n",
      "        set, this value is equal to the '--barcode-mm' value; cannot be\n",
      "        higher than the '--barcode-mm value'.\n",
      "\n",
      "    --barcode-mm-r2 N\n",
      "        Maximum number of mismatches allowed for the mate 2 barcode; if not\n",
      "        set, this value is equal to the '--barcode-mm' value; cannot be\n",
      "        higher than the '--barcode-mm value'.\n",
      "\n",
      "    --demultiplex-only  \n",
      "        Only carry out demultiplexing using the list of barcodes supplied\n",
      "        with --barcode-list. No other processing is done.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!AdapterRemoval --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "c48ecf1d-df0b-4adc-a43e-0b4ae0a0f1b9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Attempting to identify adapter sequences ...\n",
      "Opening FASTQ file 'SRR18218015_1.fastq', line numbers start at 1\n",
      "Opening FASTQ file 'SRR18218015_2.fastq', line numbers start at 1\n",
      "Processed a total of 1,410,922 reads in 8.3s; 169,000 reads per second on average ...\n",
      "   Found 673617 overlapping pairs ...\n",
      "   Of which 22581 contained adapter sequence(s) ...\n",
      "\n",
      "Printing adapter sequences, including poly-A tails:\n",
      "  --adapter1:  AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG\n",
      "                  ||       ||||   | |  | ||||  ||****** |    | |     | | | ||  \n",
      "   Consensus:  CTGTCTCTTATACACATTCCCGGCCCCCAGGAACCCCGGTTTTATCTCTTTTCCCGTTTTCTGTTTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANAAAAAAAAACAAAAACAAAAATANAATTAAATT\n",
      "     Quality:  /--+,*++)&&&%%$##%%%%\"%\"%.%%\"\"%%\"\"%%%%%\"%%%\"%%\"\"\"%\"\"%%#\"%%%%##\"$%\"%01334120-*(('&&%%%$$$$$$$#$####################################\"##########\"#\"#\"#######\"####\"##\"##\"###\"###\"##\"#\"\"##\"\"###\"\"##\"\"\"\"\"##\"#\"\"\"\"\"#\"\"\"#\"##\"#\"##\"\"\"##\"#\"\"##\"\"\"#\"##\"\"\"#\"\"\"\"\"\"###\"#\"\"\"\"##\"##\"\"###$%\n",
      "\n",
      "    Top 5 most common 9-bp 5'-kmers:\n",
      "            1: CTGTCTCTT = 57.71% (790)\n",
      "            2: CTGTCTCTA =  7.89% (108)\n",
      "            3: CTGTTCTCT =  1.83% (25)\n",
      "            4: CTGTCTCTG =  1.46% (20)\n",
      "            5: CTGTCCCTT =  1.24% (17)\n",
      "\n",
      "\n",
      "  --adapter2:  AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "                  ||       |       | |        |   ||    |||     |       |\n",
      "   Consensus:  CTGTCTCTTATACACATCTGACGCTGCCGACGACAATTAACGTGTAGATCTCGGTGGTCGCCGGATCATTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACACAAAAAAAAACACACAACCCACCCTCACCCCCTTTCCTCATCCCCCACTATTAATACTCACACTCTTCCCACNATCTCCATTCTATTCCTCTTTTANCNCTCTTTTTNACCTNTAN\n",
      "     Quality:  ,(*'*%,)-+()(&(%&'&'(('(&&)+'%'&%&&'&'&'%$$#$$######$%#$%###$$#\"\"\"#\"#$#&&''''''&&%$$$########\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"#########\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"\"#\n",
      "\n",
      "    Top 5 most common 9-bp 5'-kmers:\n",
      "            1: CTGTCTCTT = 27.63% (4259)\n",
      "            2: CTGTCCCTT = 13.43% (2071)\n",
      "            3: CTGGCTCTT =  3.99% (615)\n",
      "            4: CTGTCTCCT =  2.35% (363)\n",
      "            5: CCGTCTCTT =  1.98% (305)\n"
     ]
    }
   ],
   "source": [
    "!AdapterRemoval --identify-adapters --file1 SRR18218015_1.fastq --file2 SRR18218015_2.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3a18bdc3-f027-4dbb-8ee2-3efaa2f14401",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Trimming paired end reads ...\n",
      "Opening FASTQ file 'SRR18218015_1.fastq', line numbers start at 1\n",
      "Opening FASTQ file 'SRR18218015_2.fastq', line numbers start at 1\n",
      "Processed a total of 1,410,922 reads in 3.8s; 371,000 reads per second on average ...\n"
     ]
    }
   ],
   "source": [
    "!AdapterRemoval --file1 SRR18218015_1.fastq --file2 SRR18218015_2.fastq --basename SRR18218015_pair* --qualitybase 33 --trimqualities --trimns --minquality 20 --minlength 50 --trim5p 19 --adapter1 CTGTCTCTTATACACATTC --adapter2 CTGTCTCTTATACACATCT --threads 7 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "de135557-f833-4b74-ad16-dcdd8624f164",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "3302cc37-973c-4f7f-b432-29fbfd3189f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files\n"
     ]
    }
   ],
   "source": [
    "cd fastqc/trial4files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "cc9e09d4-940f-47ba-902f-e3c383b37f05",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "null\n",
      "null\n",
      "Started analysis of SRR18218015_.pair1.truncated\n",
      "Approx 5% complete for SRR18218015_.pair1.truncated\n",
      "Approx 10% complete for SRR18218015_.pair1.truncated\n",
      "Approx 15% complete for SRR18218015_.pair1.truncated\n",
      "Approx 20% complete for SRR18218015_.pair1.truncated\n",
      "Approx 25% complete for SRR18218015_.pair1.truncated\n",
      "Approx 30% complete for SRR18218015_.pair1.truncated\n",
      "Approx 35% complete for SRR18218015_.pair1.truncated\n",
      "Approx 40% complete for SRR18218015_.pair1.truncated\n",
      "Approx 45% complete for SRR18218015_.pair1.truncated\n",
      "Approx 50% complete for SRR18218015_.pair1.truncated\n",
      "Approx 55% complete for SRR18218015_.pair1.truncated\n",
      "Approx 60% complete for SRR18218015_.pair1.truncated\n",
      "Approx 65% complete for SRR18218015_.pair1.truncated\n",
      "Approx 70% complete for SRR18218015_.pair1.truncated\n",
      "Approx 75% complete for SRR18218015_.pair1.truncated\n",
      "Approx 80% complete for SRR18218015_.pair1.truncated\n",
      "Approx 85% complete for SRR18218015_.pair1.truncated\n",
      "Approx 90% complete for SRR18218015_.pair1.truncated\n",
      "Approx 95% complete for SRR18218015_.pair1.truncated\n",
      "Analysis complete for SRR18218015_.pair1.truncated\n",
      "Started analysis of SRR18218015_.pair2.truncated\n",
      "Approx 5% complete for SRR18218015_.pair2.truncated\n",
      "Approx 10% complete for SRR18218015_.pair2.truncated\n",
      "Approx 15% complete for SRR18218015_.pair2.truncated\n",
      "Approx 20% complete for SRR18218015_.pair2.truncated\n",
      "Approx 25% complete for SRR18218015_.pair2.truncated\n",
      "Approx 30% complete for SRR18218015_.pair2.truncated\n",
      "Approx 35% complete for SRR18218015_.pair2.truncated\n",
      "Approx 40% complete for SRR18218015_.pair2.truncated\n",
      "Approx 45% complete for SRR18218015_.pair2.truncated\n",
      "Approx 50% complete for SRR18218015_.pair2.truncated\n",
      "Approx 55% complete for SRR18218015_.pair2.truncated\n",
      "Approx 60% complete for SRR18218015_.pair2.truncated\n",
      "Approx 65% complete for SRR18218015_.pair2.truncated\n",
      "Approx 70% complete for SRR18218015_.pair2.truncated\n",
      "Approx 75% complete for SRR18218015_.pair2.truncated\n",
      "Approx 80% complete for SRR18218015_.pair2.truncated\n",
      "Approx 85% complete for SRR18218015_.pair2.truncated\n",
      "Approx 90% complete for SRR18218015_.pair2.truncated\n",
      "Approx 95% complete for SRR18218015_.pair2.truncated\n",
      "Analysis complete for SRR18218015_.pair2.truncated\n"
     ]
    }
   ],
   "source": [
    "!fastqc SRR18218015_.pair1.truncated SRR18218015_.pair2.truncated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c0d5b313-922e-4141-b05d-117e5f088cb5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SRR18218015.1 1 length=106\n",
      "GGATGCGACTGACAGTCTNNNNNNNTGAGGCCNAGNNNNTGAGTCANNNCNNNGAGGCTGNNNNNNNTGTTGACNNNGGNCTTGCTA\n",
      "+\n",
      "GGGGGGGGGGGGGGGGGF#######::CFDFG#:C####::CFDDF###:###::DFGGG#######::CCDGG###::#:CFFFGG\n",
      "@SRR18218015.2 2 length=300\n",
      "TGAGCCTTACGCATGGCTNNNNNNNGCAGACANCANNNNCAATGAGNNTGNNNAACACAANNNNNNNCTGACATNNNACNTGAGCAACCCTAAGGTCCGTAAGATGATTNNNGNCTTACANANNNNTNNNGNCGANGCGTATGCGAANNCACTCGCAGANNNNNNGAAGNNNCCACCTCAGNTTCAGCNTGGCNNNNNNNCTATTGAACCACGAGAAGNCGACATGCCNNGGANTGAGAATNGTGACGGTACTNTTACATTGAAGTTTAAATGCTTC\n",
      "+\n",
      "GFGGGGGGGGGGGGGGGG#######::CDFGG#:C####::DGFFE##::###::CDEGG#######::CFFGG###::#:CDFGGGGGGGGGGGGGGGGGGDFGBFGG###:#:BFGGG#8####:###:#::B#::AFFE8FFDE##::DFGFGGGG######38@6###66DFGGGGG#6@FGFG#6=:F#######44=FGGFFGCFGGGGGGC#4:;EGGGGG##228#2;8FGGC#2:CFGGGGGGG#2;EGGGGF?FCAFFGGGGGGGGC\n",
      "@SRR18218015.4 4 length=301\n",
      "GATTCATGCGCAACCTCGNNNNNNNCTGGGACNTANNNNCACTTGGNNGCNNCTTGTAATNTNNNNNATCAGTAANNCANGTAAGGCGCGTCAGCAGCGCGAAAGGAGCNNGGNGTTGGAAGANCNCNNNTNCAANATCGCTGAGCATAAACCACCTGCTNNNNNCAAGNGNAGCATTAAAGATATTCGTTCCGNNNNNNTCCTGGACACCGCTATTGNCGTTAACGGGAAAGTCACTACTCGTCGCAAGCGCATCGACGATAAATTGATGTCGGGT\n"
     ]
    }
   ],
   "source": [
    "!head SRR18218015_.pair1.truncated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "4defa02e-9f7d-41d6-9e06-ab069ec28412",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat SRR18218015_.pair*.truncated | jellyfish count -t 6 -m 15 -s 1000000000 -o SRR18218015Jelly -C /dev/fd/0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "83549389-1e3c-4d3c-b1ca-d5b149b30211",
   "metadata": {},
   "outputs": [],
   "source": [
    "!jellyfish histo SRR18218015Jelly > SRR18218015Jelly.histo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "f93a8372-b879-483b-8995-c59194fd9ff5",
   "metadata": {},
   "outputs": [],
   "source": [
    "!ln -s SRR18218015_.pair1.truncated SRR18218015_.pair1.truncated.fastq\n",
    "!ln -s SRR18218015_.pair2.truncated SRR18218015_.pair2.truncated.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "30536a0e-cc2d-433a-9120-0321233696e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files'"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "ebbc4cc1-20fc-4a0c-a741-c280e39c3762",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lrwxrwxrwx 1 mghalib mghalib 28 Jul 19 16:34 \u001b[0m\u001b[01;36mSRR18218015_.pair1.truncated.fastq\u001b[0m -> SRR18218015_.pair1.truncated\n"
     ]
    }
   ],
   "source": [
    "ls -l SRR18218015_.pair1.truncated.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "e21a9fda-f4d4-48c8-85c1-78b779274d5b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades.py\t--isolate\t-1\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair1.truncated.fastq\t-2\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair2.truncated.fastq\t-o\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX\t-k\t41\t-t\t6\t-m\t12\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.5\n",
      "  Python version: 3.10.8\n",
      "  OS: Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35\n",
      "\n",
      "Output dir: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX\n",
      "Mode: ONLY assembling (without read error correction)\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Isolate mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair1.truncated.fastq']\n",
      "      right reads: ['/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair2.truncated.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Assembly parameters:\n",
      "  k: [41]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/tmp\n",
      "  Threads: 6\n",
      "  Memory limit (in Gb): 12\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/spades.log\n",
      "\n",
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair1.truncated.fastq: max reads length: 282\n",
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/SRR18218015_.pair2.truncated.fastq: max reads length: 282\n",
      "\n",
      "Reads length: 282\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K41 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/configs/config.info /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/configs/isolate_mode.info\n",
      "\n",
      "  0:00:00.000     1M / 26M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/configs/config.info\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/configs/isolate_mode.info\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 12 Gb\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/dataset.info) with K=41\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 6\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.001     1M / 26M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.001     1M / 26M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.001     1M / 26M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.003     1M / 26M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.004     1M / 26M   INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.005     1M / 26M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.005     1M / 26M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.005     1M / 26M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.388    69M / 113M  INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.406    75M / 124M  INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:00.436    87M / 149M  INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.782    74M / 191M  INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.416    86M / 211M  INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.228    72M / 243M  INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:02.679    63M / 243M  INFO    General                 (binary_converter.cpp      : 111)   623525 reads written\n",
      "  0:00:02.679    41M / 243M  INFO    General                 (read_converter.cpp        :  86)   Converting single reads\n",
      "  0:00:02.680    51M / 243M  INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:02.680    41M / 243M  INFO    General                 (read_converter.cpp        :  92)   Converting merged reads\n",
      "  0:00:02.680    51M / 243M  INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:02.687     1M / 243M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:02.687     1M / 243M  INFO    General                 (construction.cpp          : 153)   Max read length 282\n",
      "  0:00:02.687     1M / 243M  INFO    General                 (construction.cpp          : 159)   Average read length 177.704\n",
      "  0:00:02.687     1M / 243M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:02.688     1M / 243M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 60 files using 6 threads. This might take a while.\n",
      "  0:00:02.688     1M / 243M  INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:02.688     1M / 243M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.66665 Gb\n",
      "  0:00:02.688     1M / 243M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 559240\n",
      "  0:00:08.449  3601M / 3601M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2494100 reads\n",
      "  0:00:08.450     1M / 3445M INFO    General                 (kmer_splitters.hpp        : 131)   Used 2494100 reads\n",
      "  0:00:08.454     1M / 3445M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:08.489     1M / 3445M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 6189256 kmers in total.\n",
      "  0:00:08.489     1M / 3445M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:08.489     1M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:08.489     1M / 3445M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 60 files using 6 threads. This might take a while.\n",
      "  0:00:08.490     1M / 3445M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:08.490     1M / 3445M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.666647 Gb\n",
      "  0:00:08.491     1M / 3445M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 559240\n",
      "  0:00:09.121  3601M / 3601M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 6189256 kmers\n",
      "  0:00:09.121  3601M / 3601M INFO    General                 (kmer_splitters.hpp        : 199)   Used 6189256 kmers.\n",
      "  0:00:09.122     1M / 3445M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:09.153     1M / 3445M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 6119302 kmers in total.\n",
      "  0:00:09.153     1M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:09.298     6M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 6119302 kmers, 4465048 bytes occupied (5.83733 bits per kmer).\n",
      "  0:00:09.298     6M / 3445M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:09.360    12M / 3445M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:09.838    12M / 3445M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:09.843    12M / 3445M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:09.843    12M / 3445M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:10.473    25M / 3445M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 282457 sequences extracted\n",
      "  0:00:10.972    25M / 3445M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:11.115    25M / 3445M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:11.129    25M / 3445M INFO    General                 (debruijn_graph_constructor: 487)   Total 564914 edges to create\n",
      "  0:00:11.129    47M / 3445M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:11.220    56M / 3445M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:11.239    56M / 3445M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:11.243    58M / 3445M INFO    General                 (debruijn_graph_constructor: 503)   Total 212503 vertices to create\n",
      "  0:00:11.243    78M / 3445M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:11.330    63M / 3445M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:11.330    63M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:11.507    68M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 6189256 kmers, 4515120 bytes occupied (5.83607 bits per kmer).\n",
      "  0:00:11.508    92M / 3445M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:18.224    92M / 3445M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:18.978    92M / 3445M INFO    General                 (coverage_filling.hpp      :  82)   Processed 564914 edges\n",
      "  0:00:19.012    52M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:19.013    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 21\n",
      "  0:00:19.013    52M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 68\n",
      "  0:00:19.013    52M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 72. Coverage mad: 17.7912\n",
      "  0:00:19.013    52M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:19.047    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:19.118    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:19.340    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:19.568    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 73.0746. Fitted coverage std. dev: 16.2469\n",
      "  0:00:19.570    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.983636\n",
      "  0:00:19.570    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 42\n",
      "  0:00:19.570    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 42\n",
      "  0:00:19.570    53M / 3445M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 38008\n",
      "  0:00:19.570    52M / 3445M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 73.0746\n",
      "  0:00:19.570    52M / 3445M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 42\n",
      "  0:00:19.570    52M / 3445M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:19.570    52M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:00:19.576    61M / 3445M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:19.601    64M / 3445M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 90236 out of 564914, length: 3578938\n",
      "  0:00:19.610    65M / 3445M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 567741)\n",
      "  0:00:19.674    65M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:20.090    68M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 3011372 kmers, 2175744 bytes occupied (5.78007 bits per kmer).\n",
      "  0:00:20.094    91M / 3445M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:29.431    90M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 623525 reads processed\n",
      "  0:00:29.438    62M / 3445M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:29.438    62M / 3445M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:29.641    62M / 3445M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 4 gaps after checking 666 candidates\n",
      "  0:00:29.648    62M / 3445M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:29.650    52M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:29.652    52M / 3445M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:29.653    52M / 3445M INFO    General                 (simplification.cpp        :  61)   Most init cleaning disabled on main iteration\n",
      "  0:00:29.653    52M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:29.660    52M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:29.660    52M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:29.660    52M / 3445M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:29.660    52M / 3445M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:29.660    52M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:29.660    52M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:29.912    50M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 79809 times\n",
      "  0:00:29.914    50M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.548   244M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 45278 times\n",
      "  0:00:39.549   244M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.720   243M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 26207 times\n",
      "  0:00:39.722   243M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:39.722   243M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.734   240M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1375 times\n",
      "  0:00:39.734   240M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.985   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2078 times\n",
      "  0:00:39.988   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.988   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 28 times\n",
      "  0:00:39.988   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:39.988   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.988   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.988   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.989   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:00:39.989   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.989   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:00:39.989   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:39.989   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.990   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.990   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.990   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.990   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.990   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:00:39.990   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.991   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.991   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.992   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.992   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.993   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.993   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.993   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.994   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.994   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:39.995   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.995   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.995   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:39.995   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:39.995   247M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:39.995   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:39.998   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:00:39.998   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:39.999   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:39.999   247M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.001   246M / 3445M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:40.001   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:40.003   246M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:00:40.007   255M / 3445M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:40.009   255M / 3445M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 7286 out of 7322, length: 1319770\n",
      "  0:00:40.010   255M / 3445M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 567741)\n",
      "  0:00:40.017   255M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:40.157   256M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 659885 kmers, 477336 bytes occupied (5.7869 bits per kmer).\n",
      "  0:00:40.159   261M / 3445M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:56.228   261M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 623525 reads processed\n",
      "  0:00:56.232   255M / 3445M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:56.233   255M / 3445M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:56.237   255M / 3445M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 1 gaps after checking 15 candidates\n",
      "  0:00:56.239   255M / 3445M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:56.240   246M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:56.240   246M / 3445M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:56.240   246M / 3445M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:56.240   246M / 3445M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:56.241   246M / 3445M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:56.241   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:56.242   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:56.242   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:56.244   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:56.244   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:56.245   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:56.245   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:56.247   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:56.248   246M / 3445M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:56.250   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:56.255   246M / 3445M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 3457 times\n",
      "  0:00:56.257   246M / 3445M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:56.257   246M / 3445M INFO    General                 (simplification.cpp        : 496)     Average coverage = 853.038\n",
      "  0:00:56.258   246M / 3445M INFO    General                 (simplification.cpp        : 497)     Total length = 194320\n",
      "  0:00:56.261   246M / 3445M INFO    General                 (simplification.cpp        : 498)     Median edge length: 2311\n",
      "  0:00:56.261   246M / 3445M INFO    General                 (simplification.cpp        : 499)     Edges: 406\n",
      "  0:00:56.261   246M / 3445M INFO    General                 (simplification.cpp        : 500)     Vertices: 764\n",
      "  0:00:56.261   246M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:00:56.261   246M / 3445M INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:00:56.261   246M / 3445M INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 567741)\n",
      "  0:00:56.264   246M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:56.281   246M / 3445M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 194320 kmers, 141096 bytes occupied (5.80881 bits per kmer).\n",
      "  0:00:56.281   247M / 3445M INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:56.291   247M / 3445M INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 4240224 kmers to process\n",
      "  0:00:57.126   247M / 3445M INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:00:57.127   247M / 3445M INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:00:58.204   248M / 3445M INFO    General                 (mismatch_correction.cpp   : 192)   Total 126 edges (out of 406) with 73105 potential mismatch positions (580.198 positions per edge)\n",
      "  0:00:58.205   248M / 3445M INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:01:04.546   254M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:04.605   260M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:04.768   266M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:04.852   272M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:05.304   278M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:05.601   283M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:01:11.822   283M / 3445M INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2262900 reads\n",
      "  0:01:13.239   289M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2494100 reads processed\n",
      "  0:01:13.243   254M / 3445M INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:01:13.244   247M / 3445M INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 0 nucleotides\n",
      "  0:01:13.244   247M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:13.244   247M / 3445M INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/before_rr.fasta\n",
      "  0:01:13.247   247M / 3445M INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph_after_simplification.gfa\n",
      "  0:01:13.249   247M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:01:13.252   248M / 3445M INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 4240224 kmers to process\n",
      "  0:01:13.252   248M / 3445M INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:01:13.253   248M / 3445M INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 2311\n",
      "  0:01:13.253   248M / 3445M INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:01:13.253   248M / 3445M INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  0:01:13.267   360M / 3445M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:01:16.157   360M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 623525 reads processed\n",
      "  0:01:16.238   360M / 3445M INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 639\n",
      "  0:01:16.239   360M / 3445M INFO    General                 (pair_info_count.cpp       : 222)   594775 paired reads (95.3891% of all) aligned to long edges\n",
      "  0:01:16.240   248M / 3445M INFO    General                 (pair_info_count.cpp       : 369)     Insert size = 184.682, deviation = 134.84, left quantile = 54, right quantile = 366, read length = 282\n",
      "  0:01:16.240   248M / 3445M WARN    General                 (pair_info_count.cpp       : 376)   Estimated mean insert size 184.682 is very small compared to read length 282\n",
      "  0:01:16.240   248M / 3445M INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:01:16.240   248M / 3445M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:01:19.080   248M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 623525 reads processed\n",
      "  0:01:19.081   248M / 3445M INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:01:19.081   248M / 3445M INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:01:19.081   248M / 3445M INFO    General                 (pair_info_count.cpp       : 299)   Left insert size quantile 54, right insert size quantile 366, filtering threshold 2, rounding threshold 0\n",
      "  0:01:19.083   257M / 3445M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:01:22.050   257M / 3445M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 623525 reads processed\n",
      "  0:01:22.055   248M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:01:22.055   248M / 3445M INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:01:22.055   248M / 3445M INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:01:22.058   248M / 3445M INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:01:22.066   248M / 3445M INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:01:22.066   248M / 3445M INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 4948\n",
      "  0:01:22.067   248M / 3445M INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 1320\n",
      "  0:01:22.067   248M / 3445M INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:01:22.068   248M / 3445M INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:01:22.068   248M / 3445M INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:01:22.071   248M / 3445M INFO   PairInfoImprover         (pair_info_improver.hpp    : 104)   Paired info stats: missing = 16; contradictional = 152\n",
      "  0:01:22.074   248M / 3445M INFO   PairInfoImprover         (pair_info_improver.hpp    : 104)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:01:22.074   248M / 3445M INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:01:22.075   248M / 3445M INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:01:22.082   248M / 3445M INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:01:22.082   248M / 3445M INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 407\n",
      "  0:01:22.083   248M / 3445M INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 407\n",
      "  0:01:22.083   248M / 3445M INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:01:22.083   248M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:01:22.083   248M / 3445M INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:01:22.084   248M / 3445M INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:01:22.086   248M / 3445M INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:01:22.087   248M / 3445M INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 853.038\n",
      "  0:01:22.087   248M / 3445M INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 1200\n",
      "  0:01:22.087   248M / 3445M INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 853.038\n",
      "  0:01:22.088   248M / 3445M INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 1200\n",
      "  0:01:22.089   248M / 3445M INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:01:22.089   248M / 3445M INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:01:22.089   248M / 3445M INFO    General                 (extenders_logic.cpp       : 545)   Using 0 single read libraries\n",
      "  0:01:22.090   248M / 3445M INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 3\n",
      "  0:01:22.090   248M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 196 (0%)\n",
      "  0:01:22.090   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 20 paths from 196 (10%)\n",
      "  0:01:22.091   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 196 (20%)\n",
      "  0:01:22.091   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 60 paths from 196 (30%)\n",
      "  0:01:22.091   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 196 (40%)\n",
      "  0:01:22.092   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 100 paths from 196 (51%)\n",
      "  0:01:22.092   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 196 (61%)\n",
      "  0:01:22.092   249M / 3445M INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 196 (65%)\n",
      "  0:01:22.093   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 140 paths from 196 (71%)\n",
      "  0:01:22.093   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 196 (81%)\n",
      "  0:01:22.093   249M / 3445M INFO    General                 (path_extenders.cpp        :  36)   Processed 180 paths from 196 (91%)\n",
      "  0:01:22.093   249M / 3445M INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:01:22.094   249M / 3445M INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:01:22.094   249M / 3445M INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:01:22.094   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:01:22.094   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:01:22.094   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:01:22.094   249M / 3445M INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:01:22.094   249M / 3445M INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:01:22.095   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:01:22.095   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:01:22.095   249M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:01:22.096   249M / 3445M INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:01:22.096   249M / 3445M INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:01:22.097   250M / 3445M INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:01:22.097   250M / 3445M INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:01:22.100   250M / 3445M INFO    General                 (launcher.cpp              : 312)   Traversed 0 loops\n",
      "  0:01:22.100   250M / 3445M INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:01:22.100   250M / 3445M INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:01:22.100   250M / 3445M INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:01:22.100   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:01:22.100   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:01:22.100   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:01:22.100   250M / 3445M INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:01:22.101   250M / 3445M INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:01:22.101   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:01:22.101   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:01:22.101   250M / 3445M INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:01:22.102   250M / 3445M INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:01:22.102   250M / 3445M INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:01:22.103   248M / 3445M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:22.103   248M / 3445M INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/before_rr.fasta\n",
      "  0:01:22.107   248M / 3445M INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph_with_scaffolds.gfa\n",
      "  0:01:22.112   248M / 3445M INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph.fastg\n",
      "  0:01:22.118   248M / 3445M INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:01:22.120   249M / 3445M INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/final_contigs.fasta\n",
      "  0:01:22.121   249M / 3445M INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/final_contigs.paths\n",
      "  0:01:22.123   249M / 3445M INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/scaffolds.fasta\n",
      "  0:01:22.123   249M / 3445M INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/scaffolds.paths\n",
      "  0:01:22.124   249M / 3445M INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:01:22.125   248M / 3445M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:01:22.363     1M / 3445M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 1 minutes 22 seconds\n",
      "\n",
      "===== K41 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/python /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades/spades_pipeline/scripts/copy_files.py /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/before_rr.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/before_rr.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph_after_simplification.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/assembly_graph_after_simplification.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/final_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/first_pe_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/first_pe_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/strain_graph.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/strain_graph.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/scaffolds.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/scaffolds.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/scaffolds.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/scaffolds.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph_with_scaffolds.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/assembly_graph_with_scaffolds.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/assembly_graph.fastg /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/assembly_graph.fastg /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/K41/final_contigs.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/python /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/scaffolds.fasta --misc_dir /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Assembled contigs are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/contigs.fasta\n",
      " * Assembled scaffolds are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/scaffolds.paths\n",
      " * Assembly graph is in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished WITH WARNINGS!\n",
      "\n",
      "=== Error correction and assembling warnings:\n",
      " * 0:01:16.240   248M / 3445M WARN    General                 (pair_info_count.cpp       : 376)   Estimated mean insert size 184.682 is very small compared to read length 282\n",
      "======= Warnings saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/warnings.log\n",
      "\n",
      "SPAdes log can be found here: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/spades_assembly_kmerXX/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "\n",
      "real\t1m23.078s\n",
      "user\t6m41.991s\n",
      "sys\t0m3.103s\n"
     ]
    }
   ],
   "source": [
    "!time spades.py --isolate -1 SRR18218015_.pair1.truncated.fastq -2 SRR18218015_.pair2.truncated.fastq -o spades_assembly_kmerXX -k 41 -t 6 -m 12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "83c95572-6f1a-4b20-bf99-960105eee081",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Python locale settings can't be changed\n",
      "/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/quast.py -o quast.soap.asmK41 ./spades_assembly_kmerXX/scaffolds.fasta\n",
      "\n",
      "Version: 5.2.0\n",
      "\n",
      "System information:\n",
      "  OS: Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 (linux_64)\n",
      "  Python version: 3.10.8\n",
      "  CPUs number: 8\n",
      "\n",
      "Started: 2024-07-19 16:38:31\n",
      "\n",
      "Logging to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/quast.log\n",
      "NOTICE: Maximum number of threads is set to 2 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files\n",
      "Main parameters: \n",
      "  MODE: default, threads: 2, min contig length: 500, min alignment length: 65, min alignment IDY: 95.0, \\\n",
      "  ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  ./spades_assembly_kmerXX/scaffolds.fasta ==> scaffolds\n",
      "\n",
      "2024-07-19 16:38:32\n",
      "Running Basic statistics processor...\n",
      "  Contig files: \n",
      "    scaffolds\n",
      "  Calculating N50 and L50...\n",
      "    scaffolds, N50 = 7594, L50 = 3, auN = 20432.2, Total length = 156748, GC % = 47.85, # N's per 100 kbp =  0.00\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/Nx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/GC_content_plot.pdf\n",
      "  Drawing scaffolds GC content plot...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/scaffolds_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 816x)...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/coverage_histogram.pdf\n",
      "  Drawing scaffolds coverage histogram (bin size: 816x)...\n",
      "    saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/basic_stats/scaffolds_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2024-07-19 16:38:33\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating PDF with all tables and plots...\n",
      "  2 of 2: Creating Icarus viewers...\n",
      "Done\n",
      "\n",
      "2024-07-19 16:38:34\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/report.html\n",
      "  PDF version (tables and plots) is saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/icarus.html\n",
      "  Log is saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files/quast.soap.asmK41/quast.log\n",
      "\n",
      "Finished: 2024-07-19 16:38:34\n",
      "Elapsed time: 0:00:02.805054\n",
      "NOTICEs: 2; WARNINGs: 1; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "!quast.py -o quast.soap.asmK41 ./spades_assembly_kmerXX/scaffolds.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "50aef505-8443-48ff-ae86-f24aa2ce4c0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
      "\n",
      "Assembly                    scaffolds\n",
      "# contigs (>= 0 bp)         198      \n",
      "# contigs (>= 1000 bp)      25       \n",
      "# contigs (>= 5000 bp)      3        \n",
      "# contigs (>= 10000 bp)     2        \n",
      "# contigs (>= 25000 bp)     2        \n",
      "# contigs (>= 50000 bp)     0        \n",
      "Total length (>= 0 bp)      202455   \n",
      "Total length (>= 1000 bp)   124676   \n",
      "Total length (>= 5000 bp)   85621    \n",
      "Total length (>= 10000 bp)  78027    \n",
      "Total length (>= 25000 bp)  78027    \n",
      "Total length (>= 50000 bp)  0        \n",
      "# contigs                   75       \n",
      "Largest contig              39905    \n",
      "Total length                156748   \n",
      "GC (%)                      47.85    \n",
      "N50                         7594     \n",
      "N90                         640      \n",
      "auN                         20432.2  \n",
      "L50                         3        \n",
      "L90                         47       \n",
      "# N's per 100 kbp           0.00     \n"
     ]
    }
   ],
   "source": [
    "cat ./quast.soap.asmK41/report.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "428c235b-2786-44f6-ae39-a8d7fd2ff6ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "#From the output of the fastg file it was discovered that this was a Salmonella Phage Genome :D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "c976a8c5-c201-498d-8ff6-2908b6a1ab9c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file                                  format  type  num_seqs  sum_len  min_len  avg_len  max_len   Q1     Q2   Q3  sum_gap    N50  N50_num  Q20(%)  Q30(%)  AvgQual  GC(%)\n",
      "spades_assembly_kmerXX/contigs.fasta  FASTA   DNA        198  202,455       42  1,022.5   39,905  376  454.5  594        0  2,293        9       0       0        0  49.19\n"
     ]
    }
   ],
   "source": [
    "!seqkit stats -a spades_assembly_kmerXX/contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "8128f54e-51ab-49df-91e6-bb98062e4d8c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NODE_1_length_39905_cov_4082.125477\n",
      "ACAACCTACAGTCTAACAACCTACAGTCTAACAACCTACAGTCTAACAACCTACAGTCCT\n",
      "ACAACCTACAGTCACAACCATAAGACACTCACAGCCTACAGTCATAGACCTACAGTTAAT\n",
      "AAACACGAAGCTATAGACTTAAGGTCTAAGACATGTTAGTCAAAGACTATATCATCTTAG\n",
      "TCACTATGGGGCCGACGATAGCTTAATCAATATTATTTCAAAATTACCTATTGACTATAG\n",
      "GTCCAGACTATGGCTTAATAGCTTCCGTCAACGCGACACGGCAACAACCGGATAGTGAAG\n",
      "ACGCCGGGTCATCCCGGTTAAGTAGACAGCCTGATAAGTCATACGAACAACAGGCAGATG\n",
      "ACAGAAAGCAGTAGGGTCGCTGGAGGCTCCCACTTCAGCAGGTCGGAAAGGTCCTACAAC\n",
      "CTAACGAGAGCCTGACCAGCTCGCCGAGTCTACCACTGAGGTGGAGAGATAGAGGAGGCA\n",
      "ACTGATCGCCAACTGAGGGGCTACAGCTCACGGTTTAGGTTAACAGGCTAAACAAAACGC\n",
      "TCTTTAACAATCTGGATAAACTCTTAACGTGCGCCGATAGCGACTAACTACAGGGTCTTT\n",
      "GATTCTACATCTGAAGGCCCTGACTGATAGTCACTAACAATTGAGAGGATATAACTATGG\n",
      "AACGTAACGCTAACGCATACTATGACCTTCAGTCTCACGCTTTTGAACTGCTTAAAGAGC\n",
      "GGATTGCTTACGAAGATATTCAAGATGCAAGCGACGCGGGTGATGCAATTCACGAGGTTG\n",
      "CAGGTGATGTAACGCCGCATTACTATCATGAAATCTTTACAGTGATGGCGGCTGACGGCA\n",
      "TCGATATCGATTTCGAGGACAGCGGCCTTATCCCTGAAACGAAGGATGTTAGCCGTATCT\n",
      "GCCAAGCTCGCATTTATGAACAGCTCACCATTGACATGTGGAATAACGTTGACCAAATCG\n",
      "TTGATGAATATTTAGAGACTCTTGAGGACACCAGCGATAGCGTAGTGTTCGACCATGCGG\n",
      "CTGCTGAGGATTGCGAGTGATGGTAACTTATGGTCTCTGTCAACACAACGTCAACAACGC\n",
      "CCGGCTTATGGTCGAGACCGGGCAGTTACACCATGATGCGACTATGCGCCACCTGAAAGC\n",
      "GGTCTATAAGGGGCGCAAGCGCATACACGATAGCCTACACGCTGAGGACAAGCAAAATGA\n",
      "TAACTGAAGGTTCAATCATAGAGTACCGTTGCGAACACGGTTATAACCTGCAAGGTAAGG\n",
      "TGTCTCAGGTCTATACTTCAGTAACAGGTAAGCGCTGCGTGGAGCTGACAAGTGGTGTGC\n",
      "GGATAGGTCTACATGAATGTAGGCTACTGAATAAGCCAAGAACTGAGTGATAGACGAGAG\n",
      "GTCATCTAGCTGATAGGTGGCCTCAAAGATTACCACTTACATTGATAGGAAATAACTATG\n",
      "AATTCGCATGTAATATATCTTAAAAACCAAACCGTAACCTGCTATGAACACACTGATGAA\n",
      "ACTATTGTCGAGCTAGTTAAAAGGTACTACGAGGGTAAGTACGGGTTACACTGCAAAGTG\n",
      "GTAGACGCGAGTGGTCGTATAGTTTATGAGTCTGAAGGTATATAACTATGAGTATCCCAC\n",
      "TATTGATAGTTATAGTCTATGGACTTATAACCTACGTGCTGGTGAGAGACATAAACAAGG\n",
      "CGCGTAAAGTCTATAAATTCAACTATGTACGTCTGGGCCGCTGGACTGTACGTCAACCTA\n",
      "ACGGGCGATTCATGCGCAACCTCGCGAACGTCTGGGACATAGCAACACTTGGGAGCAACT\n",
      "TGTAATGTATAAGATCAGTAAGACATGTAAGGCGCGTCAGCAGCGCGAAAGGAGCAAGGA\n",
      "GTTGGAAGAGCGCTTGTCCAATATCGCTGAGCATAAACCACCTGCTGAAACCAAGCGCAG\n",
      "CATTAAAGATATTCGTTCCGATAAAATCCTGGACACCGCTATTGACGTTAACGGGAAAGT\n",
      "CACTACTCGTCGCAAGCGCATCGACGATAAATTGATGTCGGGTTTCCGTGTGAAAATGGT\n",
      "TGACGGTAAGCGTCGGTTCACTGACGGCAAATCATACTTCGACGAAAGTGACCTCACCAA\n",
      "AACCGAGATGGACTACGCCGAGAAGTGTACCCTCGAAAATAATATCCCTGACGTGGACTC\n",
      "GATGATCGGTAAAAAGTACGAAGCGTTCGGCACAACTTTCACAATCGTCGGTCAAAACAA\n",
      "AAATGGCTATGACGCTAAAGGTGAGGACGGTACCACTGACGAAATCGGTCCGGTTTATGC\n",
      "TGAGAAGCATGTGACCGAGTACGAGAAGAAGTCCAACGGTGCTACCAGCGTTAAGCCTAC\n",
      "AGCTGAGGAATCGTGCTATTAAGGACTCGCCAGAATTGTTAAGAAACGCATTCCACTACT\n",
      "TGACAGGAGCAAACTGTAATGAACAAATCATACGGGATTAATCTGGTACACTCTATGGAC\n",
      "GAAGAAATGAACATGTTAAGCCTACTGGCTACCGGGTATCGTCACGGAACGTCAAGTCAC\n",
      "TCAGAACGTCAACAGCAGGAGCGTGACCGCGTACTTCAGGCACGACTTCGGGCCGAGGGC\n",
      "CATAAGGCTGAGCTGATGTGTCTGCCGTATGGCGTCCGACCAATCACCGATGACGGCAAA\n",
      "CTTCTGGTCTCAGGCTGGAAGGCGATAGACCAGCGGTCAACCATCAAGCACACGACCAAC\n",
      "GGTGACTTCAGTCATCTTCACGCTAACCCACTGATATGCAAGTAATTCTGACTTAACTAT\n",
      "CACTATAGGACTCAAGGTCTAAGACTCCAAGTGAAAGACCAAAGAGACTTTAAGTGAAAG\n",
      "ACTAATAACAAGGACTTTAAGTATGAGCGTCATCTCAATTGACAAGCACGACTTCTCTGA\n",
      "TGTGTCGAATGCAATCGAACCCTTCAACGTACTGGCGGACCACTACGGTCAGGACCTCGC\n",
      "AGTCAAGCAGCTACAACTTGAGCACGAGGCATACACTGAAGGAGAGCGACGTTTCATCAA\n",
      "GAACCTTGAGCGCCAGACTGAGCGCGGGGAACTGGCAGACAATCAGGTCGCCAAGCCTTT\n",
      "GATGCAGACTCTGGTCCCTAAGATTGCGCAAGCCGTCAAGGAGTGGCACGAAGGTCCAGA\n",
      "CGGTAAGCTGTCAACCTCTCGTCCAAGCGTAGCGTTCACCATGTTGAGCACTGAAGAGAA\n",
      "GGCCGTTAAGGACCGCTCTCTGCGCATCTCCTGTGAGTCTGCTGCGGTTATCATACTGAA\n",
      "GGTCATCCTCTCCAAGCTGGTCAAGCCTGAAGGGATACCGATTACACCGATGGCCTCCGC\n",
      "GATAGGTCGCACACTTGAGGATGAAATCCGCTTCGGTCGCATCCGTGACAAGGAGCAGGA\n",
      "GCACTTCAAGAAGGCGATAGCTGAGAACCTGAACAAGCGAGCCGGGGCATCCTACAAGAA\n",
      "AGCCTACATGCAAGCGGTTGAGACCTCCATGCTGGAGCAAGGCCAACTGGAAGATGCGTG\n",
      "GGGAACTTGGAGTCCTACCGAAGCGGTACACGTTGGCATCAAGATGCTTGAGATTGTCAT\n",
      "CCAGTATACACAACTGGTCGAGCTTAAGCGTTATGGTGCTGGCAATGCAGCGGCTGACGT\n",
      "TGAGATGGTCCACCTGTCAGACTTCTGGGTAAAGAAGATGGCGCAACGTGGCTTCAGCCT\n",
      "TGCGGGTATCGCTCCAGTCTATCAACCTTGCGTCGTTCCACCTAAGCCATGGACTGGTGT\n",
      "CGTAGGCGGTGGGTATTGGGCAAAAGGACGCAGACCATTACCACTGATTCGCTTAGGGTC\n",
      "TAAGTCTGCGGTAGCACGCTACGAAGACGTGTACATGCCTGAAGTCTATGACGCTGTGAA\n",
      "CATCATCCAGAATACACCTTGGAAAGTTAACAAGAAGGTGCTGGACGTGGTGAACAGGGT\n",
      "AGAGAAGCTGAACAACACGCCTATTGACGACATCCCTCAGATGGAGCCACTGAGGCCTGA\n",
      "AGACTATGCGGGTGAGACTGAAGAAGAACTCAAGGCATGGAAGAAAGCTGCCGCTGGTAT\n",
      "CTATCGACGCGAGAAGGCCCGACAGTCACGCAGATTGTCACTGAGCTTTATCGTCAACCA\n",
      "AGCGAACAAGTTCTCTCAGTTCAAGGCCATCTGGTTCCCATACAACATGGACTGGCGCGG\n",
      "CCGCGTCTACGCTGTTCCGATGTTCAACCCTCAGGGTAACGACATGCAGAAGGGTCTCCT\n",
      "GACTCTGGCAGTCGGCAAGCCTATTGGTGCTGATGGTTTCAAATGGCTGAAGGTCCACGG\n",
      "TGCAAACTGTGCGGGTATCGATAAAGTCACCTTCGAGGAGCGCATCAAGTGGGTGGAAGA\n",
      "CAACCACGACAACATCATGGCTGCTGCTAAGGCACCGATGGATAGCATTGAGTGGTGGGG\n",
      "AAAGTTAGACTCTCCGTTCTGCTTCCTTGCGTTCTGCTTCGAGTATGCTGGCGTAATGCG\n",
      "CCACGGCCTGTCTTACTCTTGCTCGCTACCGATAGCGTTCGATGGGTCATGCTCTGGGAT\n",
      "TCAGCACTTCAGCGCGATGCTTCGTGACCACATCGGTGGACATGCAGTAAACCTGACTCC\n",
      "ATCCGGTAAAGTCCAAGACATCTACCGCATCGTGTCTGACCGCATTGAGGAGGAACTCAA\n",
      "AGTCCTGCTGGTTAACGGTACAGACAACGAGATGGTAACTCACGAGGACAAGAAAACTGG\n",
      "TGAGATTACCGAGCGTCTCAAGCTGGGGACACGAGAGCTGGCCCGTCAGTGGCTGACCTA\n",
      "TGGTATGTCACGAAAGGTCACTAAGCGCTCAGTCATGACTTTGGCCTACGGGTCGAAAGA\n",
      "GTACGGCTTCGCAGACCAAGTGTACGAGGACATCGTGATGCCAGCGATTGACTCAGGGTC\n",
      "TGGCGCTATGTTCACTGAGCCAAGCCAAGCGTCTCGCTTCATGGCTAAGATGATTTGGGA\n",
      "AGCTGTGAGTGTGACCGTAGTTGCTGCGGTTGACGCGATGAAGTGGCTTCAAGGTGCCGC\n",
      "CAAGCTGCTGGCTGCTGAAGTGAAGGACAAGAAGACTGGCGAAATCCTGAAGCCTTGCCT\n",
      "TCCGGTACACTGGGTCACACCTGATGGGTTCCCGGTCTGGCAGGAATACCGCAAGAAGGA\n",
      "TACCACTCGTCTGAACCTGATGTTCCTAGGGTCATTCAACCTTCAGCCTACCGTCAACAA\n",
      "AGGAACGAAGAAAGAGCTGGACAAGCACAAGCAGGAGTCAGGCATTAGCCCGAACTTCGT\n",
      "CCACTCACAAGATGGTAGCCACCTACGCAAGACTGTAGTCCACACGCACCGTAAGTATGG\n",
      "CGTGATGTCCTTCGCAGTGATTCACGATAGCTTCGGGACCATCCCGGCTGATGCTGAGTA\n",
      "TCTGTTCCGTGGCGTCCGTGAGACGATGGTCGAGACCTACCGCGACAACGATGTGCTGCA\n",
      "AGACTTCTACGAGCAGTTCGAAGACCAGCTTCATGAGAGCCAGCGCGACAAGTTGCCTGA\n",
      "GCTGCCGAAGCGAGGTAAACTTAACATCGAAGACATTCTGTTATCTGAATTTGCATTCGC\n",
      "CTAAGGAGGCACCTACATGAAATTCGCACACAAGCAAACTGGCGTTAAGGGTGGCACTCA\n",
      "AATCGTGACCGTGACTGAACACAATGGCAAGGGTTTAGTGAAGACCACGGTCATTCCGAC\n",
      "CGAGATGTCAAAGCAGCTTAACGTCCCATTCAAATGGCTGGTGCAGCAGGTCGAGAAGTC\n",
      "ACACGAACAGGCATTGACCGAGGCGGCAATGAAATGACAGACCTACAGTTCCTGGCCTTG\n",
      "TGGCTGGGAGCACTAGCAGTATTCACTTTAATCCAACGCAGAAGAGGTTAAACTAAACTA\n",
      "TCACTATAGGATTAGACTCAAGGTCATGACTCAAAGTCGTGGCCTTCATGATTAACCCTA\n",
      "TAACCTAAACAACTGGAGCTACAACATGTATCAGAACACTATCAACTTCGAGCGCATCCG\n",
      "TGAACGTCAGAAGACTGAAGGTTACATTCCGAAGGGCCGCAAGCTGAACAAGACCAAGCG\n",
      "TGGCGGCGGTGTCAAGGGTGCTTTCCGTAACGCTGAGAGTAAAGACTCTCTGGTCAACCA\n",
      "AGAGAAATACTTCGTAGGAGCTTAAGGCATGAGCAGACGTTGGTTATTCGATGCGTCCAC\n",
      "GAGCCAGTGGTCATCGTTGGGCAGGACACACCGCGAAATGAGTGAGGCAGGACTTGATGT\n",
      "CAAAATGGATGACCTGCCGTTGTCAAGGACAACTCTGGTCCGAATCATAGAAGCAGACGG\n",
      "GACGCCTCTAGTCTCCCGCGAGTTTACCGAACACGAAATCTTTAGCTGTGAGCAATGGTG\n",
      "CTTGCAGCAATTGAATGATTACAGGAGCTGGAAATAATGACTACGACTATTAAAACTAAC\n",
      "CCGCACCGCGCTGTAGATTACTCTGAGTCCGGCGTTAAGAAAGCACTGGAAGCCGCTGGG\n",
      "TCGCTGGAAGCTGAAGTGAAGTATGACGGCGTCCGGTTGAATCTTCCGGTCTTCCCTGAT\n",
      "GGGCGTACTGAGTGGCTAAGCCGAGAGTCCAAACCACTGCCAGCGCTCGACCATTTGTCA\n",
      "ACAGTTGGTCAGGACATCGGAGCGTGCCGTGCTTCCGACTGGCGATGGTTCCTGAAGCAA\n",
      "GCAGGACATGAAGTCTCTGGCTTGATGATTGACGGCGAGGTGATGGTCAAGGGTGTAGAC\n",
      "TTCAACACATCCTCAGGTCTTATCCGCACGAAGTATGTCAAGTCAAAGAACTACGACTTC\n",
      "CATGATGGTGATTGGGACCATAAGTCAAAGACTACGCAATTCAGGTTAGACCCTGAGCAC\n",
      "ATTAAGGTTATCGTTTACGGTATCATCGACCTTAACTTCATCCTCGACCCGAAAGCTGAA\n",
      "GGTCCTATCCATAGCGTCACTCGCCTGAAGGCCGAAGCTATCGTCCCTCTACTCCAGAAA\n",
      "TACTTCCCGGAAATCGACTGGGTTCTGTCTGAGTCACACACGGTCTATGACCTTGAGTCG\n",
      "CTCAACTCCCTGTATGAAGAGAAGCGTCTGGAAGGACACGAGGGTCTGGTAGTCAAGGAC\n",
      "CCACTTGGTAAGTATAAGCGTGGCAAGAAGTCAGGCATGTGGAAGATGAAGCCTGAGGAG\n",
      "ACCATCGACGGGACCGTGTGTGGCCTCGTGTGGGGGACTCCTGGCAAGGCTAACGAAGGT\n",
      "AGGGTCGTCGGCTTCGAGGTTCTACTTGAGGATGGCATGGTGGTGAATGCCTGTGGTCTG\n",
      "ACTGAAGAACAGAAGGACGAGTTCACAGCTAAGGTCTTTGAAAACTGCAAGGACCAGATG\n",
      "CTCGACATCGATAGCTCGGAATCATACATAGATAGTACAGTTCAACAGCATGTAGTAAGT\n",
      "CCGTATGAAGGGTGGCAATGTGAAGTTCTCTTCATGGAGCGGTTCCCTGATGGCTCCCTT\n",
      "CGTCACCCTAACTTCAAGTGCTGGCGTGGAACCGAAGACAACCCAACCGTTAAGATTTAA\n",
      "GGTTGATATATATATATATAGATATTTAATGTAAACGAGAGAGGTCAATAAATGGTTATT\n",
      "TGGTCAATGTTTGACGGCTCTGGGATTGCAGTAAAAGATTGGGCTGAAGCTGGTCACGAG\n",
      "TGTTATTGTTTTAACTTTGATGATGCTGACCATGGTGATTATAAAAATCTCGGCGCCAGG\n",
      "ATTGAACATGAGAAAATCAAGTACGTTAACGCATGGATTGATGATGAATTTGACTTTGAA\n",
      "GTTAAGCCTGATTTCATAATGGCATTTCCTCCATGCACTGATTTGGCGGTTAGTGGGGCA\n",
      "GCGCACTTTGCAGCAAAGAGAGCTAAGAATGAAAATTTTCAGATTGATGCCGCCAACACC\n",
      "TGTAAGGTAGCTTTTAATCTCGCTGAGAAGTATGGGATTCCATATATGATAGAGAACCCG\n",
      "GTTAGCGTTCTTTCGTCATTGTGGAGGAAGCCAGATTACGCGTTCCACCCTTACGAGTAT\n",
      "GGCGGCTATCTTCCTGATGGGGATGCTCACCCAATTTTTCCTGAGTTCATAAAAGCAAAT\n",
      "GATGCGTACCCAAAAAAGACATGCTTATGGACTGGTAATGGCTTCATCATGCCAGAGAAG\n",
      "AAGATTGTTGAGTGTAATAATGGATATTCCAACCAGCACAAGAAGCTAGGAGGTAAGTCA\n",
      "ATTAAAACAAAGACAATCCGCAGCTTAACTCCTCGCGGTTTTGCAAAGGCAGTATTCATT\n",
      "GCAAATAGCAATCGTGATAGTAGGCTGCTAGCGGAGAACGTGCGGTCCGTTACGTCAAGA\n",
      "CCATAGCCTAGGCAGTAGACTTAATAGATTCATTACAATGTCAACCCAGTGGTCTTCGGA\n",
      "CTGCTGGGTTTTTTGCTTGTACTAGCAAGCTATCGTCTGGGCCACCAGCATGACCCTAAG\n",
      "CTATCACTATAGGACAACCATAACTTCGACCTAAGGAGGCCGACCATGTTTAAGAACTTA\n",
      "ATGTTCAACCGTTATTCCAGTACCTTCCACCTGTCCCACAACCCGTTCGCTTGCATTAAG\n",
      "CGCAACGAGAAGCTGGGCTACTTCGGGAAGTCCGTTAAACTGTCACCTACAGTCTTCGCT\n",
      "CTCATTACTCCGGGCAAAGCTGAAGAGGCTCGTCAGAAGCGTGAGACCAACGTACCTGTG\n",
      "GTCTACACCAAGTGGCCTCGTGTCCGCCTGTTCATTGAGTTCGTGAAGGAGGTGGTCAAT\n",
      "GACTAAGCACAAGGAGGGGTGCCGCTCGGTATTGAGCTTAGGGTCTTGTGACTGTGGTGC\n",
      "TGGGGACATCGGCAGGGTCAGCGAAGATGATGGTGTGCGCAAGCCGTCCCACTATCAGGT\n",
      "CTTTGACGGTGTGGAGTCCATTGAGATTATCGCCCGGTCCATGACCGTGAGTGAGTTCCG\n",
      "TGGGTTCTGCCTCGGTAACGTCCTGAAGTATCGACTCCGAGCTGGTAAGAAGTCAGATCT\n",
      "GGCGACCATGGAGAAGGACCTTAAGAAAGCGGCCTTCTATCAGGAGCTGTTCGACCTGCA\n",
      "TAAAGGCAAGTGTCATGCTTAATGATATTCTGTACTTCGGATTAATCTGTGCAGGAGTAG\n",
      "GTGCATCATTGCGTTTAATGTATTGGGCGATAGAGGTGGAATATGGTAAAACCAAGTGAG\n",
      "TGGTGCCGGGAGATGTATGAGAAGACTCTCAACCCGGACTACATCACGTTATACAACCAA\n",
      "TGGAAGGAGCGCGGTTTATGACTGAGGTTGAGAAGAAGTACATCGTGGAGCTTGAGGGTC\n",
      "GCGTTCAGTCCTTCGAGGTTCCGGTGTACGCAAAGTCTCTTGAAGAGGCCACCCTGAAGT\n",
      "CCCAAGAGTATGAGGACGCTGGGTTTGTGGTCGGACGAATCCGTCCTGAGACCTAAACTA\n",
      "TCACTGTAGGACAGACGTCCAGTTCGTAACTTTAAATTAGGAGATTTACACCGATGGCTA\n",
      "AAGAGCAACTGAAGACTTTCACCACTCCTGTAGCTGGTATCGTTGAGCCTTACGCATGGC\n",
      "TGAACAAAGCAGACACCAAGTTCAATGAGCGTGGTGAACACAAAGCTAACCTGACATTCG\n",
      "ACCTGAGCAACCCTAAGGTCCGTAAGATGATTGACGTCTTACAGAAGATTCACGACGATG\n",
      "CGTATGCGAAAGCACTCGCAGACCACGAGAAGAACCCACCTCAGGTTCAGCGTGGCAAGA\n",
      "AGCCTATTGAACCACGAGAAGGCGACATGCCGTGGATTGAGAATGGTGACGGTACTGTTA\n",
      "CATTGAAGTTTAAATGCTTCGCGTCTTACCTGAAGGATGGTAAGTCAGAGCCTATCGTAT\n",
      "TACGGTTCTACGACACCGATGCTAAGTTAATCCGTGACGTCCCGAATATTGGCGCTGGCT\n",
      "CAAAGCTGAAGGTCAAGTTTAAAGTCCTGCCGTTCAAGTGGAACGCTGCGACTGGTGCAA\n",
      "GCGTTAAGCTCCAGCTTGAGTCCTGCCTTCTGGTAGAACTGAAGGAGTGGAAAGGTGATG\n",
      "GTGCTGGTGGCGATGGTGGCTGGGGTGATGATGAAGACCTCGGTACTGGCTACAAAGCGT\n",
      "CAACCGATGGTGACTTCGGGTCTGATGACTTCGGTGACGATGGTTCCGAAGGTGGCGACG\n",
      "ACTCGAACTCTGGTGGCGATTACGACTTCTAATGGCTCAATGGTCTGCAAAACGGGGGCA\n",
      "CTCTGTGGGTGCCTACCGCTCTGGACTTGAAGCCAAAAACCAGCAGTGGCTGGAACAGAA\n",
      "CGGCGTCAAAGCGGAGTACGAAAGCCATTATATCAACTATGTGATTCCGGCTTCCGACCA\n",
      "CAAGTACACACCAGACTTTATCCTTCCTAACGGTATCATCGTGGAGACCAAGGGTATCTT\n",
      "CGACAGTGATGACCGCAAGAAGCACCTTCTGGTACGAGAACAGCACCCTGAGTTAGACAT\n",
      "TCGGTTCGTGTTCTCAAGTTCCCGCTCCAAGTTATACAAAGGGTCTCCGACCACGTATGG\n",
      "CGCATGGTGCGAAAAGAACGGCTATAAGTATGCCGATAAGTTCATCCCGGTTGAGTGGCT\n",
      "TAGAGAGGCGACTGTACGTCTGCCTTCAGGCATCCTCATCCCAAAGAAAGGAGTTAAGAA\n",
      "ATGATTCGTAAAGCTGTTATTATCAGCGAGGGTGGCAGCCACCATTGCCCGCCTAACGGG\n",
      "ACAGAGGTATTCGTTAAACCTTCTAGAATTGACAATAATTTCCTGCGGTTCTCACTGGAT\n",
      "GGAAAGAAATGGCTGGATGATATTCGTGCCTACCGCTTCAAGATGATTGAAGAAGAGAAA\n",
      "CGTGAAGTCCTTCCAGCATCAAGCGTCCCTCCTGTAACCGTTAAACGTGAAGTCCTGACC\n",
      "ATCAACAAGATTGGTGTAGGTCAGACGTTCATCGTCAACGGTAAGCCTGACGAAGTGTAC\n",
      "GTGAAGATTAGCAACTCTCACGTCTTCAACCATAAGTTGCTTCAGATGCACACCACGGAT\n",
      "GCTAACCGTTTCCAGCGTCATCTGAACCTAGTAGCTGTAGAGCTGGTGGTGTACGATGGT\n",
      "GAGTGAGTCCATTCGGTCAAGACTTCTGGCTGGCTACGAGGTGGACTACAATGGGTGCTG\n",
      "GAACTGGAAAGGTGCCACGGCATCCGGCTATGGAGTGATTGTCCTTAACCGGAAGGTCGC\n",
      "TGCTGAGCATGGAGTTGGTAGAAACTACCAGACCCACAGGGCATCCTATGAAATCCACCA\n",
      "CGGGCCAATACCTGAGGGGAATGTGGTCAGGCATAAGTGTGACAATAAGTTATGTTTGAA\n",
      "CCCTGACCACCTTGAAGTCGGGACCCAGCGTGAGAACATTCAGGACGCTATAGACCGAGG\n",
      "TCGTATGGCGAGCAAACTGACTGAGAAGGACATCGAGTTCATCCGGTCATCAACGTTAAG\n",
      "TCAACGGAGGTTAGGTAAGCTGCTGGGCGTGAGTCCCACTGTTATCTGGCATGTTCGTAA\n",
      "TGGCAGCAAATGGAGACATGTCTAATGGTCACTAAGGTACAATTCAAACCGCGTCAAGCT\n",
      "ACAGACGCAATCTTCGTCCACTGTTCAGCGACCCCGCCGACTATGGACATTGGTCGTGAG\n",
      "ACCATTGAGATGTGGCATAAGCAGCAGGGCTGGCTGGCAATTGGCTACCACTTTATCATC\n",
      "AAGCGTGATGGCACTGTGGAAGAGGGCCGCCCGGTCGATGTAGTAGGGTCACACGTTAAG\n",
      "TACTGGAACTCCCGGTCAGTAGGCGTCTGCCTTGTAGGTGGAGTCAATGATAAGGGTCAG\n",
      "TTTGAAGCTAACTTCACTCCAGCCCAGATGAACTCCCTGCGCAACAAGCTGGCTGACCTG\n",
      "AAGTCCTTGTATCCTCAGGCAGAAATCAAAGCACACCATGACATAGCACCTAAGGCGTGT\n",
      "CCAAGCTTCGACTTGCAACGCTGGTTGACTACCAACGAACTGGTCACTTCCGACCGCGGT\n",
      "TAATAACAATGCTTAATAACGATTGGGGGACCTTAGGGTCCCTTTGTTCGCATTTCCGAT\n",
      "TAAGGAGTGACCAATGTCATACGATGACCAAGACGACGAGAGTGTCTTTCTGTATCACAC\n",
      "CCAGTGTCCAGACTGTGGGTCCTCGGATGCCAATGGTGTTTACTCAGACGGCCATCAATT\n",
      "TTGTTTTGCATGTGACCCTTCAGTCGCATGGAAGAAAGGGGACATGGAGTTGACCGAGGG\n",
      "ATACACACCCTCAGGAGGTAAGAAGCAAGTGAGCAACCTGTTAACGTGGGGTGAGAACTC\n",
      "AGGACGTTATGTCCCACTCCCAGCTCGTGGACTATCGGCTGACATCTGCAAGAAGTACAG\n",
      "CTACTGGGTAGGCATGATGCAGGGCAAGATGGTCCAGATTGCGGACTACTACGACAGGTC\n",
      "CGGGACCAAGGTAGGACAGAAGGTCCGAGACGCTGACAAGAACTTCACGGCTATCGGTAG\n",
      "CGTCAAGAATGACCTACTGTTCGGCTCACAGCTCTGGTCTGGTGGCAAGAAGATTGTCAT\n",
      "CACAGAAGGGGAGATTGACTGCCTGTCAGTCGCTCAGGTTCAGGACGGGAAGTATCCTGT\n",
      "GGTCTCTCTACCGCTGGGCGCTAAGTCCGCGAAGAAGACACTGGCAGCGAACCTTGAGTA\n",
      "CCTCGACCAGTTCGAAGAGATTATCCTTATGTTCGACATGGACGAGCCGGGACGCGAGGC\n",
      "TGTAGAGCAGGCTGCTCCGGTCCTCCCTGCTGGTAAGGTAAAAGTCGCGTTCATCAATGG\n",
      "GTACAAAGACGCCAACGCTGCACTTCAGGCCAAGGACTTCAGGGCGATTCAGGACGCAAT\n",
      "CTGGAATGCTAAACCTTTCGTCCCGGCTGGGGTGGTGAGCGCCAAGTCTCTGAAGGACAG\n",
      "GACCCGCGAGGCAATGCTGAAGGCTGAGACTGAAGGTCTCCTCTTCTCGTCCTGCACCAC\n",
      "ACTCAACGCGATGACCCTAGGTGCGCGGTCTGGTGAGCTTATCATGGTGACTTCAGGGTC\n",
      "AGGCATGGGTAAGTCCACCTTCGTGCGTCAGCTCCTCCTAGAGTGGGGCAGACAGGGCAA\n",
      "GCGTGTAGGCATGGCGATGCTGGAGGAAGCTGTTGAAGAGACAGTGCAGGACCTCATAGG\n",
      "TCTTGACAATAACGTCCGTCTCCGTCAGTCCAAAGAGCTGAAAGAGGCAATACTTCAGGA\n",
      "CGGTCGATTCGATGAATGGTATGACAAACTGTTCAACGACGATATGTTCCACCTTTACGA\n",
      "CTCATTCGCTGAGTCAGAAGAGGACACATTGTTCGCCAAGTTAGCCTACATGGTTGACGG\n",
      "GCTGGACTGTGATGTCATACTGCTGGACCACATCTCAATCGTTGTGTCTGGCATGGACGA\n",
      "TAACTCGGATGAACGTAAGACCATCGACCGAATCATGACCCGTCTCAAGAAGTTTGCGAA\n",
      "GACGAAGGGCGTTGTGGTCGTGGTAATATGTCACCTGAAGAACCCGGAGAAAGGTAAATC\n",
      "GCATGAAGAAGGACGACCAGTTAGCATTACTGACTTGCGCGGTAGTGGTGCTTTACGTCA\n",
      "GCTATCCGACACTATCATCGCACTTGAGAGAAATCAGCAAGGTGATACTCCTAACGTTGT\n",
      "GCAGCTTCGTCTGCTTAAGTGTCGCTTTACTGGTGACACGGGCTTGGCTGGAAACCTTGA\n",
      "GTACAACAAGACGACCGGGTGGCTCGAACCGCTTGCGTCGTTTAGCGGAGGCGGAGAAGA\n",
      "GGATAGCGGCTCGTGGGAAAACAACGACTTCTAGCCTTGAGTCGAGACTTGAAGAATCAC\n",
      "TTAAACGGAGAGTAACTAATGCTACGAAAACTTAAGGCCCGATACCACAGGTTCATGTTC\n",
      "AAGTGGTGGAGCGAGGAGGCAGACTGCCTGTCCCGAATCCTCGGAGAGCAGCGCTTTGAG\n",
      "TCTGCTGCGTGGAAGCGAGCGAACCGAAAGTTCATGTATCACTTCCTGCGTACCGACTAC\n",
      "TAAGCTATCACTGTAGGATAGACTCAAGGTCATTCACTTCGAGTGGCCTTTATGATTAGA\n",
      "CTATAGGAGACATTATGATTGACTTACAGAACATCTGGGGTTCGGACATCGAGACCAACG\n",
      "GTCTTCTTGATACCGTCTCTCAGTTCCACTGTGGGGTCCTGATTAACGCCGAGTCGAATG\n",
      "AGACCCTGAAGTACGGGGTAGCTCCGATGGTAGGTATCGTCGGTGGCTTCAAAGAGTATG\n",
      "TGCAGAAGGTCGAAGAGATTGCAGCCACACCGCATGGACTTCTGGTCTTCCACAACGGGA\n",
      "TTAACTATGACGTCCCGGCTATTGACATCCTGAAGCGTAAGTATTTTGGTAAGCGTTTCA\n",
      "ACTTCCCGAAACACAAGATGATTGACACTCTGGTCATGGGCCGCCTGATGTACCCGAACA\n",
      "TTAAGTTCTCGGACATGGGTGCTGTGAAAGCTGGTCGTCTTCCGCCTAAGATGATGGGAC\n",
      "GACAGTCTCTTGAAGCATGGGGCTACCGTCTCGGTGAGATGAAGGGTGAGTACAAGACCG\n",
      "ATTACATTGCCAAGTGCAAGGCAGAAGGTATCGAATATAAGGCTGGTGACGAGTGGTTAT\n",
      "TCCCAAGCCAAGAGATGCTGGACTATAACGTCCAAGACGTTGTGGTCACTCTGGCCCTGT\n",
      "TCAAGAAGTTCCTTACCGACAAGTTCTACTTTGGGACACCTGAGGCCGGACTTGAGGCTG\n",
      "TGTATGCACTGCGTCTCGAACACGATGCTGCGTGGACATGCGCGAAGATGGAGCGTAACG\n",
      "GCTACCCGATGAACTCGGAGATAGTCGAATGCTTGTATCGTGAGCTATCAGTCAAACGTG\n",
      "CAGAGCTTTTGGATAAGCTGCGGTCGACGTTCGGTAGTTGGTATTCACCTAAGGGTGGCA\n",
      "AAGAGTTCTTCAAGCACCCACGAACCGGGGTAGACCTGCCGAAATACCCGCGAGTCATCT\n",
      "ACCCGAAGGTAGGGGCAATCTTCAAGAAGCCGAAGAACAAAGCTCAGCGCTTAGGTCTGG\n",
      "AGCCTTGCGAACGAGACTCCCGCGATACGATGGAAGGCGCACCGTTCACACCGATTACCT\n",
      "TCGTTGAGTTCAATCCGGGGAGTGGAGACCACTTAGCTAAGGTTCTGATGGACCGGGGCT\n",
      "GGGAGCCTACTGAGTTCACGGATACAGGGAAGCCTAAGTGTGATGACGAAACTCTGGAGC\n",
      "ACGTTAAGCTGGCAGACCCAGAAGCTCAGGCTTGCGTAGAGCTGGTCCGTGAGTATCTGG\n",
      "TGGTCCAGAAGCGCATTGGGCAGGCGGCTGAAGGTAAGAACGCATGGCTTCGACTTGTGG\n",
      "GTCCTGATGGGAGGATGCACGGGTCGATTAACCCCTGTGGTGCTGTAACCGGACGTGCGA\n",
      "CTCACAGTTCACCGAACATGGCTCAGGTCCCGGCTAACGGTGCTCCGTGGGGTGAAATCT\n",
      "GCCGAAGTGCTTTCGGTGCTCAGTGGAACAAGAACAATGGCAAGCCAGACCCGTGGATTC\n",
      "AAGTTGGGGTCGATGCCTCAGGCCTTGAACTCCGTTGTCTGGGGAACCGAGCGTCTCCAT\n",
      "TTGATGGAGGAGAGTATGCGAAGACTGTAGTCGACGGTGACATCCACTGGGCCAATGCGG\n",
      "TCAACGCCGGGTTAGCGCCTAACGTCCCACGCGATAAGTCCAGCCACGACCACGATGCTT\n",
      "TCCGTAACAACGCCAAGACGTTCATCTATGCGTTCCTGTATGGTGCAGGGGCCGCGAAGA\n",
      "TTGGACTGATAGTTGGTGGCGGTAAGAAGGAAGGTTCGGCCCTGATGAAGAAATTCATTG\n",
      "AGGGTACACCAGCTATCAAGGACCTGCGAGAGGCTGTGCAGAACACGCTCATCTCAGAGT\n",
      "CCAAGTGGGTAGATGGTGAGAACATCGTGAAGTGGAAACGTCGCTGGCTGAAAGGTCTGG\n",
      "ACGGTCGCCGCATTCACATCCGGTCGCCACACTCAGCACTTAACGCCCTACTTCAGGGTG\n",
      "ACGGTGCGGTGGTCTGTAAGCACTGGATTGTCGAGACCGAGCGTATGCTCGAAGAGGCCG\n",
      "GGTATGTCCACGGCTGGGAAGGTGACTTCGCTTACATGGCTTGGGTTCACGACGAATTAC\n",
      "AGATAGCGTGTCGTACTCAGGAGATTGCCGAAGAAGTCGTCAAGATTGCTCAGCTTGCTA\n",
      "TGCGTAAGGTGGGTGAGTTCTATAACTTTAAATGCGTCCTTGATACCGAAGGGAAGATTG\n",
      "GACCAACGTGGAAGGAGTGTCACTAATGGCTATTACTAAACGTCTGACCTGTAAGTTTGA\n",
      "CATGAAGATTGTCTTTACGTCTGAGGACGTAGAGTCCTACTCTCGCGAGATGGTAGAGAT\n",
      "TACCAAGGGTTACGTCTCAGGTGAGAAGCAGGACGGCCTGACCCGAAAGATGGTGGAGAC\n",
      "TGCTGTAGAGCATGGGGTCGAAGCTGCTATTGAGCTACACCTGAAGTCTGTGCTGGCGTC\n",
      "CAGGCTGAAAGATGAGCTGCAGGAAGACGGTGCTGCCCTCTCTAATATCAGCGTGGGGTT\n",
      "CAAGCAATGAGCGAATACTTACGGGTCCTTGCGGCCCTTAAGTCCTGCCCGAAGACATTT\n",
      "CAGTCCAACTACGTGCGCAACAACGCTGCACTGGTGGCTGAGGCCGCGAGTCGTGGACAT\n",
      "CTGAGCTGCCTGTCTATGGACGGGCGTAACAACGGCGCGTGGGAGATTACCGCTGCTGGC\n",
      "ACCAAGTTCCTTAACCAACACGGAGGCTGTCTATGAAGATACCTGAAAGTCTGAAAGAGC\n",
      "TGGTCCTTGAAATGGGTGAGTGCGAGAAGGAGCACCGCAAGTACGCGGAGTTCTACGCAT\n",
      "CTTGGTCGTACACCCTGTCAGAGAAGGACGTTGAGAACTACGCCGAGTCTCCTTCTGAGG\n",
      "CTTATGAATGGTGTCCGTTCATCGGCCTGACCATCGTAGCCAGCGGTATGTGGAATGACA\n",
      "GCTATGGCTCAGAGGTAAATGATTGGGGAGTGTATCGCACGGTTGAGAAGGAGACCGAAA\n",
      "TGGCTTCAAACTTCTGGAGCCTAATGTCCCACCTCAACTCAGAGGACAACGACATGGCCC\n",
      "ATGAGTTCGCCCACAAATGGCTGAAGCCTGAGTTGATTCTAGAGCGTATTGAGGAGCCGA\n",
      "AAGTTTCCTTGATGCCAGCTACAGTTGCTATGGCCACAGGAGGTCTGGACGATGGGTTCT\n",
      "GAGAAGAAGATTGCTCTGGTGCTTGATGGTGACTATCTGGTATTCTCCTCTATGGCTGCG\n",
      "GCTGAAGATGAGACTGACTGGGGAGACGACATCTGGACCCTTATCTGCGACCACGAGAAG\n",
      "GCTCGTCGTATCCTTGAGAACACCATCGCTGAAATCGTCAAGAAGCGCAAGGAGTGGAAA\n",
      "GACGCTAAGATTGTGATGTGCTTCACCGATGACGCCAACTGGCGCAAGGACGTCCTGCCC\n",
      "ACCTATAAGGCCAACCGTAAAGGTTCTCGCAAGCCTGTAGGTTACAAGAAGTTCGTAGCC\n",
      "GAAGTGATGGCTGACCCACGGTTCAACAGCTTCCTGCGTCCTACGCTTGAGGGTGATGAC\n",
      "TGTATGGGTATCATCGGGACCAGACCTCAGATTGTAGGCTGTGACCATGCGGTGCTGGTG\n",
      "TCCTGTGATAAGGACTTCAAGACCATCCCGAACTGTGAGTTCTTCTGGCTAACCACTGGT\n",
      "GAAATCCTGAGTCATACGACTGCCGAGGCAGACTACTGGCACATGGAGCAGACCATCAAG\n",
      "GGTGACACTACAGATGGCTACGGCGGTATTCCTGGGATGGGTGAGGACACCACTCGTGCA\n",
      "TTCCTTGACGAGCCGTACTACTTCGTGCAGGAGAGCCGTGAGCTTAAGACTGGTAAGAAC\n",
      "AAAGGCCAGATTAAGACCGAGTGGAAGAAGTACCCTAAGCGTGAAGACATGACGTTGTGG\n",
      "GACTGCATGGTGACTCTGGCTGCTAAAGCTGGGATGACCGAGGAGGAACTTCTGGTCCAA\n",
      "GCTCAGGTAGCTCGTATTTGTCGAGCCTCCGACTACGACCCTAAGTCAAAGGAGGTCATC\n",
      "CTGTGGACACCATCCATGTAATCTACTGGGCCGGACTTCTGGCCCTTTACTGCATGTACA\n",
      "AGCGGTTCGGGTCGAACAACCGTCCTAAACACTGAGTCTAGCCGATAGTCATATCCTATC\n",
      "AATCCAACAGTCATCCATAGGTGAAACACTAAACTATCACTATAGGGACTTTAGGACCTA\n",
      "AGATATGACTATAAGATAGACTTTAGTCTTAACTTAAAGAGGAGATTCAAGATGGCAATT\n",
      "AATGCTATTGAAAACGTTGTTAAGCAGTTACAGGAAGAAAGACTTGATGTCCCGAACATC\n",
      "TCCCAGTCTGCCATCCAGTTTATGCACGTACTGTTCAACGCAAGCTACGCTGAGAAGACT\n",
      "GGAGCTATTAGTCTCCTCAAGCAGCAGGGCTACAGTGATGCGTTCATTGCCGGGTTTATC\n",
      "AAGGGTCTCCAGTATTGCTCTGACACTCTCGACTCTGCGATTGCTATGCGTCGTGAGCTG\n",
      "AAAGATACCGTTCAGTTCGATTAACTGTAGGAGGGACTATGTGTTTCAGTCCAAAGATTA\n",
      "GCACTCCGAAGCCTTCGGTCCAAGCACCTGAGCCAGCACCTCTGAGTGAGGAAGTTGCGT\n",
      "CAGTTGACATCGGGGCCGAATCGGATGTGAACACCAACGAGACCAAAGGTATCAAAGATC\n",
      "TGAAGGTCAAGAAGGAGTCTACATCTAAAGATAAATCGTCAGTTAGCCGCGCTATGCGAG\n",
      "CCTCTGGCGTCAACATGGGGTAAGACAATGCTACCATATCTCAACTCACGCGAAGGCCGC\n",
      "CACATGTGCGCTTGTCGCCTCTGGGAAGACGGGCAGTCTAACTTCAAGTCATTCGAGGAC\n",
      "TTCAAGGCTCATACTTACCGTATGGCTGACGAGTTCGACGGTGAAGAGTACACAATCTAC\n",
      "GATGTCTCAGGTCAACCAGTAGCGTATCTCTACATGCTGGCTACCGCATCTTGGCACCGA\n",
      "CCGACGCCCGGTCTGGACCTTTCAATCGTCGCTATTCGTCGTGACTCGCAGTCCTCCCGC\n",
      "AAGGTTCTTGAGACTGTCAGACACATCATAGACGAAGAGTGCAAGCGTTGGGGTCTTGGC\n",
      "TGGTATTCTCGTGTCAAGCATGTTTCTGGGTCGGTAGACATCGTAACAACCAAGTGCGTT\n",
      "AGCACGAACAAGGAGATTAATCGTGGGTAAATCAATCAGTAAGGCCTTCAAGAAAGTAGT\n",
      "AAAAGGTGCGTTAGGGACAGTTGGTCTTGGAGCTAGTGATGCACCTAAGGTTGTTGAGGC\n",
      "TCCGACCCCAGCAGCAGCAGCAGTGGAAGTACCGAATGACAAAGTGGAAGATGTGGATAC\n",
      "CGAAGTTACTGCATCTGACGAGAAGAAAGTGAAGCGTTCCGGTAAGCGTAGCCTTCAGGT\n",
      "CTCTCGTACCTCTGGTGGTGGTATTTCTTTATAAGGAGGCACGTAGATGGCTGAACGTGA\n",
      "AGGATTTGCTGCTGAAGGCGCGAAGTCAGTTTATGATAGATTGAAGAACGGACGACAGCC\n",
      "TTACGAGACACGCGCTCAGAACTGTGCTGCTGTCACTATCCCGTCACTGTTTCCAAAGGA\n",
      "GTCCGACAACTCGTCTACGGAATACGTGACTCCGTGGCAAGCTGTAGGTGCTCGCTGTTT\n",
      "GAACAACTTGGCTGCAAAGCTGATGTTGGCATTATTTCCTCAGTCACCGTGGATGCGACT\n",
      "GACAGTCTCCGAATATGAGGCCAAGACCTTGAGTCAGGACTCAGAGGCTGCTGCTCGTGT\n",
      "TGACGAAGGGCTTGCTATGGTCGAGCGTGTGTTGATGGCATACATGGAGACCAACAGTTT\n",
      "TCGTGTCCCATTGTTCGAAGCTCTTAAGCAGCTTATCGTCTCAGGTAACTGTCTGCTCTA\n",
      "CATTCCAGAGCCTGAACAGGGAACCTACAGTCCTATGCGAATGTACCGCTTAGTGTCCTA\n",
      "CGTTGTTCAACGTGATGCTTTCGGTAACATCTTACAGATTGTGACTCTCGACAAGGTAGC\n",
      "GTTTAGTGCTCTACCGGAAGACGTGAAGTCTCAACTCAACGCAGACGACTATGAGCCTGA\n",
      "CACCGAGCTGGAAGTATATACACACATATACCGTCAGGACGACGAGTATCTTCGCTACGA\n",
      "GGAAGTGGAGGGCATTGAGGTAGCAGGGACCGATGGTTCATATCCGCTGACTGCCTGTCC\n",
      "ATACATCCCGGTACGAATGGTTCGACTAGATGGTGAAGACTATGGTCGTTCTTATTGCGA\n",
      "GGAGTATCTTGGGGACCTGAACTCGCTGGAGACGATTACAGAAGCTATCACCAAAATGGC\n",
      "TAAGGTAGCCTCCAAGGTGGTTGGACTCGTTAACCCGAACGGTATCACGCAACCTCGTCG\n",
      "TCTGAACAAGGCGGATACAGGTGAGTTCGTGGCTGGTCGCGTTGAGGACATCAACTTCCT\n",
      "GCAACTGACGAAAGGTCAGGACTTTACGATTGCCAAGTCGGTAGCTGACGCTATCGAGCA\n",
      "ACGTTTAGGTTGGGCCTTCCTTCTTAATAGTGCTGTTCAGCGTAATGCCGAGCGTGTCAC\n",
      "TGCTGAAGAGATTCGTTATGTTGCTGGCGAACTGGAGGCGACCTTAGGTGGCGTGTACTC\n",
      "AGTACAGTCACAAGAGCTTCAGTTACCTATCGTCCGTGTGCTGATGAACCAGCTTCAGTC\n",
      "TGCTGGCATGATTCCTGACCTTCCGAAAGAAGCGGTAGAGCCTACTGTCTCCACTGGCCT\n",
      "TGAAGCGTTAGGTCGTGGTCAAGACTTGGAGAAGCTAACTCAGGCAGTCAACATGATGAC\n",
      "AGGGCTACAGCCTCTGCAACAGGACCCAGACATCAACTTGCCGACCCTTAAGCTGAGACT\n",
      "TCTGAATGCCTTAGGCATTGACACCGCTGGTCTGCTTCTGACTCAGGACGAGAAGATGCA\n",
      "GCGTATTGCTGAACAATCAGCTCAAGGAGCTGTGGTCAACGGTGCGTCTGCTGCTGGCGC\n",
      "TAACATGGGTGCTGCTGTAGGTCAGGGAGCTGGTGAGGACATGGCTCAAGCCTAAACTAT\n",
      "CACTATAGGAACACCGAAAGTAAGAGCGACTAAGCCGACCATATGGTAGCCGTTAGTCCC\n",
      "ACACTGCAAGTAGGTGTTTACCTATTAACGACTTAAAGGAGAATGACTCAATGTCTCAAT\n",
      "CAGTTTATGCCGAGTTCGGCGTTAGCTCTAATGCAATCACTGGTTCCGTTGAGGACCTGA\n",
      "ACGAACACCAGAAGTCTATGCTTGAACAGGACGTAGCTGTTCGCGATGGCGATGACGCTA\n",
      "TTACCTTCAAGCAACTGGAAGCCGAAAACGAAGAGGCGGCCGAAGAAGACGAAAACGTCG\n",
      "AAGAGACTGAAGGTGAAGAAGACCACGAGCCAGATGACGAAGAGTCTGAGACCGATGGTG\n",
      "AGCAGCCTGAGTTCATCGAACTGGGTGATGCACCGAAAGAGTTGACAGAAAGTGTCACAG\n",
      "CTCTGGATGAAAACGAAGCTGCATTTGACGACATGGTGTCTTCTGCTGTAGAAGCTGGCA\n",
      "AGGTCACTGCTGATGAAATTACCGCTATCAAGGCTGAATACGCCAAGGACGGTAAGCTGT\n",
      "CTGAAGCATCCTACGCCAAGTTGCAGGAAGCAGGTTACACCAAGCGTTTCGTAGATTCGT\n",
      "TTGTACGTGGGCAGGAAGCTCTGGCTGAACAGTATGCTGCTGGTGTGGTTCGCTACGCTG\n",
      "GTGGTGCTGAACAGTTTAATCGAATCCTGTCACACCTTGAGTCCAACGACCCATCAACTC\n",
      "GTGAAGCACTGGAAGCTGCTATCGTTCGTAAGGACATTGCGACTACAAAAGCTCTGCTGA\n",
      "ATCTGGCTGGCAAGACTCTGGGTAAAGCTGTGGGTGTTAAACCTCAGCGTACCCTCACAG\n",
      "CTCAAGCTAAACCTGTGGTCGCACCTAAGGCTCCTCAGACCGAAGCATTCAGCTCCAAGT\n",
      "CTGACATGATTAAGGCTATGAGTGACCCGCGATACCTGCGTGACGCTAAGTACACGATGG\n",
      "AAGTTCGCGCTAAGGTAGCTGCCTCAAGCCTGTAGGACTAAACTATCACTATAGGGAGAC\n",
      "CAAGAGATAGACTCAAGTTTTCCCTATTACTTCAGTCCATACGGATTGGGCATACAGTAA\n",
      "GTAATAAACTTTATCTTTCAATTGAATAGGAGAATTATCATATGGCAAACGTTCCGGGTC\n",
      "AGAAAATTGGTACAGACCAAGGTAAAGGCAAATCCAGTTCCGACGCTCTGGCGTTGTTCC\n",
      "TGAAGGTATTTGCTGGTGAAGTCCTGACCGCATTCACTCGCCGCTCTGTAACTGCTGACA\n",
      "AGCATATTGTCCGTACCATTCAGAACGGTAAGTCTGCTCAGTTCCCGGTCATGGGTCGCA\n",
      "CCTCTGGTGTGTATCTGGCTCCGGGTGAGCGACTGTCCGATAAGCGTAAAGGTATCAAAC\n",
      "ATACCGAGAAAGTGATTACCATTGATGGTCTGCTGACTGCCGATGTGATGATTTTCGACA\n",
      "TTGAAGACGCGATGAACCACTATGACGTGGCTGGTGAGTATTCCAACCAGTTGGGTGAAG\n",
      "CTCTGGCTATCGCTGCTGATGGTGCGGTTCTGGCTGAAATGGCTATCCTGTGTAACCTCC\n",
      "CGGCTGCATCCAACGAGAATATTTCCGGTCTAGGCACTGCGTCCGTACTTGAAGTAGGTA\n",
      "AGAAAGTTGACCTCGACACCCCAGCTAAACTGGGTGAAGCAATCATTGGTCAATTGACCA\n",
      "TTGCTCGTGCGAAGCTGACCTCCAACTACGTTCCTGCTGGCGACCGTTACTTCTACACCA\n",
      "CGCCGGACAACTACTCTGCAATCCTAGCGGCTCTGATGCCGAACTCTGCTAACTATGCTG\n",
      "CGCTGATTGACCCAGAGACTGGTAACATCCGTAACGTGATGGGCTTCGTTGTTGTTGAAG\n",
      "TTCCACATCTGGTGCAGGGCGGTGCCGGTGAGACCCGTGGTGACGATGGTATCACTATTG\n",
      "CTTCCGGCCAGAAACACGCATTCCCGGCAACTGCTACTGGCGACGTTAAAGTTGCTATGG\n",
      "ATAACGTTGTGGGTCTGTTCTCTCACCGTTCTGCTGTTGGTACTGTGAAGCTGCGTGACT\n",
      "TGGCGCTGGAACGTGACCGTGACGTCGATGCTCAGGGCGACCTGATTGTCGGTAAATACG\n",
      "CTATGGGTCACGGTGGCCTGCGTCCTGAAGCTGCTGGCGCACTGGTTTTCAGCCCAGCGG\n",
      "CGTAAGAACCTTTAGCCAACCTAACGTCGCTACAGTAGCGGCTGCACCTGAAGAGGAGAC\n",
      "TCTAACTCCTCAACAGAAAGCTGCGCGTACTCGTGCTGCGAACAGGGCCGCTAAACTGGC\n",
      "TGAGTCCAACAACTAATTGAAACCCCTTGGGTGCCTTCGGGTGCTTGAGGGGTTTTTTGC\n",
      "TTAAAGTGAGAGGAGACTTATGGCTCAATACATTCCACTGAATGCTAACGATGACTTAGA\n",
      "TGCCATCAACGATATGTTAGCTGCTATCGGTGAACCAGCAGTCCTACAGCTTGACGAAGG\n",
      "GAACGCAGACGTCTCGAATGCTCAACGCATCCTGCATCGTGTCAATCGTCAGGTCCAAGC\n",
      "TAAAGGCTGGAACTTTAACATCAACGAAGCTGCTGTCCTGACACCAGATGTCCAAGACAA\n",
      "TAGGATTCGATTCCTACCGTCTTACCTTCGGGTCATGACTGCTGGTGCTACCAGTTACTA\n",
      "CAGCAATATGGGTGGCTATCTATACGACCTGTCCACTCAGTCTACGACCTTCACTGGCCC\n",
      "AATTACAGTCGAGCTGGTAGAGATGAAGCCGTTCGCTGAGATGCCTGTGGTCTTCAGGGA\n",
      "CTACATCGTCACTAAAGCTAGCCGTGAGTTTAACGCTAAGTTCTTCGGTAGCCCAGAGTC\n",
      "TGAACAGTATCTTCGTGAGCAGGAAGCAGAACTATATCAACAGGTGATGGAGTACGAGAT\n",
      "GGACACTGGTCGCTACAACATGATGTCTGACATCGGGAGGGACTAATGGCTAGTCACAAC\n",
      "CTGAAGATTCACCGCCAGCACTTTGGTCCGGTACAACTAGGTCTAAAGACATCAGAACTG\n",
      "AGACTTAACGACCGGGACTTTAAGGTTGGCGATTGGCTTATCCTGAATGAGTGGGACAAC\n",
      "GGATACACTGGACAACAGGTAGCGCGTAAGGTTGTCCACATTGCTGACGTTGGTGACATC\n",
      "GCTGAAGGATACGTCCTAATGAGTATGATTTAAGGAGGGACTATGCCGCTTATCACACAA\n",
      "AGCATAAAAAATCTTAAAGGGGGCATTAGCCAGCAGCCTGATATACTGAGGTTCTCAGAC\n",
      "CAAGGCGAGGCACAGGTTAACTGCTGGTCATCCGAGAGTGATGGCCTCCAGAAGCGCCCA\n",
      "CCTACAGTCTTCAAGCGCAGACTTAACATCGACGTGGGGAGCAACCCTAAGTTCCACCTG\n",
      "ATTAACCGTGACGAGCATGAGCAGTATTACATCGTGTTCAATGGGTCCAACATTCAGGTA\n",
      "GTTGACTTGAGTGGCAATCAATACTCAGTGTCAGGTGAGGTAGACTACGTTAAGTCCTCC\n",
      "AACCCACGAGATGACATCAGTGTCGTTACCGTTGCAGACTATACGTTCGTAGTTAACCGT\n",
      "AAGGTTGTCGTCAAAGGTGGAAGCGAGAAGTCACACTCTGGCTATAACCGCAAAGCTCGT\n",
      "GCTTTGATTAACCTGCGTGGTGGACAGTATGGTCGCACCCTTAAGGTGGGAATCAACGGT\n",
      "GGCGTTAAGGTCTCACATAAGCTACCAGCAGGTAATGATGCCGAGAATGACCCACCTAAG\n",
      "GTGGATGCTCAGGCAATCGGCGCCGCTCTGAGAGACCTTCTTGTTGCTGCCTATCCTAAC\n",
      "TTCACGTTCGACCTTGGGTCTGGATTCCTGTTAATCACTGCCCCTGAAGGGACCGACATT\n",
      "AACTCAGTGGAGACGGAGGATGGCTACGCTAACCAGCTAATAAGCCCAGTCCTAGACACT\n",
      "GTGCAGACAATCTCTAAACTACCTCTTGCCGCTCCTAATGGGTATATCATTAAGATTCAA\n",
      "GGTGAGACAAACAGTAGCTCCGATGAATACTACGTGATGTATGACTCCAACACTAAGACG\n",
      "TGGAAGGAGACAGTGGCGCCGGGAGTTGTCACTGGTTTTGATGTCACCACAATGCCACAT\n",
      "GCTCTGGTCAGACAGCCTGATGGCTCCTTTGAGTTTAAGACTCTTGATTGGTCCAAGCGT\n",
      "GGTGCTGGTAACGATGACACAAACCCTATGCCTAGCTTCGTGGATGCTGCGATTAACGAT\n",
      "GTGTTCTTCTACAGGAACAGACTTGGGTTCTTGTCTGGTGAGAACGTAATAATGAGCCGT\n",
      "TCAGCCAGCTACTTTTCGTTCTTCCCTAAGAGTGTGGCGACACTGAGTGATGATGACCCT\n",
      "ATTGATGTAGCTGTAAGTCACCCTCGAATCTCAATCCTTAAGTATGCAGTTCCGTTTAGC\n",
      "GAGCAGCTACTACTTTGGTCGGATGAGGTTCAGTTCGTGATGACAAGCTCTGGGGTCCTT\n",
      "ACCTCGAAGTCTATCCAGCTTGATGTAGGCTCAGAGTTTGCCTTAGGTGACAATGCCAGA\n",
      "CCGTTCGCTGTAGGACGCTCAGTCTTCTTCTCAGCGCCTCGTGGGTCATTCACCAGCATT\n",
      "AAGCGATACTTTGCTGTAGCTGATGTGTCTGATGTGAAGGATGCCGATGATACAACTGGT\n",
      "CACGTACTGTCCTACATCCCTAACGGGGTGTTTGATATTCAAGGTACAGGTACTGAGAAC\n",
      "TACATCTGCGTCAACTCTACAGGTGCATACAACAGAATCTACATCTACAAGTTCTTGTTT\n",
      "AAGGACGGCGTACAACTTCAAGCCTCTTGGTCACACTGGGAGTTCCCTAAAGATGATAAG\n",
      "ATTCTGGCGTCTGCCTCTATTGGCTCTACCATGTTCATCGTTCGTCAGCATCAAGGAGGG\n",
      "GTAGACATTGAGCACCTTAAGTTCATAAAGGAGGCAACTGACTTCCCGTCAGAGCCATAT\n",
      "AGACTACACATTGACTCAAAGGTGTCTATGGTAATTCCAATTGGCTCATATAATGCTGAC\n",
      "ACCTACAAGACTACGGTTGACATTGGTTCTGCTTATGGTGGAAACGCTCCGTCTCCCGGT\n",
      "CGGTACTATCTGATTGACAGTAAGGGTGCTTATGTGGAACTTGGGGACTTGACCGGTATC\n",
      "TCTACTGTGGTTACCCTCAACGGCGATTGGTCAGGACGTACAGTGTTCATCGGACGGTCC\n",
      "TATCTGATGTCCTACAAGTTCTCACGGTTCCTGATTAAGATTGAAGATGATAGTGGCACT\n",
      "CAGTCTGAAGACACTGGTCGTCTACAGCTTCGTCGCGCTTGGGTCAACTACAAAGACACT\n",
      "GGCGCTCTGAGACTCATTGTCAGAAATGGTGAGCGGGAGTTCGTAAATACCTTCAACGGG\n",
      "TACACACTTGGTCAGCAAACCATCGGGACTACAAACATTGGCGACGGGCAGTATCGCTTC\n",
      "GCTATGAATGGTAACGCATTGACCACGAGTCTAACCTTAGAGTCCGACTATCCTACCCCA\n",
      "GTGTCAATCGTTGGTTGTGGCTGGGAGGCGTCATACGCTAAGAAAGCTCGTTCCGTTTAA\n",
      "CTTATTGAATGGCCTATAGATTTACCTTAACTATCACTATAGGGACTATAGGCCCTTTAA\n",
      "GTTATAATACTTTAATAGGAGACTTTATGTATATTCGTAAAGCTACGGAACCAGATGTCC\n",
      "ACTACTTTCTGTGGCATCTTTCAGCAGATGATGTTAATGAATGCAAAGCAAACTACGGGT\n",
      "CAACCGTGGGTCTCTCTGAAAGACTGCTTAAGCACCTATCTCCGTCATCTGTGGTTTTAA\n",
      "CGAACGGTGTAGGCGAAGTGTTTGCCTATGGTGGGAACCAAGGGGATAACGTATGGTTCT\n",
      "TGACTTCAGGTCTGGTCTACAATCTGGGGCCTAAAGAGAAGCGAGAGTTCATAAAGCGAA\n",
      "TCTCTGAGTACAGGGACTTAATGTTAGACCAATACGGGACCATCTGGAACTACGTGTGGT\n",
      "CAGGCAATAAGTCTCACATTAAATTCTTGAAGTTGCTTGGGGCTAAGTTCCATGATGATT\n",
      "GGACTATCAGCCCAGTAACTGGTGAGCGTTTTCAATTATTCACTATCTCTAAGGAGGTGA\n",
      "ACCGTGTGCGAGCCAATTAGCATCGGTCTAGGAGTTATGTCTGTAGCCGGGGCCGCCATG\n",
      "TCCGCATCCCAACAGGCCAAGTCTGAAGGTGCAGCAATTGACGCTCAGAACCGACAGGCT\n",
      "CAGGAAATGGTTAAGCAGATGAACTACTCTGACGCCAACCTGAAGATGCAGGAGCGAGAC\n",
      "CTGAAGGAGCAGCAGATGGCTGAACTGACAGAGACCACGCTCAACGGTATCCGCAATCAG\n",
      "GGTATGGTCCGAGCTGCGGTGGCTGAGTCCGGTCTGGAAGGTAACTCTATGGATAGGATT\n",
      "GAGCGTCAGGTCAAAGGAGATACAGTCAAGGAGAGAGCAGGGATTACCGAAAGTTACAAC\n",
      "CGCGATTATGCGGCTATCTTTGGGAACCGTATCGCCAACATTGAGAACACCAAGTCTGCT\n",
      "ATCCGTGGTCAAGGTAAAATCATCAAGACCAGCCCACTAGCTCATGCACTTAATGTTGCT\n",
      "AACGCCGGGATGCAAGGTTACGCCGCTGGTGAGTCAATCTCTGGAGCGTCCAGCTCTGGC\n",
      "GGTGCTGCACCAATTAGTGCTGCTAAAGGCACACCTACAGGTCATATCTAAGAGGAGGAC\n",
      "TAATGGCTAGTAATATTGAATCAGCTCTGGCTAATCGGACTATGGGTCGTGGCAGAGCGC\n",
      "CGGGTAAAACTATCGCCGTCAACTACCAAGCAGCTAACGTTCAGGCTCCGACTGGAGATT\n",
      "CCGGTCTGGCTCGTGCGTTTACTAACTTCGTTGAGGCTGGTACAGGATTGTACAAGCAGT\n",
      "TCAAGGACGAGGAGAAGACCAGAGCTGACGAGAGGTCTAACGAGATTATCCGTAAGCTGA\n",
      "CACCTCAGCAAAGACGTGAGGCTATCCAGAACGGCACACTGCTGTATCAGGATGACCCTT\n",
      "ACGCTATGGAAGCACTTCGAGTCAAGACTGGTCGTAATGCTGCTTTTGCTGTAGACGACG\n",
      "AGATTAACGTTAAGATTCAGAACGGTGAGTTCCGCACACGTCAGGAAATGGAAGAATATC\n",
      "GCCACCAGCGACTTCAGGATGCCGCTAAGTCCTATGCTGAAGAGGCGGGTATTAACCCTA\n",
      "CAGACGAGTTCTTCCAGCGTGGATTCAACGATAACATCACGGACCGAAACATCGCTATCT\n",
      "ACGGGTCCTTCAATAAGTATTTCAGCAAGCAGTCTGAAGAGACAGCAATGTTGAACACTC\n",
      "GTATTGAGCTGAACTCGTTCCTTAACGATGGGGACCTGATGCGTTCGCCTGAGTCTGGAA\n",
      "AGACCTTTATGGCCTACCTTCGGGATGGATTGACGACTGCTGCTATACCTTCTGACCAGA\n",
      "GAGCACGAGAGGTCATCACCCAGACTGTCCGTGATGCAATCCAGAAGTCAGGAGGCTCAA\n",
      "ACTTCCTACAGCAAGTCCGTGGCGAGCGTATCACCCTTAACGGTGTTGACGCTACCATCG\n",
      "AAGAGATTGTAGGACCTGACATCTTCAATGCTGCTATGGTTGAGGCACAAGGTACTGAGT\n",
      "ATAAACTGGTGGCTAAGTATCAGGAAGACTTGTCGTTAGGCGTTCAGTCCGCAATTCTTC\n",
      "AGGACGACCCAACCATTGGTCTTGCCCAGATTCAGAAACTCAAGGAGCAGAACAACCAGC\n",
      "TTCAACCCGGTGAAGAACTCACTCCTCAGCGCCAGATGCTTATTAACGCAGAAGCCACAC\n",
      "TACTGGAAGCGGTAAAGCGTAAGTCTGCTGAACAGGCGAAGGAGAACACGAAGTTAATCC\n",
      "AGACCCAGAACAAGCAACTGGTCATTGACCAAGTGTATCAGCGACGTCTGGCTGGGGACA\n",
      "ACGTGTCCACCAACTATGAGGACCTTCCGGTCTCTGACGCTACAGGTGAGTTTAAGCGTT\n",
      "CAGACATGAACAACTATGCGGCTGCCAAGCTACAGCAGATTGACCAGATGGACATCCCTG\n",
      "AGGCTGCTAAGGACGCTCAGAAGGTGGCATTGTTGAGGGCTGACACTAACAACGGTCCGT\n",
      "TCCGTAATGCCTTCCAGACGCTGACTCAGGATGCTGCTGGTGAGTGGCAAGCTGCGGTCA\n",
      "TCCGTGGACAGTACGACCCAGACAAGATGAAACGCTTCGAGTCTCTGCGTAAGGTCTACA\n",
      "CTCAGGACCCTTACAGCTTCGCGGCCCTCTACCCTGAGCAGGCCCAATTGTTCACGACTT\n",
      "TCGATCAGATGGACAAGATAGGTCTGGACCCTCAGACGCTGATTGAAGCTGACAAGCAAG\n",
      "CTGCAAGTTTATCGCGTGAGATGCGAAGTGAGGCAGACAAGGAGTGGAGAGAGCTGACGA\n",
      "ATGACTCAAAGGAGACTCAACTCTCACGGATGCCGACTGAATTGAAGATGATTGCACGAA\n",
      "AAGTATGGGACTCATGGTACTACAGGACAGGTAACGCTGACGGAGCCACTCAGAATACCC\n",
      "AGAAGTGGTTGAATGAACACACCGTGACTTTCACCGATGATGGTCAGGATGGTAAGTCCA\n",
      "TCGGCATGGTGTCCCACCATCAACTTATGGTCGGTGATAACCCAGAGTCGTGGCAGGTTG\n",
      "GTCGAGACATTATCGACATAGCTCGCAAGCAGCTCATTAAGACCAACCCTTGGGTAGTGA\n",
      "ACTCTCAGCTCTCTGTAGTTGAGCAGAACGGCTCTGTCTTCCTGCAGAGCGCTAACGGAA\n",
      "CCCTACGCATTCGATACGATAAAGAACTTGTTGGTAAACTTTACCGTGAACAACAGAAAC\n",
      "AAGCCGAGGATAAAGCCTACGCTGAGGCTGAACGTAAAGCCAACACTCGCGCTCGCATTG\n",
      "TCAAGTCTAAAGAGGCTGGTGATAAGCGTCGAGCGGAGCGTGAGGTCAACATCGAGAATC\n",
      "GCGGTGGGATGTACAATGACGTCTCACTGGAGGGTATCGCAAACATATTAATTGGTAAGG\n",
      "AGTAACATAATGGCGACTCGCGGTATTCGCAATCACAACATTGGAAACATCCGTAAGTCC\n",
      "AAAGACCAATGGGAAGGTTCTGTAGGGGATGATGGAGAGTTCGTTGTATTTGACACTCCT\n",
      "GACTCTGGCGTCCGCGCTCTGGCGAAGAACCTGATGTCATATGGTCGACAGGGCTATGAC\n",
      "TCTATCGAGAAGATTATCAACCGATGGGCACCACCTAGCGAGAACGACATTCAGTCTTAC\n",
      "ATCGACTCTGTTGTGGCTGCGACTGGCATTCCAGCTACCCAAAGTATCGACCTGACGAAC\n",
      "CCTGATGTCCTAGCCTCTCTGTCTGAGGCTATCGGCTACCACGAGACTGGTTCACGGTAT\n",
      "GACCCTGAGGTCTACAAGCTCGGAGTCTCTCGCGCCTTAAGTGGTGCGGGAGGCATCACC\n",
      "CCAAAGACTCCACCAGTAAGCGCTAACGTCTTCGACGCTCTCACGGAGGGATTGAAGGCT\n",
      "AAACCTAAAGTCGCTCTTGGAGAGAACCTTCCTGGCGTTACTGGCATGAACATCGAAGGT\n",
      "CAGGAACCTGAGGCTCCTAACGAGTCTTTCGGAGAGATGTTCTATAAGTCAACTGGCGAG\n",
      "ACGCTGGACCAAAGAGCTGACCGCTCTACATGGTTCGGCTTCGGTGGAGCTGCTGAGGCT\n",
      "GAAGTGAAGAACTCTCTGGTGGGCGTGGCTATCCGCGCTGGTCAGACCGAGGAACCACTT\n",
      "GATGTCATTGGCGATGTGTTTAACCCTACCCGCTGGAATAACCACAAGTGGACTCGTGAG\n",
      "GAGCTGGACCAGATTCGTAACGCTGGGGTTCTGCCTCAGTATTACGGGGTCATTACTGGT\n",
      "GGCTCTTCTCAGAATCTGACCGAGCTGATTAACTTGGCGCTGGAGAACCAGAAGTTAGAC\n",
      "GCTGAGAAGGCTAAGGCCGGGACTGGCGCTCAACTGGCGGCTGGTGTGATTGGTGCTGGC\n",
      "GTGGACCCACTTACCTATGTTCCTATTGCCGGACAGGTAGGTAAAGGCGGTAAGCTGGTT\n",
      "AACAAGATGTTCACCGTGGCTGCTCAGTCTGGTGCGGTGGCTGGCGCGTCTGAGATTGCT\n",
      "CGTACCTCTGTTGCCGGTGGTGATGCTCACGTTGCCGAGGCAATTATGGGAGGCGCTCTC\n",
      "TTCGGTGGTGGCATGACTGCTATCGGGGACGCTCTTGGCAAGGTTCTTGGAAAGTCGACA\n",
      "AATGAGTTCGCTGGCCCAGCTACACGTCTTGAGGCACGTGAGACAGCTCGTAACGTTGAT\n",
      "GGTCAGGACCTGTCTCGTCTCCCTATTCAGGAAGGTGAGCAGACCTTTAGTTATCAAGGC\n",
      "GTTAAGTTCGCTGACGTTCCGAACGAGCCGGGTAGTGTACGACTGGAAGATGGTTCAATC\n",
      "CTGATTGGTGAGAACCCTCTGAACCCTAAGACACGTCAAGTCTTTGACGAAGTGATTGAG\n",
      "CCAGAACGTGCCGCTGCTGGTGTGAACCTTGGGGGACTTACCGAGATTGGCCTGAAGCTG\n",
      "CTTCGGTCTGAGAACCCTGAGATTCGTGGAGTCGCCGCTGACTTAGTGCGTTCACCCACT\n",
      "GGTATGCAGTCTGGGGCTTCAGGTAAAATCGGGACCACTGCGTCAGACGTATTCGAGAGA\n",
      "CTTCGTGCTGTGGACCACAGGGCGTACAACGACTTGGATGACGCTATAACAGAATCTTTG\n",
      "GAAGACCCGTTGTATTCCCTTAAGTTCTGGCGTGACCGTTCAGCTCAGCGTGTGGACGTC\n",
      "TATGAACGTGCATCTCTTGCTATCGAGGACACCTCAGGTCAGCTCAAAGCTCAGCTTACA\n",
      "TCTAAAGAGCTTAAAGTATACGACATTATGAAGCAAGTTCTGGACTCTAAGCGTGAGATG\n",
      "ATGGAGAACCCTGCAATGTTCGGGAACCCTAAGGCTGTAAACATCTTCCCAGGCAGTCGC\n",
      "TTCAAAGGGTCTTATGTACCTATCGTCTACAGCAGTAAGAAGAAGCAGATGTACACTGAG\n",
      "ATGCTTGGTGGGGCTGAAGGTTTCCAGAATGCTATCAAACAGTCATGGCTGGCAGGATAT\n",
      "GCATCACGTCCACACGTCAAGCAGAGACTGGACGATGCATTAATGGCTGAGAACCCTAAG\n",
      "ATGACTCCTGAGGAGCTGGCTAAGGCAGTCGATAAGTACGCCAACGACAAGGCATATGGT\n",
      "ATCTCACATACTGAAAGATTTGACCGCTCGTCCGTCATGGAGGAGAACATCAACGGTCTG\n",
      "GTAGGTCTGGAGAATAACAACTTCCTTGAGGCTCGTAACCTGTTTGATAATGACGTGGAG\n",
      "ATTACCCTGCCTAATGGTGAGACCTTCAGTGTAAACAAGCTTCGTGAGTTCGACATGGAT\n",
      "AAGGTTATTCCAGCCTATAACCGTCGAGTTAATGGTGACATTGCTATCATGGCTGGTACA\n",
      "GGCAAGACCACGAAGGAAATGAAGGACTTAGTTGAGACCATGATGAACAAGGCTGGTGAT\n",
      "GACGGCAAGTTGAAAGGTGAGGTCTCGACCTTACGTGACACCTTGAAGATTCTCACTGGT\n",
      "CGTGCTCGACGTGATGGTGCTGATGATGCAGCCTTCGCTACAGTGATGCGCACAATGACA\n",
      "GACCTATCGTTCTTCGCTAAGAATGCCTATATGGGAGTTCAGAACTTAACGGAGATTGGT\n",
      "GGTATGCTAGCTCGTGGTAACGTTCGTGCGCTGCTGCATGGAGTCCCGATGTTCCGTGAC\n",
      "CTAGCCTTCCGTAGCAAGAAGATGGGAGCCTCGGAGATTAAGGAACTTCACAACGTTGTC\n",
      "TTCGGTAAGGAGCTTGACGACTCCATCCGTCCATCAAAACAGGATGTCATCGACCGCCTG\n",
      "CGGGCGTACAGCGCCCTGAGTAAACCCGTAGCTACCGCTTTAGGTTCTGCCAAGTATTAC\n",
      "ACTGGCGAACTTGCTGTCCGCTCTCCGTTTACGAAGGTCCTGAACGGGACGGCGAACTAC\n",
      "CTGCTGGATGCTGGACGTCAGGGATTCTTGTCTGACATCGTGGAACACAGCCTTACCGGA\n",
      "AGTAAGCGTAAGTTTGATGACCGCTGGTTGAAGACCGCTGGTATTTCGGACGAGCAGTGG\n",
      "AAAGGCATTAAGTCCCTCATCCGTGAGTCAGTGACTCGTGGTCCAGACGGGAAGTACACC\n",
      "ATCAAGGACAAGAAGGCGTTCAGTCAGGACCCAAGGGCTATGGACCTCTGGCGTATGGGT\n",
      "GATACCATCGCCGATGAGACCCTGCTCCGTCCGCATAAGCTGAGCAACATGGATTCCAAG\n",
      "GCGTATAGGCCGTGGGCGAAGGCTGTATTGCAGTTTAAGAACTTCGTCATCAAGTCCATC\n",
      "AACGGGCGGACCATGCGTACCTTCTACAATGCCACCAAGAATAACCGAGCGATGGACGCT\n",
      "GCACTGTCCACGGTTATGTCTATGGGTCTTGCTGGAATGTACTACATGGCTCAGGCACAC\n",
      "ATCAAGGCATACGCTATGCAGGACGGGCGTGACCGTGAGTACCTTAAGCAAGCCCTTAAC\n",
      "CCGACAATGATTGGCTATGCGGCTCTGTCCCGTAGCTCCCACTTAGGTGGTCCTCTTGGG\n",
      "GTCGCCAACATTCTGGGTGGAATCGCTGGGTATGAGGACACTAAGATGCTCCGCTCGTCA\n",
      "ATCCTTCCGCGCTCACCTACCGATAAACCTGAACGTGCTATTACCTATGGCGCTGCGAAG\n",
      "AGTGACCCGGTGATGAATGTGGTTGGAAACTTTCTTGAGCAGGTTCCTGCTTTCGGATAC\n",
      "GCCGCTAACGTGGGGGCTTCGGCCTACAACTTGGCTGGATACCTGAAAGCTGACACACGT\n",
      "GTGAACGAACGTGACTACATGACAGGGATGTACAATACCTTCCGTGAGCTGGTCCCGAAC\n",
      "GACCCAATCACCCAGAGGCTTCTTCTCGGAACCTTTGAGGAACAAGGTATTCACATCAAG\n",
      "GACTAAACTATCACTATAGGAAACGGGCGCTCCCACAGGCCCTAAACACAACTTAAGGAG\n",
      "ACCATATGTCCGGAACTCTTACTAAATTCCCAACTGGAAGCACACAATATAAGATTAACT\n",
      "TCGACTACTTATCCCGTAAGTTTGTTAAAGTTACTTTAAGCAACTCTCTTGATGCTACCT\n",
      "TAAATAAAGAGCTAGTGGTTGGTAACGATTACCGCTTCCTTAGTGCAACCACATTAGAAA\n",
      "TACTAGCCAGTCAATCTGGGTTCGACGTCATACAGATTTATCGCCAAACGTCTTCAGAGC\n",
      "TTATGGTTGATTTCAGAGATGGCTCAGTATTGACCTCATCCGACCTCACAACGGCTGAGC\n",
      "TTCAGGCCATCCATATCGCCGAGGAGGGCAGAGACCAGACGGTTGAACAGGCCGTGCAGT\n",
      "TTGCGGAAGCTGCTGAGAAATCTGCCGAGGACGCTGCCGCATCACTTAGCGGGATTAAGG\n",
      "ACATAATCGCTAACAACAATATCGATTCAACAATCCGTTCAGACCTGTCCAAGTCGGATG\n",
      "GATACACAATGATTGGCGGACTGTCTGAACGATTCAGTCTACCCTACAAGTGGGTGGTAG\n",
      "TGGATAACGCTCCTTATAATGGCGACCTTGAGAAGGCCGTATCCGAAAGTTCAGACGGTA\n",
      "CGGGATTCCTATTAGGTATAGGGAAATCCTACGACTTCCTTACATTCCGCAACGCTCGTA\n",
      "ACACTAAGAAGAACCTAACGTTCGTTGGTGGAGGGGTCCCGGAGCTGTCTGCCGATACAC\n",
      "GGAGCCTTGTTGCTGGTACTGGAACGATAATCAGAGGACCAATGAAGACTGAGGCTGGTG\n",
      "GGTTCCAGTTAGTGAATCTAGGCGTTGACTGTGGTGGCGTGGTGTCAGACGAACTTCTAG\n",
      "GCGGTCAATGGGACGACCCTATCCAGTGCTATGGTATTGGTAAAGAGGCAAATGTTTTTA\n",
      "TAGATAACGTTAAAACCCTGAGCCATACAAACCTCACTGGTCCAAACTACCCAGGAACTC\n",
      "ACAGCTTGCTTCTTGAGATTCTTGACGGAGCTTATGTTGGAAAGGTAGAACTGATTGGCG\n",
      "GCTACCACGGTCTGACCCTTAAGGGGACTAACTTAGATGTGATGGACGCCCACTGTTACG\n",
      "GACAAAGAGGCGATGCGTTTATTATTAAGTCTGACTCTGCTGCGCCGGCTGCTGATATCT\n",
      "ACGTCGGAAGACTTAAGGTAGGGCTTCGTGATAACTCTAAGTTTACAGATGTTACTATGG\n",
      "GCGGCATCTACGACGCTCACGATGGGGTTAACATAGATAACGTGGAGATAGGCTCTCTTC\n",
      "AGGTCTACAACGCATCATGGGGGCTCATTCCCTCAAATGAAAATACTGGTTATATCAGCA\n",
      "ACTTGCGTATTGGGAACTACGTGGCTATGAACGTATATGGTAACTACTATTCGCTAGTGA\n",
      "TTGATGGTCGCTGCGTTGGTGTTAACATAGGGGAGCACCAGTGTTCTAATACATCTGGCG\n",
      "GAATCAGCATAGACCCTTCCTCTGTTAACGTGTCATTGGGTAACGGATTCTCAAAGTGTG\n",
      "CCACAAAGTCTGGTTACAAGCTGGGTGGTAACTCTCTGGTTCACGGGAACCTGATAGCCG\n",
      "ATGAGAATGGTGAGTACGGTGTTGATTATGCTGGTGGCCTAGGGTTCGATGCATCAAAAG\n",
      "TTCTAGGGTATAGTAACCCATTCGGTCTGGTGTCGACTGAGGTTAGCGCCCTTGGGTCAA\n",
      "CTACTGGACCTAAGAATGGATGGGGAGACACCGGGGACTTTTCATGGGTAATGCATGGCA\n",
      "AGATGTCGATAGTAAACGGAGCCCTTATTAAGGGAGCTGGGGCTGTTCCATACACATCCC\n",
      "TACCAATAACGGCGCCTCGCAGGAGAACGCGCATCGCTGCATGGGGTGTGCAAGGGTCGA\n",
      "GCATGATTCCAGTCGAGTGCTATGTGGAAGCGAATGGGGACCTTAACGTCGTTGGCTGGG\n",
      "AGTCCATTACAGACGGAAACGTGATAAACTTCTCCGGTCAGTACCTAGCTAAATAACTAT\n",
      "GGCCCTCTACGGAGGGTCAATAATGGAGGTTTTATGATTGAGTTCGACTTCAAGAATGAG\n",
      "GTCCTAAAAGCCTCGCCTATCGTCGGGACCGCTGCGGCTGATGGTGCCAGTCGGTTCTTC\n",
      "TTTGGGTTAACACTCAACGAATGGTTCTACGTCGCTGCTATCGCGTACACCGTGGTGCAG\n",
      "ATTGGCGTCCTAATTTACAAGACGATTAAGAGCGGAGGTAAGACATGACGCAGATGGACT\n",
      "TAGAGAAGTTCCTGTTAATGCTGGACACTGAACGCGCTCGACTCATGCTGCAAGACCTGC\n",
      "GGGATGACACAAAGCGTTCACCTCAGCTTTACAACGCTATCGAGAAGCTGCTAGCTCGCC\n",
      "ACAACTTTGTACTTAGCAAGGTGTCTGTGGATGAGAAACAGTTAGCTGACATGGAGGCGT\n",
      "TGAACAGGGAGTACGATAAGGTGCTCTCAGCTACTGAGGACAATGACACCGGGTATGGTG\n",
      "TACAGTAAGTGTTAGACTCAAGGTCATTACCACTATATGTAGTGGCCTTTATGATTAACA\n",
      "CACACTACTGGCGGAGACGCTACGTGGATTCTGATAAACGAGGAGGGCAACTATGCTCGA\n",
      "ACTTTTACGTAGATGCACCCCTTGGCTGGTGGCAGGACTCCTGTTCTGCTCAGGTTATTG\n",
      "GGTGGCAGACAATAAGTGGGAGGCGAAGGTAAACAATGAGTACATCACGAAACTTGAGGC\n",
      "ACGAGAGAATCAACGGGCTGCTGTCCAAGGTGAAATCGACAAAGTGTCAGCAGAGTGGCA\n",
      "AGACAAGATGTCCGCGCTGGAAGGCTCTACTGATAGGATTATTGCTGACCTTAACCGCGA\n",
      "TAACAAGCGGATGCGCGTCAAAGTCAACACCACAGGTATCACCGAGTCAGATTACAGTCG\n",
      "ATGCTTCCCTGATGGTCGAGTCGAACTACACCCAGAGACTTCTAAAAGTCTTATCCGAAT\n",
      "AACGCAGGAAGCCGACCTTAAGGAGAAAGCTCTTCAGGATACCATACGTAGGCTTCAGCA\n",
      "AGAGAAGGAGAAGTGATGCTCAGTGATTCGTATGTCTATGATGAGTCCAGTAAGTCTGGC\n",
      "CTGAAGTCCCTGCGTTCTGGTAAGGATGTGGGGTGTGTGGGAAACCACGGGTATTATCAG\n",
      "GTAGGAACTCGACGTCATGGACTCCTGCTGGCCCATAGAGTCGTCTGGGAGCTAACCAAC\n",
      "GGTCCTATACCAGAAGGTATGGAGATTGACCATATTGACCGCTGCCGGACTAACAACAGG\n",
      "CTTGACAACCTCCGGCTAGTTACTTCAAGCGAGAACTCCTGCAACACCTCAGGACATTCT\n",
      "GACTCAAAGACTGGAGTTAAGAACATATCGTTCCATCAGAATGGCTACGCTATCGAGGTC\n",
      "ATGCGTAATGGAGTCCGTATGCGAACAACACGTCCTACTTTAGAGGAGGCTATCTCTGTC\n",
      "CGGGACCTAATGCTGGAGGGACTATATGTCTAAGAAAACACCTGCTGAAAGACAGGCTGA\n",
      "AATGACCCAAAGGATGAGGGACGACTTCGTGTTCTTCATCTTTGTTCTCTGGAAAGCGCT\n",
      "GAGCCTTCCGGTTCCAACCAGATGTCAGATAGATATGTGTAAGAAGCTGGCTGCTGGCGA\n",
      "TAACCGACGATTCATCCTTCAGGCGTTCCGAGGTATTGGCAAATCGTTTCTGACCTGTGG\n",
      "TTTCGTGGTCTGGAAGCTATGGAATAACCCGGACTTGAAGTTCATGATTGTGTCGGCCTC\n",
      "AAAGGAACGAGCCGATGCGAACTCAATCTTCATCAAGCGTATCATCGACCTCATGCCTCA\n",
      "GCTACAAGAGTTGAAGCCTAAGCAGGGACAGCGAGATGCAGTAATCAGCTTTGACGTAGG\n",
      "GCCTGCCAAGCCAGACCACTCTCCTTCGGTTAAGTCCGTTGGTATCACTGGTCAGTTAAC\n",
      "TGGTAGCCGTGCTGACATCCTGATTGCCGATGACGTGGAGGTTCCGAACAACTCAGCAAC\n",
      "TCAAGCTGCCCGTGACCGTCTGTCAGAGCTTGTGAAAGAGTTTGACGCTATCCTGAAGCC\n",
      "GGGAGGTACAATCATCTATCTGGGTACTCCTCAGAACGAGATGACCCTGTACCGTGAGTT\n",
      "AGAAGGTCGTGGGTACACCACTACTATCTGGCCCGCTCGCTATCCGCGCGACAGGAAGGA\n",
      "CTGGCAGTCTTACGGCGACCGTCTGGCTCCTATGCTTCAGGCAGAGCTGGAAGAGGACCC\n",
      "TGAGTCCTTCTACTGGCGTCCGACCGATGAAGTGCGCTTCGATGATACAGACCTGAAGGA\n",
      "ACGTGAGCTGTCATACGGTAAGGCTGGCTTCGCTCTACAGTTCATGCTGAACCCGAACCT\n",
      "AAGTGATGCCGAGAAGTACCCGCTGAAGCTCCGTGACCTTATCGTAGCGGACTTGGACCC\n",
      "AGCGTCCAGCCCAATGGTCTACCAATGGTTGCCGAACCCTCAGAACAAGCGTGAGGACGT\n",
      "GCCTAACGTTGGACTCATGGGTGACTCGTACCACACGTATCAGACTGTAGGTTCTGCCTT\n",
      "CAGTTCGTACACCCAGAAGATTCTGGTCATAGACCCTAGTGGTCGAGGTAAGGATGAGAC\n",
      "CGGGTATGTGGTTCTGTACCAGCTCAACGGCTACATCTTCGCTATGGAAGTAGGGGGTAT\n",
      "GCGTGGCGGCTATGAGGACTCTACGCTGGAAGCTCTGGCTAAGATTGGTCGTAAGTGGAA\n",
      "GGTCAACGAATACGTCATTGAGGGTAACTTCGGTGATGGTATGTACCTTGAGTTGTTCAA\n",
      "GCCTGTAGCGGCCCGTACCCATCCAGCGGCTGTAACTGAAGTGAAGAGTAAAGGACAGAA\n",
      "GGAACTCCGCATCTGTGACGTTCTGGAGCCTATCATGGGGTCTCACCGACTTATCGTCAA\n",
      "CGCTGCTGCTATCGTCCAAGACTACCAGACAGCCTATGATAAGGATGGCGTTCGTAACCC\n",
      "TATCTACTCTCTCTTCTACCAGATGACCCGCATCTCTCGTGAACGTGGAGCATTGGCACA\n",
      "CGATGACCGACTTGATGCGCTGGCTATCGGTGTGCAGTTCTTCGTTGAGTCTATGGCTAA\n",
      "GGATGCCAACAAAGGAGAACGTGAAGTCACTGAGGAGTGGCTGGAGGAGCAGATGGAGAA\n",
      "CCCACGGAAAGGCTTCGAGTCCATCCACACTGAGTTCTGGGACAATGGGGTCCGGGTAAC\n",
      "TCATGATACGGACGACGAGCTGGGACTAGGGTCATACGTTACGTTCCACTAGCTAAATGA\n",
      "ATAACTATAGGCGAAAACTGAATTAATACGCGGTTAGTAACCTATAGTTATTACCAGTCT\n",
      "AACCTACTGTTTTACAAGGAGTTTGGACTTAACTATCACTATAGGAAAGACCAATGGTTA\n",
      "CTTATAGTATTACTGTAGTGAATATACATATGCAGACTTTATGCAAGACCTTAGGAGGCA\n",
      "GACTCCGAGTTCTTACCTAAGGCTTGCACCGATGGAAGGAGGGTGATATTAATCATAATC\n",
      "CCTCCAATACAGATAGTCACCGACCATAGATACAGGAGGTATGTAGCATATGGCAAAGAC\n",
      "CAAAGCTGTTCTCAAAGCTCTGGCGACCAATAGAGCTACATACAGGTTTCTTGCTGCTGT\n",
      "TCTACTTGCTGCTGGCGTTACTGCTGGAAGTCAGTGGGTCGGGTGGGTCGAGACTCTCGT\n",
      "ATGTACTCTGGTGTCTCAGTGTAATTAACGCAATCATGGTAACAATAAACGAAAGACGGA\n",
      "CCTAAGGTCAGTTCATAGCTGACGCTACTATACTGACCTTAGCTACTGTAGTCAAGGACT\n",
      "TTAGGTAACACCTTAATAGAAGCTCACTTAGGGTCATCCTGCTTACTGGTCTATCCTAGT\n",
      "GTCGCCTGACCTACGGTCCTTGACCTACAGTGGCTGGGTCCTACAGTATGGACGATGAGT\n",
      "TTTGGACCAAAAGTTTGAGACCACATCTCACAGTTCAAGAACCTCAAGTCTCCCCCATAG\n",
      "GGTCACTTCCAGTCAGACCAAAGGCCCTACCCCTAGACCTTCGGTCTAAACTTCAGGTCA\n",
      "TTGACTCAAGGTCTACCAGAAGTCATGACGATAGGTCAAGGCTATGGCTTATGGAAGGAC\n",
      "GATAGGTAGGGACTATGGGTTGTCTCTGAGTCTCTATCTGTATAGTGTTATAGTATAACA\n",
      "TTACGTCCTACAACCTACAGTCTAACAACCTACAGTCTAACAACCTACAGTCTAACAACC\n",
      "TACAG\n"
     ]
    }
   ],
   "source": [
    "!cat spades_assembly_kmerXX/contigs.fasta | seqkit grep -r -p NODE_1_> longestcontig.fasta\n",
    "!cat longestcontig.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "952ec354-8e96-4620-9616-948f686de45a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Results\n",
    "'Salmonella phage PRF-SP9, complete genome\n",
    "Sequence ID: OR344853.1Length: 39956Number of Matches: 1\n",
    "Range 1: 35 to 39939GenBankGraphicsNext MatchPrevious Match\n",
    "Alignment statistics for match #1\n",
    "Score\tExpect\tIdentities\tGaps\tStrand\n",
    "73691 bits(39905)\t0.0\t39905/39905(100%)\t0/39905(0%)\tPlus/Plus\n",
    "'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "6e1b341f-1c31-4fdb-a9d8-d556f2b54353",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files'"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "c4c4d55e-901a-4459-a316-7e2a541ebe09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "d784e6af-372a-4448-bb5c-5b9ffd56e080",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "9cb277d3-d3de-470f-a2a0-40f54ec5dc43",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "java -ea -Xms300m -cp /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/opt/bbmap-39.01-1/current/ jgi.ReformatReads in1=SRR18218015_1.fastq.gz in2=SRR18218015_2.fastq.gz reads=10000 out1=SRR18218015_10K1.fastq.gz out2=SRR18218015_10K2.fastq.gz\n",
      "Executing jgi.ReformatReads [in1=SRR18218015_1.fastq.gz, in2=SRR18218015_2.fastq.gz, reads=10000, out1=SRR18218015_10K1.fastq.gz, out2=SRR18218015_10K2.fastq.gz]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t20000 reads          \t3617633 bases\n",
      "Output:                 \t20000 reads (100.00%) \t3617633 bases (100.00%)\n",
      "\n",
      "Time:                         \t0.312 seconds.\n",
      "Reads Processed:       20000 \t64.19k reads/sec\n",
      "Bases Processed:       3617k \t11.61m bases/sec\n",
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades.py\t-1\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz\t-2\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz\t-o\t/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.5\n",
      "  Python version: 3.10.8\n",
      "  OS: Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35\n",
      "\n",
      "Output dir: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz']\n",
      "      right reads: ['/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 7\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/spades.log\n",
      "\n",
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz: max reads length: 301\n",
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz: max reads length: 301\n",
      "\n",
      "Reads length: 301\n",
      "\n",
      "Default k-mer sizes were set to [21, 33, 55, 77, 99, 127] because estimated read length (301) is equal to or greater than 250\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-hammer /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (main.cpp                  :  76)   Loading config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/configs/config.info\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.004     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 8 threads. This might take a while.\n",
      "  0:00:00.004     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.004     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291663 Gb\n",
      "  0:00:00.005     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 2446643\n",
      "  0:00:00.080  2689M / 2689M INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz\n",
      "  0:00:00.203  2689M / 2689M INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz\n",
      "  0:00:00.328  2689M / 2689M INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 20000 reads processed\n",
      "  0:00:00.329     1M / 305M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.335     1M / 305M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 365192 kmers in total.\n",
      "  0:00:00.335     1M / 305M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.344     1M / 305M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 365192 kmers, 275912 bytes occupied (6.04421 bits per kmer).\n",
      "  0:00:00.344     1M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:00:00.352     7M / 305M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:00.632     7M / 305M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:00:00.635     7M / 305M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:00:00.637     7M / 305M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:00:00.680    16M / 305M  INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:00:00.719     7M / 305M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 244128\n",
      "  0:00:00.720     4M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:00:00.721    13M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz\n",
      "  0:00:00.938    13M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz\n",
      "  0:00:01.160    13M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:00:01.162    13M / 305M  INFO   K-mer Counting           (kmer_data.cpp             : 397)   There are 365192 kmers in total. Among them 227050 (62.1728%) are singletons.\n",
      "  0:00:01.162    13M / 305M  INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:01.221    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 7 non-read kmers were generated.\n",
      "  0:00:01.223    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:00:01.223    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 181104. Among them 72396 (39.9748%) are good\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13. Among them 13 (100%) are good\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 63139. Among them 28879 (45.7388%) are good\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 2.91571 kmers\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.00203\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 101288\n",
      "  0:00:01.224    13M / 305M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.94029,0.0291135,0.0209906,0.00960564),(0.0217745,0.959819,0.00703565,0.0113712),(0.0111453,0.00687005,0.959644,0.0223406),(0.00942844,0.0208064,0.0285425,0.941223))\n",
      "  0:00:01.224    13M / 305M  INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:01.225    13M / 305M  INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 8 threads.\n",
      "  0:00:01.385    13M / 305M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 346 new k-mers.\n",
      "  0:00:01.500    13M / 305M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 33 new k-mers.\n",
      "  0:00:01.628    13M / 305M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:01.635    13M / 305M  INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:01.636    13M / 305M  INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 8 threads.\n",
      "  0:00:01.641    13M / 305M  INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K1.fastq.gz and /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/SRR18218015_10K2.fastq.gz\n",
      "  0:00:01.762   198M / 305M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 10000 reads.\n",
      "  0:00:02.034   199M / 305M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:00:02.086   199M / 305M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:00:02.150    13M / 305M  INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 10187 bases in 2894 reads.\n",
      "  0:00:02.151    13M / 305M  INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 2725 bases out of 3607698.\n",
      "  0:00:02.151     1M / 305M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/corrected.yaml\n",
      "  0:00:02.151     1M / 305M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/python /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades/spades_pipeline/scripts/compress_all.py --input_file /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/corrected.yaml --ext_python_modules_home /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades --max_threads 16 --output_dir /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with pigz)\n",
      "== Files to compress: ['/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/SRR18218015_10K1.fastq.00.0_0.cor.fastq', '/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/SRR18218015_10K2.fastq.00.0_0.cor.fastq', '/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/SRR18218015_10K_unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K21/configs/config.info\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=21\n",
      "  0:00:00.002     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.002     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.002     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.002     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.006     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.006     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.008     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.010     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.010     1M / 27M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.014     1M / 27M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.108    63M / 63M   INFO    General                 (binary_converter.cpp      : 111)   9838 reads written\n",
      "  0:00:00.109    41M / 41M   INFO    General                 (read_converter.cpp        :  86)   Converting single reads\n",
      "  0:00:00.110    71M / 71M   INFO    General                 (binary_converter.cpp      : 111)   23 reads written\n",
      "  0:00:00.110    61M / 61M   INFO    General                 (read_converter.cpp        :  92)   Converting merged reads\n",
      "  0:00:00.110    71M / 71M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:00.111     1M / 33M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.112     1M / 33M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.112     1M / 33M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.112     1M / 33M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.112     1M / 33M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.115     1M / 33M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.115     1M / 33M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291652 Gb\n",
      "  0:00:00.115     1M / 33M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 489310\n",
      "  0:00:00.526  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39398 reads\n",
      "  0:00:00.526     1M / 1344M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39398 reads\n",
      "  0:00:00.529     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.542     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 92785 kmers in total.\n",
      "  0:00:00.544     1M / 1344M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.552     1M / 1344M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.554     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.560     1M / 1344M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.571     1M / 1344M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.579     1M / 1344M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 489305\n",
      "  0:00:00.692  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 92785 kmers\n",
      "  0:00:00.693  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 92785 kmers.\n",
      "  0:00:00.694     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.703     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 92719 kmers in total.\n",
      "  0:00:00.703     1M / 1344M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.716     1M / 1344M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 92719 kmers, 128656 bytes occupied (11.1007 bits per kmer).\n",
      "  0:00:00.718     1M / 1344M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.723     1M / 1344M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.730     1M / 1344M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.730     1M / 1344M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:00.730     1M / 1344M INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:00.731     1M / 1344M INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:00.736     1M / 1344M INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 819\n",
      "  0:00:00.736     1M / 1344M INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 829\n",
      "  0:00:00.737     1M / 1344M INFO   Early tip clipping       (early_simplification.hpp  :  49)   17348 22-mers were removed by early tip clipper\n",
      "  0:00:00.737     1M / 1344M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.737     1M / 1344M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.742     1M / 1344M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 652 sequences extracted\n",
      "  0:00:00.745     1M / 1344M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.747     1M / 1344M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 487)   Total 1304 edges to create\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 503)   Total 586 vertices to create\n",
      "  0:00:00.748     1M / 1344M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:00.749     1M / 1344M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:00.749     1M / 1344M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.760     2M / 1344M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 92785 kmers, 128784 bytes occupied (11.1039 bits per kmer).\n",
      "  0:00:00.765     2M / 1344M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.851     2M / 1344M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.855     2M / 1344M INFO    General                 (coverage_filling.hpp      :  82)   Processed 1304 edges\n",
      "  0:00:00.860     1M / 1344M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:00.862     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
      "  0:00:00.872     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 71\n",
      "  0:00:00.872     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 75. Coverage mad: 16.3086\n",
      "  0:00:00.872     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:00.903     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:00.961     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:01.140     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:01.446     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:00:01.496     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.682. Fitted coverage std. dev: 14.1437\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 1\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39831\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.682\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:00:01.498     1M / 1344M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:01.498     1M / 1344M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:01.502     1M / 1344M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:01.503     1M / 1344M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.504     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:01.504     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.504     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:01.504     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 12 times\n",
      "  0:00:01.504     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:01.506     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 119 times\n",
      "  0:00:01.506     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:01.506     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 71 times\n",
      "  0:00:01.506     1M / 1344M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:01.506     1M / 1344M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:01.506     1M / 1344M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.506     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.506     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.507     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 12 times\n",
      "  0:00:01.507     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.508     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 13 times\n",
      "  0:00:01.508     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.509     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 29 times\n",
      "  0:00:01.509     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.509     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.510     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:00:01.510     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.510     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.510     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.513     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.516     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.516     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.516     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:00:01.517     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.517     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.518     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.518     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.519     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.519     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.520     1M / 1344M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.520     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.522     1M / 1344M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.523     1M / 1344M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.523     1M / 1344M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.523     1M / 1344M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.523     1M / 1344M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.523     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.523     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.523     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.523     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.525     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.539     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.553     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.563     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.565     1M / 1344M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:01.569     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:01.570     1M / 1344M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:01.574     1M / 1344M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:01.582     1M / 1344M INFO    General                 (simplification.cpp        : 496)     Average coverage = 53.7876\n",
      "  0:00:01.590     1M / 1344M INFO    General                 (simplification.cpp        : 497)     Total length = 58398\n",
      "  0:00:01.598     1M / 1344M INFO    General                 (simplification.cpp        : 498)     Median edge length: 5366\n",
      "  0:00:01.606     1M / 1344M INFO    General                 (simplification.cpp        : 499)     Edges: 100\n",
      "  0:00:01.607     1M / 1344M INFO    General                 (simplification.cpp        : 500)     Vertices: 162\n",
      "  0:00:01.608     1M / 1344M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:01.609     1M / 1344M INFO    General                 (read_converter.cpp        : 107)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K21/simplified_contigs\n",
      "  0:00:01.613     3M / 1344M INFO    General                 (binary_converter.cpp      : 111)   50 reads written\n",
      "  0:00:01.617     1M / 1344M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:01.617     1M / 1344M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "===== K21 finished. \n",
      "\n",
      "\n",
      "===== K33 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K33/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K33/configs/config.info\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=33\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.001     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.001     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.001     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.002     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.002     1M / 27M   INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K21/simplified_contigs\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.003     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 244652\n",
      "  0:00:00.172  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39498 reads\n",
      "  0:00:00.179     1M / 1324M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39498 reads\n",
      "  0:00:00.184     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.196     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 92865 kmers in total.\n",
      "  0:00:00.196     1M / 1324M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.197     1M / 1324M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.197     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.197     1M / 1324M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.197     1M / 1324M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291646 Gb\n",
      "  0:00:00.197     1M / 1324M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 244650\n",
      "  0:00:00.294  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 92865 kmers\n",
      "  0:00:00.294  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 92865 kmers.\n",
      "  0:00:00.295     1M / 1330M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.310     1M / 1330M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 92884 kmers in total.\n",
      "  0:00:00.310     1M / 1330M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.323     1M / 1330M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 92884 kmers, 128832 bytes occupied (11.0962 bits per kmer).\n",
      "  0:00:00.324     1M / 1330M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.328     1M / 1330M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.344     1M / 1330M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.347     1M / 1330M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:00.350     1M / 1330M INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:00.357     1M / 1330M INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:00.367     1M / 1330M INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 841\n",
      "  0:00:00.381     1M / 1330M INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 848\n",
      "  0:00:00.388     1M / 1330M INFO   Early tip clipping       (early_simplification.hpp  :  49)   20866 34-mers were removed by early tip clipper\n",
      "  0:00:00.390     1M / 1330M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.392     1M / 1330M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.403     1M / 1330M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 400 sequences extracted\n",
      "  0:00:00.411     1M / 1330M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.425     1M / 1330M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.430     1M / 1330M INFO    General                 (debruijn_graph_constructor: 487)   Total 800 edges to create\n",
      "  0:00:00.433     1M / 1330M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.440     1M / 1330M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.445     1M / 1330M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.445     1M / 1330M INFO    General                 (debruijn_graph_constructor: 503)   Total 419 vertices to create\n",
      "  0:00:00.445     1M / 1330M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:00.446     1M / 1330M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:00.446     1M / 1330M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.459     2M / 1330M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 92865 kmers, 128864 bytes occupied (11.1012 bits per kmer).\n",
      "  0:00:00.460     2M / 1330M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.533     2M / 1330M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.539     1M / 1330M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:00.540     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 7\n",
      "  0:00:00.540     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 65\n",
      "  0:00:00.541     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 70. Coverage mad: 14.826\n",
      "  0:00:00.541     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:00.562     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:00.613     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:00.759     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:00.869     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 70.9848. Fitted coverage std. dev: 13.722\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 1\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 25\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 25\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39840\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 70.9848\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 25\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:00.873     1M / 1330M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:00.873     1M / 1330M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:00.873     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:00.873     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:00.873     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:00.874     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 11 times\n",
      "  0:00:00.874     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:00.875     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 63 times\n",
      "  0:00:00.875     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:00.875     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 75 times\n",
      "  0:00:00.875     1M / 1330M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:00.875     1M / 1330M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:00.875     1M / 1330M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:00.875     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:00.876     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.876     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 6 times\n",
      "  0:00:00.876     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.876     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:00:00.876     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.878     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:00:00.879     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:00:00.879     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:00.879     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:00:00.880     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:00.880     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:00.880     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:00.881     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:00.881     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:00.903     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:00.931     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:00.974     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.010     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.049     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.063     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.070     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.076     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.078     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.084     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.084     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.084     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.085     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.086     1M / 1330M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.086     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.086     1M / 1330M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.086     1M / 1330M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.086     1M / 1330M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.087     1M / 1330M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.087     1M / 1330M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.087     1M / 1330M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:01.087     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:01.088     1M / 1330M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:01.088     1M / 1330M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:01.091     1M / 1330M INFO    General                 (simplification.cpp        : 496)     Average coverage = 50.7738\n",
      "  0:00:01.095     1M / 1330M INFO    General                 (simplification.cpp        : 497)     Total length = 57195\n",
      "  0:00:01.096     1M / 1330M INFO    General                 (simplification.cpp        : 498)     Median edge length: 39824\n",
      "  0:00:01.096     1M / 1330M INFO    General                 (simplification.cpp        : 499)     Edges: 82\n",
      "  0:00:01.101     1M / 1330M INFO    General                 (simplification.cpp        : 500)     Vertices: 144\n",
      "  0:00:01.115     1M / 1330M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:01.117     1M / 1330M INFO    General                 (read_converter.cpp        : 107)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K33/simplified_contigs\n",
      "  0:00:01.118     3M / 1330M INFO    General                 (binary_converter.cpp      : 111)   41 reads written\n",
      "  0:00:01.126     1M / 1330M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:01.128     1M / 1330M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K55/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K55/configs/config.info\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=55\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.007     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.007     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.008     1M / 27M   INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.009     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.010     1M / 27M   INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.013     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.016     1M / 27M   INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K33/simplified_contigs\n",
      "  0:00:00.023     1M / 27M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.024     1M / 27M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.028     1M / 27M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.030     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.032     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.038     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.041     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 244652\n",
      "  0:00:00.331  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39480 reads\n",
      "  0:00:00.332     1M / 1336M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39480 reads\n",
      "  0:00:00.336     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.346     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 91674 kmers in total.\n",
      "  0:00:00.352     1M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.355     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.358     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.360     1M / 1336M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.372     1M / 1336M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291646 Gb\n",
      "  0:00:00.376     1M / 1336M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 244650\n",
      "  0:00:00.679  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 91674 kmers\n",
      "  0:00:00.679  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 91674 kmers.\n",
      "  0:00:00.683     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.698     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 91735 kmers in total.\n",
      "  0:00:00.698     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.718     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 91735 kmers, 127984 bytes occupied (11.1612 bits per kmer).\n",
      "  0:00:00.721     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.728     1M / 1336M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.749     1M / 1336M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.763     1M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.769     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.795     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 1963 sequences extracted\n",
      "  0:00:00.813     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.833     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.897     1M / 1336M INFO    General                 (debruijn_graph_constructor: 487)   Total 3926 edges to create\n",
      "  0:00:00.913     2M / 1336M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.949     2M / 1336M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.993     2M / 1336M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.995     2M / 1336M INFO    General                 (debruijn_graph_constructor: 503)   Total 2024 vertices to create\n",
      "  0:00:00.997     2M / 1336M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:01.008     2M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:01.013     2M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.036     2M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 91674 kmers, 127960 bytes occupied (11.1665 bits per kmer).\n",
      "  0:00:01.040     2M / 1336M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.113     2M / 1336M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.125     2M / 1336M INFO    General                 (coverage_filling.hpp      :  82)   Processed 3926 edges\n",
      "  0:00:01.127     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:01.127     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 12\n",
      "  0:00:01.128     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 58\n",
      "  0:00:01.128     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 59. Coverage mad: 13.3434\n",
      "  0:00:01.128     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:01.146     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:01.189     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:01.247     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 60.8928. Fitted coverage std. dev: 12.7266\n",
      "  0:00:01.250     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.990633\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 19\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 19\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39862\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 60.8928\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 19\n",
      "  0:00:01.251     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:01.251     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:00:01.257    10M / 1336M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.259    10M / 1336M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 1138 out of 3926, length: 65698\n",
      "  0:00:01.259    10M / 1336M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3948)\n",
      "  0:00:01.260    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.264    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 48078 kmers, 35488 bytes occupied (5.90507 bits per kmer).\n",
      "  0:00:01.264    11M / 1336M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.341    11M / 1336M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.350    10M / 1336M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.352    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.355    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 42 candidates\n",
      "  0:00:01.359    10M / 1336M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.360     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:01.360     1M / 1336M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:01.361     1M / 1336M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.361     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:01.361     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.361     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:01.369     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 871 times\n",
      "  0:00:01.370     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:01.379     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 31 times\n",
      "  0:00:01.385     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:01.388     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 75 times\n",
      "  0:00:01.393     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:01.395     1M / 1336M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:01.397     1M / 1336M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.401     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.402     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.404     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.413     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.430     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:00:01.433     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.440     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:00:01.441     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.441     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.445     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 2 times\n",
      "  0:00:01.448     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.448     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.448     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.449     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.449     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.450     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.450     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.454     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.455     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.458     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.459     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.463     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.464     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.466     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.472     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.476     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.478     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.484     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.487     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.494     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.495     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.497     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.497     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.499     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.499     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.499     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.499     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.499     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.499     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.500     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.500     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.500     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.500     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.500     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.500     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.501     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.503     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.503     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.503     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.506     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.510     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:00:01.518    10M / 1336M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.525    10M / 1336M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 61 out of 72, length: 30968\n",
      "  0:00:01.529    10M / 1336M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3948)\n",
      "  0:00:01.542    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.606    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 16023 kmers, 12328 bytes occupied (6.15515 bits per kmer).\n",
      "  0:00:01.606    10M / 1336M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.715    10M / 1336M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.720    10M / 1336M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.720    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.720    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
      "  0:00:01.720    10M / 1336M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.721     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.721     1M / 1336M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.721     1M / 1336M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.721     1M / 1336M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.722     1M / 1336M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.722     1M / 1336M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:01.722     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 496)     Average coverage = 44.4193\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 497)     Total length = 55900\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 498)     Median edge length: 39846\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 499)     Edges: 72\n",
      "  0:00:01.723     1M / 1336M INFO    General                 (simplification.cpp        : 500)     Vertices: 130\n",
      "  0:00:01.723     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:01.725     1M / 1336M INFO    General                 (read_converter.cpp        : 107)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K55/simplified_contigs\n",
      "  0:00:01.725     4M / 1336M INFO    General                 (binary_converter.cpp      : 111)   36 reads written\n",
      "  0:00:01.741     1M / 1336M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:01.741     1M / 1336M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== K77 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K77/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K77/configs/config.info\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=77\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.001     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.001     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.001     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.001     1M / 27M   INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.006     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.006     1M / 27M   INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K55/simplified_contigs\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 163101\n",
      "  0:00:00.324  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39470 reads\n",
      "  0:00:00.326     1M / 1336M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39470 reads\n",
      "  0:00:00.329     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.348     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 89882 kmers in total.\n",
      "  0:00:00.348     1M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.349     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.349     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.349     1M / 1336M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.349     1M / 1336M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291646 Gb\n",
      "  0:00:00.350     1M / 1336M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 163100\n",
      "  0:00:00.443  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 89882 kmers\n",
      "  0:00:00.443  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 89882 kmers.\n",
      "  0:00:00.443     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.451     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 89977 kmers in total.\n",
      "  0:00:00.452     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.462     1M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 89977 kmers, 126720 bytes occupied (11.2669 bits per kmer).\n",
      "  0:00:00.462     1M / 1336M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.467     1M / 1336M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.474     1M / 1336M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.475     1M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.475     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.482     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 1887 sequences extracted\n",
      "  0:00:00.495     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.510     1M / 1336M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.510     1M / 1336M INFO    General                 (debruijn_graph_constructor: 487)   Total 3774 edges to create\n",
      "  0:00:00.510     2M / 1336M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.511     2M / 1336M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.511     2M / 1336M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.512     2M / 1336M INFO    General                 (debruijn_graph_constructor: 503)   Total 1982 vertices to create\n",
      "  0:00:00.512     2M / 1336M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:00.513     2M / 1336M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:00.513     2M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.525     2M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 89882 kmers, 126672 bytes occupied (11.2745 bits per kmer).\n",
      "  0:00:00.529     2M / 1336M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.598     2M / 1336M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.611     2M / 1336M INFO    General                 (coverage_filling.hpp      :  82)   Processed 3774 edges\n",
      "  0:00:00.614     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:00.616     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
      "  0:00:00.616     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 45\n",
      "  0:00:00.619     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 50. Coverage mad: 11.8608\n",
      "  0:00:00.619     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:00.634     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:00.668     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:00.772     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:00.821     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 51.5031. Fitted coverage std. dev: 11.5407\n",
      "  0:00:00.826     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.998173\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 15\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 15\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39865\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 51.5031\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 15\n",
      "  0:00:00.827     1M / 1336M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:00.827     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:00:00.830    10M / 1336M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.831    10M / 1336M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 1145 out of 3774, length: 65216\n",
      "  0:00:00.831    10M / 1336M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3795)\n",
      "  0:00:00.834    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.893    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 47234 kmers, 34880 bytes occupied (5.90761 bits per kmer).\n",
      "  0:00:00.893    11M / 1336M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:00.963    11M / 1336M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:00.969    10M / 1336M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:00.970    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:00.979    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 40 candidates\n",
      "  0:00:00.983    10M / 1336M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:00.992     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:00.995     1M / 1336M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:00.998     1M / 1336M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.003     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:01.013     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.015     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:01.025     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 858 times\n",
      "  0:00:01.028     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:01.041     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 18 times\n",
      "  0:00:01.044     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:01.052     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 87 times\n",
      "  0:00:01.057     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:01.062     1M / 1336M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:01.066     1M / 1336M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.067     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.071     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.075     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 4 times\n",
      "  0:00:01.080     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.086     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:00:01.087     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.093     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.097     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.101     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.107     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.110     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.113     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.116     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.125     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.128     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.133     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.136     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.140     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.143     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.146     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.148     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.157     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.157     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.157     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.157     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.157     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.158     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.158     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.158     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.160     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.174     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.179     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.194     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.202     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.204     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.213     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.219     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.221     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.224     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.231     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.235     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.239     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.240     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.249     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.253     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.257     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.259     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.266     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.275     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.276     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.277     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.282     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.284     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.288     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.291     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.292     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.298     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.300     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.307     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.309     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.317     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.325     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.328     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.331     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.334     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.337     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.340     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.344     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.349     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.359     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.362     1M / 1336M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.364     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.376     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.385     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.388     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.392     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.396     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.402     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:00:01.407    10M / 1336M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.412    10M / 1336M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 57 out of 64, length: 28439\n",
      "  0:00:01.418    10M / 1336M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3795)\n",
      "  0:00:01.430    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.530    10M / 1336M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 14741 kmers, 11408 bytes occupied (6.19117 bits per kmer).\n",
      "  0:00:01.531    10M / 1336M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.623    10M / 1336M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.633    10M / 1336M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.634    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.635    10M / 1336M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
      "  0:00:01.636    10M / 1336M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.637     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.638     1M / 1336M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.639     1M / 1336M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.641     1M / 1336M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.641     1M / 1336M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.642     1M / 1336M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:01.642     1M / 1336M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:01.642     1M / 1336M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (simplification.cpp        : 496)     Average coverage = 38.2022\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (simplification.cpp        : 497)     Total length = 54816\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (simplification.cpp        : 498)     Median edge length: 39864\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (simplification.cpp        : 499)     Edges: 64\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (simplification.cpp        : 500)     Vertices: 120\n",
      "  0:00:01.643     1M / 1336M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:01.643     1M / 1336M INFO    General                 (read_converter.cpp        : 107)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K77/simplified_contigs\n",
      "  0:00:01.644     4M / 1336M INFO    General                 (binary_converter.cpp      : 111)   32 reads written\n",
      "  0:00:01.652     1M / 1336M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:01.653     1M / 1336M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "===== K77 finished. \n",
      "\n",
      "\n",
      "===== K99 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K99/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K99/configs/config.info\n",
      "  0:00:00.004     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.016     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=99\n",
      "  0:00:00.019     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.020     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.024     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.028     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.032     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.033     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.034     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.038     1M / 27M   INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.041     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.043     1M / 27M   INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.045     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.050     1M / 27M   INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K77/simplified_contigs\n",
      "  0:00:00.056     1M / 27M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.062     1M / 27M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.069     1M / 27M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.070     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.072     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.079     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.088     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 122326\n",
      "  0:00:00.247  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39462 reads\n",
      "  0:00:00.248     1M / 1328M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39462 reads\n",
      "  0:00:00.250     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.256     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 87598 kmers in total.\n",
      "  0:00:00.257     1M / 1328M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.257     1M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.257     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.257     1M / 1328M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.257     1M / 1328M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291646 Gb\n",
      "  0:00:00.257     1M / 1328M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 122325\n",
      "  0:00:00.343  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 87598 kmers\n",
      "  0:00:00.343  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 87598 kmers.\n",
      "  0:00:00.344     1M / 1332M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.366     1M / 1332M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 87704 kmers in total.\n",
      "  0:00:00.368     1M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.380     1M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 87704 kmers, 125056 bytes occupied (11.4071 bits per kmer).\n",
      "  0:00:00.394     1M / 1332M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.405     1M / 1332M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.414     1M / 1332M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.420     1M / 1332M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.421     1M / 1332M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.436     1M / 1332M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 1828 sequences extracted\n",
      "  0:00:00.446     1M / 1332M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.451     1M / 1332M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.453     1M / 1332M INFO    General                 (debruijn_graph_constructor: 487)   Total 3656 edges to create\n",
      "  0:00:00.454     2M / 1332M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.459     2M / 1332M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.476     2M / 1332M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.480     2M / 1332M INFO    General                 (debruijn_graph_constructor: 503)   Total 1934 vertices to create\n",
      "  0:00:00.487     2M / 1332M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:00.496     2M / 1332M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:00.497     2M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.519     2M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 87598 kmers, 124936 bytes occupied (11.4099 bits per kmer).\n",
      "  0:00:00.519     2M / 1332M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.571     2M / 1332M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.582     2M / 1332M INFO    General                 (coverage_filling.hpp      :  82)   Processed 3656 edges\n",
      "  0:00:00.588     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:00.589     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 7\n",
      "  0:00:00.590     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 40\n",
      "  0:00:00.597     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 42. Coverage mad: 11.8608\n",
      "  0:00:00.599     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:00.620     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:00.652     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:00.739     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:00.795     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 42.9484. Fitted coverage std. dev: 10.3758\n",
      "  0:00:00.809     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.983811\n",
      "  0:00:00.850     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 12\n",
      "  0:00:00.888     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 12\n",
      "  0:00:00.913     1M / 1332M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39865\n",
      "  0:00:00.915     1M / 1332M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 42.9484\n",
      "  0:00:00.916     1M / 1332M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 12\n",
      "  0:00:00.926     1M / 1332M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:00.927     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:00:00.934    10M / 1332M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.940    10M / 1332M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 1132 out of 3656, length: 64038\n",
      "  0:00:00.941    10M / 1332M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3677)\n",
      "  0:00:00.948    10M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.026    10M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 45849 kmers, 33856 bytes occupied (5.90739 bits per kmer).\n",
      "  0:00:01.031    11M / 1332M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.104    11M / 1332M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.111    10M / 1332M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.114    10M / 1332M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.116    10M / 1332M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 40 candidates\n",
      "  0:00:01.121    10M / 1332M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.125     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:01.126     1M / 1332M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:01.127     1M / 1332M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.131     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:01.135     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.139     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:01.149     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 847 times\n",
      "  0:00:01.151     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:01.160     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 8 times\n",
      "  0:00:01.162     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:01.166     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 87 times\n",
      "  0:00:01.169     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:01.171     1M / 1332M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:01.174     1M / 1332M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.178     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.187     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.194     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.199     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.204     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:00:01.207     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.214     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.220     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.223     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.226     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.229     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.232     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.235     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.236     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.240     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.241     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.245     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.249     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.253     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.256     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.262     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.270     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.271     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.275     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.281     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.286     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.292     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.309     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.323     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.340     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.346     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.353     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.357     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.357     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.357     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.357     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.357     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.358     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.359     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.359     1M / 1332M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.360     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.360     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:00:01.364    10M / 1332M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.372    10M / 1332M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 57 out of 64, length: 27207\n",
      "  0:00:01.380    10M / 1332M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3677)\n",
      "  0:00:01.390    10M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.411    10M / 1332M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 14125 kmers, 10968 bytes occupied (6.21196 bits per kmer).\n",
      "  0:00:01.411    10M / 1332M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.501    10M / 1332M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.507    10M / 1332M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.508    10M / 1332M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.511    10M / 1332M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
      "  0:00:01.514    10M / 1332M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.522     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.523     1M / 1332M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.527     1M / 1332M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.530     1M / 1332M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.532     1M / 1332M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.533     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.542     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.546     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.554     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.560     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.564     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.572     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.582     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.585     1M / 1332M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:01.591     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:01.598     1M / 1332M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:01.601     1M / 1332M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:01.606     1M / 1332M INFO    General                 (simplification.cpp        : 496)     Average coverage = 32.146\n",
      "  0:00:01.611     1M / 1332M INFO    General                 (simplification.cpp        : 497)     Total length = 54215\n",
      "  0:00:01.614     1M / 1332M INFO    General                 (simplification.cpp        : 498)     Median edge length: 39864\n",
      "  0:00:01.616     1M / 1332M INFO    General                 (simplification.cpp        : 499)     Edges: 64\n",
      "  0:00:01.619     1M / 1332M INFO    General                 (simplification.cpp        : 500)     Vertices: 120\n",
      "  0:00:01.622     1M / 1332M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:01.624     1M / 1332M INFO    General                 (read_converter.cpp        : 107)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K99/simplified_contigs\n",
      "  0:00:01.625     4M / 1332M INFO    General                 (binary_converter.cpp      : 111)   32 reads written\n",
      "  0:00:01.633     1M / 1332M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:01.635     1M / 1332M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "===== K99 finished. \n",
      "\n",
      "\n",
      "===== K127 started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/spades-core /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 27M   INFO    General                 (main.cpp                  :  99)   Loaded config from /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/configs/config.info\n",
      "  0:00:00.007     1M / 27M   INFO    General                 (memory_limit.cpp          :  54)   Memory limit set to 7 Gb\n",
      "  0:00:00.010     1M / 27M   INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (main.cpp                  : 109)   Assembling dataset (/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/dataset.info) with K=127\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 8\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.011     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.011     1M / 27M   INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:00.011     1M / 27M   INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.011     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.012     1M / 27M   INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.012     1M / 27M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.012     1M / 27M   INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K99/simplified_contigs\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (construction.cpp          : 153)   Max read length 301\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (construction.cpp          : 159)   Average read length 182.765\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291649 Gb\n",
      "  0:00:00.013     1M / 27M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 122326\n",
      "  0:00:00.276  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 125)   Processed 39462 reads\n",
      "  0:00:00.280     1M / 1324M INFO    General                 (kmer_splitters.hpp        : 131)   Used 39462 reads\n",
      "  0:00:00.286     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.304     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 84668 kmers in total.\n",
      "  0:00:00.307     1M / 1324M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:00.311     1M / 1324M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:00.312     1M / 1324M INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 80 files using 8 threads. This might take a while.\n",
      "  0:00:00.317     1M / 1324M INFO    General                 (file_limit.hpp            :  42)   Open file limit set to 4096\n",
      "  0:00:00.326     1M / 1324M INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 0.291646 Gb\n",
      "  0:00:00.328     1M / 1324M INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 122325\n",
      "  0:00:00.595  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 194)   Processed 84668 kmers\n",
      "  0:00:00.599  2721M / 2721M INFO    General                 (kmer_splitters.hpp        : 199)   Used 84668 kmers.\n",
      "  0:00:00.603     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:00.624     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 84772 kmers in total.\n",
      "  0:00:00.628     1M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.658     1M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 84772 kmers, 122872 bytes occupied (11.5955 bits per kmer).\n",
      "  0:00:00.661     1M / 1328M INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:00.672     1M / 1328M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.694     1M / 1328M INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.711     1M / 1328M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:00.806     1M / 1328M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:00.868     1M / 1328M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 1771 sequences extracted\n",
      "  0:00:00.883     1M / 1328M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:00.892     1M / 1328M INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.894     1M / 1328M INFO    General                 (debruijn_graph_constructor: 487)   Total 3542 edges to create\n",
      "  0:00:00.895     2M / 1328M INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:00.906     2M / 1328M INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:00.917     2M / 1328M INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:00.917     2M / 1328M INFO    General                 (debruijn_graph_constructor: 503)   Total 1875 vertices to create\n",
      "  0:00:00.918     2M / 1328M INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:00.919     2M / 1328M INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:00.923     2M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:00.950     2M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 84668 kmers, 122784 bytes occupied (11.6015 bits per kmer).\n",
      "  0:00:00.954     2M / 1328M INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.002     2M / 1328M INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.010     2M / 1328M INFO    General                 (coverage_filling.hpp      :  82)   Processed 3542 edges\n",
      "  0:00:01.019     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:00:01.020     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:00:01.028     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 31\n",
      "  0:00:01.030     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 32. Coverage mad: 10.3782\n",
      "  0:00:01.038     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:00:01.054     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:00:01.082     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:00:01.142     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:00:01.188     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 33.3532. Fitted coverage std. dev: 9.01542\n",
      "  0:00:01.188     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.774351\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 10\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 10\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 39865\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 33.3532\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 10\n",
      "  0:00:01.189     1M / 1328M INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:00:01.189     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:00:01.197    10M / 1328M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.205    10M / 1328M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 1106 out of 3542, length: 61422\n",
      "  0:00:01.212    10M / 1328M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3562)\n",
      "  0:00:01.220    10M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.291    10M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 43311 kmers, 32024 bytes occupied (5.91517 bits per kmer).\n",
      "  0:00:01.293    11M / 1328M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.376    11M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.408    10M / 1328M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.440    10M / 1328M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.482    10M / 1328M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 34 candidates\n",
      "  0:00:01.515    10M / 1328M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.519     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:00:01.520     1M / 1328M INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:00:01.523     1M / 1328M INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.526     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:00:01.532     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.534     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:00:01.552     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 828 times\n",
      "  0:00:01.555     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:00:01.565     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4 times\n",
      "  0:00:01.566     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:00:01.571     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 88 times\n",
      "  0:00:01.575     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:00:01.576     1M / 1328M INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:00:01.580     1M / 1328M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.582     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.587     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.593     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.598     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.607     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:00:01.611     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.621     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.626     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.630     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.634     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.635     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.642     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.646     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.650     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.652     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.656     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.659     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.660     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.664     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.671     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.671     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.675     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.679     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.687     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.687     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.691     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.695     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.700     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.704     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.707     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.709     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.714     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.721     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.723     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.726     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.731     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.733     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.740     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.741     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.745     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.749     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.751     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.751     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.751     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.752     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.753     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.753     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.753     1M / 1328M INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.753     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.755     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.759     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.770     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.771     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:00:01.777     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.782     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:00:01.790    10M / 1328M INFO    General                 (edge_index.hpp            : 113)   Size of edge index entries: 12/8\n",
      "  0:00:01.796    10M / 1328M INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 42 out of 46, length: 22530\n",
      "  0:00:01.802    10M / 1328M INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 3562)\n",
      "  0:00:01.804    10M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:01.823    10M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 11265 kmers, 8904 bytes occupied (6.3233 bits per kmer).\n",
      "  0:00:01.823    10M / 1328M INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.917    10M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:01.925    10M / 1328M INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:00:01.926    10M / 1328M INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:00:01.928    10M / 1328M INFO   GapCloser                (gap_closer.cpp            : 391)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
      "  0:00:01.938    10M / 1328M INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:00:01.941     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:00:01.942     1M / 1328M INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:00:01.948     1M / 1328M INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.949     1M / 1328M INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:00:01.954     1M / 1328M INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.957     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.968     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.973     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:01.979     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:01.981     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:00:01.989     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:00:01.991     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:00:02.001     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:00:02.005     1M / 1328M INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:00:02.007     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:00:02.011     1M / 1328M INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 0 times\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 495)   After simplification:\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 496)     Average coverage = 26.3148\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 497)     Total length = 51130\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 498)     Median edge length: 39864\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 499)     Edges: 46\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (simplification.cpp        : 500)     Vertices: 88\n",
      "  0:00:02.011     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:00:02.011     1M / 1328M INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 3562)\n",
      "  0:00:02.014     1M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:02.046     1M / 1328M INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 51130 kmers, 37680 bytes occupied (5.89556 bits per kmer).\n",
      "  0:00:02.046     1M / 1328M INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.061     1M / 1328M INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 230 kmers to process\n",
      "  0:00:02.063     1M / 1328M INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:00:02.063     1M / 1328M INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:00:02.064     1M / 1328M INFO    General                 (mismatch_correction.cpp   : 192)   Total 2 edges (out of 46) with 50 potential mismatch positions (25 positions per edge)\n",
      "  0:00:02.064     1M / 1328M INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:00:02.120     2M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 39398 reads processed\n",
      "  0:00:02.123     1M / 1328M INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:00:02.124     1M / 1328M INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 0 nucleotides\n",
      "  0:00:02.128     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:02.130     1M / 1328M INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/before_rr.fasta\n",
      "  0:00:02.134     1M / 1328M INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph_after_simplification.gfa\n",
      "  0:00:02.137     1M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:00:02.142     1M / 1328M INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 230 kmers to process\n",
      "  0:00:02.146     1M / 1328M INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:00:02.153     1M / 1328M INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 39864\n",
      "  0:00:02.159     1M / 1328M INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:00:02.161     1M / 1328M INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  0:00:02.188   146M / 1328M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:00:02.375   146M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:02.476   146M / 1328M INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 37\n",
      "  0:00:02.483   146M / 1328M INFO    General                 (pair_info_count.cpp       : 222)   6360 paired reads (64.6473% of all) aligned to long edges\n",
      "  0:00:02.484     2M / 1328M INFO    General                 (pair_info_count.cpp       : 369)     Insert size = 273.676, deviation = 136.946, left quantile = 146, right quantile = 463, read length = 301\n",
      "  0:00:02.491     2M / 1328M WARN    General                 (pair_info_count.cpp       : 376)   Estimated mean insert size 273.676 is very small compared to read length 301\n",
      "  0:00:02.493     2M / 1328M INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:00:02.498     2M / 1328M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:00:02.517     2M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:02.522     2M / 1328M INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:00:02.525     2M / 1328M INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:00:02.527     2M / 1328M INFO    General                 (pair_info_count.cpp       : 299)   Left insert size quantile 146, right insert size quantile 463, filtering threshold 2, rounding threshold 0\n",
      "  0:00:02.532    11M / 1328M INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:00:02.552    11M / 1328M INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9838 reads processed\n",
      "  0:00:02.559     2M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:00:02.560     2M / 1328M INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:00:02.561     2M / 1328M INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:00:02.564     2M / 1328M INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:02.570     2M / 1328M INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:00:02.574     2M / 1328M INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 48\n",
      "  0:00:02.581     2M / 1328M INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 14\n",
      "  0:00:02.595     2M / 1328M INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:02.602     2M / 1328M INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:02.604     2M / 1328M INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:02.614     2M / 1328M INFO   PairInfoImprover         (pair_info_improver.hpp    : 104)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:00:02.624     2M / 1328M INFO   PairInfoImprover         (pair_info_improver.hpp    : 104)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:00:02.629     2M / 1328M INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:00:02.637     2M / 1328M INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:02.655     2M / 1328M INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:00:02.658     2M / 1328M INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 46\n",
      "  0:00:02.661     2M / 1328M INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 46\n",
      "  0:00:02.663     2M / 1328M INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:00:02.665     2M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:00:02.667     2M / 1328M INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:00:02.673     2M / 1328M INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:00:02.674     2M / 1328M INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 33.3468 calculated of length 41140\n",
      "  0:00:02.677     2M / 1328M INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:02.689     2M / 1328M INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 26.3148\n",
      "  0:00:02.692     2M / 1328M INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 14\n",
      "  0:00:02.694     2M / 1328M INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 26.3148\n",
      "  0:00:02.702     2M / 1328M INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 14\n",
      "  0:00:02.708     2M / 1328M INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:00:02.710     2M / 1328M INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:00:02.712     2M / 1328M INFO    General                 (extenders_logic.cpp       : 545)   Using 0 single read libraries\n",
      "  0:00:02.715     2M / 1328M INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 3\n",
      "  0:00:02.721     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 22 (0%)\n",
      "  0:00:02.727     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 3 paths from 22 (13%)\n",
      "  0:00:02.731     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 6 paths from 22 (27%)\n",
      "  0:00:02.741     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 9 paths from 22 (40%)\n",
      "  0:00:02.743     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 12 paths from 22 (54%)\n",
      "  0:00:02.754     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 15 paths from 22 (68%)\n",
      "  0:00:02.759     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 18 paths from 22 (81%)\n",
      "  0:00:02.761     2M / 1328M INFO    General                 (path_extenders.cpp        :  36)   Processed 21 paths from 22 (95%)\n",
      "  0:00:02.766     2M / 1328M INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:00:02.766     2M / 1328M INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:00:02.766     2M / 1328M INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:00:02.766     2M / 1328M INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:00:02.766     2M / 1328M INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:00:02.766     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:00:02.767     2M / 1328M INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:00:02.767     2M / 1328M INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:00:02.767     2M / 1328M INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:00:02.767     2M / 1328M INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:00:02.768     2M / 1328M INFO    General                 (launcher.cpp              : 312)   Traversed 0 loops\n",
      "  0:00:02.768     2M / 1328M INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:00:02.769     2M / 1328M INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:00:02.769     2M / 1328M INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:00:02.769     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:00:02.769     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:00:02.772     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:00:02.777     2M / 1328M INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:00:02.780     2M / 1328M INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:00:02.782     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:00:02.782     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:00:02.782     2M / 1328M INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:00:02.782     2M / 1328M INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:00:02.782     2M / 1328M INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:02.782     2M / 1328M INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:00:02.782     2M / 1328M INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/before_rr.fasta\n",
      "  0:00:02.783     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.783     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph.fastg\n",
      "  0:00:02.785     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:00:02.787     2M / 1328M INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/final_contigs.fasta\n",
      "  0:00:02.791     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/final_contigs.paths\n",
      "  0:00:02.795     2M / 1328M INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/scaffolds.fasta\n",
      "  0:00:02.796     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/scaffolds.paths\n",
      "  0:00:02.798     2M / 1328M INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:00:02.799     2M / 1328M INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:00:02.799     1M / 1328M INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 0 minutes 2 seconds\n",
      "\n",
      "===== K127 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/python /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades/spades_pipeline/scripts/copy_files.py /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/before_rr.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/before_rr.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph_after_simplification.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/assembly_graph_after_simplification.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/final_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/first_pe_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/first_pe_contigs.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/strain_graph.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/strain_graph.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/scaffolds.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/scaffolds.fasta /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/scaffolds.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/scaffolds.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph_with_scaffolds.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/assembly_graph_with_scaffolds.gfa /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/assembly_graph.fastg /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/assembly_graph.fastg /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/K127/final_contigs.paths /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/python /home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/scaffolds.fasta --misc_dir /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/corrected/\n",
      " * Assembled contigs are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/contigs.fasta\n",
      " * Assembled scaffolds are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/scaffolds.paths\n",
      " * Assembly graph is in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished WITH WARNINGS!\n",
      "\n",
      "=== Error correction and assembling warnings:\n",
      " * 0:00:02.491     2M / 1328M WARN    General                 (pair_info_count.cpp       : 376)   Estimated mean insert size 273.676 is very small compared to read length 301\n",
      "======= Warnings saved to /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/warnings.log\n",
      "\n",
      "SPAdes log can be found here: /home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/newspades_results/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "!reformat.sh in1=SRR18218015_1.fastq.gz in2=SRR18218015_2.fastq.gz reads=10000 out1=SRR18218015_10K1.fastq.gz out2=SRR18218015_10K2.fastq.gz\n",
    "!spades.py -1 SRR18218015_10K1.fastq.gz -2 SRR18218015_10K2.fastq.gz -o newspades_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "cafe04b3-7088-42e4-9546-367d112ebc1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2'"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "b3268019-4386-4dc7-9602-8ac7e96c8acb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mghalib/miniconda3/Excercises/phageinformatics_UCPH_June_2024/Rainforestdata_2/fastqc/trial4files\n"
     ]
    }
   ],
   "source": [
    "cd fastqc/trial4files/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "a69127ab-1f8b-47fa-920a-75cd5fc633ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "!tar -xzf annpharokka_v1.4.0_databases.tar.gz\n",
    "#after downloading the database from a server manually"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "b5ea82ea-76ea-4874-970f-02e3c334af72",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m2024-07-19 18:37:00.456\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36minstantiate_dirs\u001b[0m:\u001b[36m183\u001b[0m - \u001b[1mRemoving output directory finalcontig_pharokka as -f or --force was specified.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.477\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m95\u001b[0m - \u001b[1mStarting Pharokka v1.7.2\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.477\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m96\u001b[0m - \u001b[1mCommand executed: Namespace(infile='newspades_results/contigs.fasta', outdir='finalcontig_pharokka', database='pharokka_v1.4.0_databases', threads='7', force=True, prefix='Default', locustag='Default', gene_predictor='prodigal', meta=False, split=False, coding_table='11', evalue='1E-05', fast=False, mmseqs2_only=False, meta_hmm=False, dnaapler=True, custom_hmm='', genbank=False, terminase=False, terminase_strand='nothing', terminase_start='nothing', skip_extra_annotations=False, skip_mash=False, minced_args='', mash_distance=0.2, citation=False)\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.478\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m97\u001b[0m - \u001b[1mRepository homepage is https://github.com/gbouras13/pharokka\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.478\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m98\u001b[0m - \u001b[1mWritten by George Bouras: george.bouras@adelaide.edu.au\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.478\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m100\u001b[0m - \u001b[1mChecking database installation in pharokka_v1.4.0_databases.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.478\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m103\u001b[0m - \u001b[1mAll databases have been successfully checked.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.479\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m119\u001b[0m - \u001b[1mChecking dependencies.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.983\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m383\u001b[0m - \u001b[1mPhanotate version found is v1.5.1\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:00.983\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m392\u001b[0m - \u001b[1mPhanotate version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.006\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m414\u001b[0m - \u001b[1mMMseqs2 version found is v13.45111\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.007\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m423\u001b[0m - \u001b[1mMMseqs2 version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.166\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m447\u001b[0m - \u001b[1mtRNAscan-SE version found is v2.0.12\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.167\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m458\u001b[0m - \u001b[1mtRNAscan-SE version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.446\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m482\u001b[0m - \u001b[1mMinCED version found is v0.4.2\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.446\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m493\u001b[0m - \u001b[1mMinCED version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.451\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m519\u001b[0m - \u001b[1mARAGORN version found is v1.2.41\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.451\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m530\u001b[0m - \u001b[1mARAGORN version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.493\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m547\u001b[0m - \u001b[1mmash version found is v2.3\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:01.493\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m554\u001b[0m - \u001b[1mmash version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.502\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m573\u001b[0m - \u001b[1mDnaapler version found is v0.7.0\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.502\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m580\u001b[0m - \u001b[1mDnaapler version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.503\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m592\u001b[0m - \u001b[1mPyrodigal version is v3.4.1\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.503\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m593\u001b[0m - \u001b[1mPyrodigal version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.503\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m605\u001b[0m - \u001b[1mPyrodigal_gv version is v0.3.1\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.503\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m606\u001b[0m - \u001b[1mPyrodigal_gv version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.526\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m230\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.528\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m232\u001b[0m - \u001b[1mInput newspades_results/contigs.fasta is in FASTA format.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.528\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m237\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta for duplicate FASTA headers.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.529\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m239\u001b[0m - \u001b[1mAll headers in newspades_results/contigs.fasta are unique.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.529\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_gene_predictor\u001b[0m:\u001b[36m270\u001b[0m - \u001b[1mProdigal implemented with pyrodigal will be used for gene prediction.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.530\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m154\u001b[0m - \u001b[1mYou have chosen to reorient your contigs by specifying --dnaapler. Checking the input.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.530\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_dnaapler\u001b[0m:\u001b[36m992\u001b[0m - \u001b[1mRunning Dnaapler to rerorient all contigs to begin with the terminase large subunit.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:02.532\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running dnaapler all -o finalcontig_pharokka/dnaapler -t 7 -i newspades_results/contigs.fasta ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:07.376\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running dnaapler all -o finalcontig_pharokka/dnaapler -t 7 -i newspades_results/contigs.fasta\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:07.377\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_meta\u001b[0m:\u001b[36m305\u001b[0m - \u001b[1mMore than one contig detected in the input file. Re-running pharokka with -m meta mode is recommended unless this is a fragmented isolate genome. Continuing.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:07.377\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m289\u001b[0m - \u001b[1mImplementing Prodigal using Pyrodigal.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:07.631\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m309\u001b[0m - \u001b[1mStarting tRNA-scanSE.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:07.632\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running tRNAscan-SE --thread 7 -G -Q -j finalcontig_pharokka/trnascan_out.gff finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:08.969\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running tRNAscan-SE --thread 7 -G -Q -j finalcontig_pharokka/trnascan_out.gff finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:08.970\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_minced\u001b[0m:\u001b[36m822\u001b[0m - \u001b[1mRunning MinCED.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:08.971\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running minced finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta finalcontig_pharokka/pharokka_minced_spacers.txt finalcontig_pharokka/pharokka_minced.gff ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.185\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running minced finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta finalcontig_pharokka/pharokka_minced_spacers.txt finalcontig_pharokka/pharokka_minced.gff\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.186\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_aragorn\u001b[0m:\u001b[36m854\u001b[0m - \u001b[1mRunning Aragorn.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.188\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running aragorn -l -gcbact -w -m -o finalcontig_pharokka/pharokka_aragorn.txt finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.223\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running aragorn -l -gcbact -w -m -o finalcontig_pharokka/pharokka_aragorn.txt finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.224\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m317\u001b[0m - \u001b[1mStarting MMseqs2.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.225\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_mmseqs\u001b[0m:\u001b[36m675\u001b[0m - \u001b[1mRunning MMseqs2 on PHROG Database.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.228\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/target_dir/target_seqs ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.257\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/target_dir/target_seqs\u001b[0m\n",
      "\u001b[32m2024-07-19 18:37:09.259\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs search -e 1E-05 pharokka_v1.4.0_databases/phrogs_profile_db finalcontig_pharokka/target_dir/target_seqs finalcontig_pharokka/mmseqs/results_mmseqs finalcontig_pharokka/tmp_dir/ -s 8.5 --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:11.999\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs search -e 1E-05 pharokka_v1.4.0_databases/phrogs_profile_db finalcontig_pharokka/target_dir/target_seqs finalcontig_pharokka/mmseqs/results_mmseqs finalcontig_pharokka/tmp_dir/ -s 8.5 --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.000\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createtsv pharokka_v1.4.0_databases/phrogs_profile_db finalcontig_pharokka/target_dir/target_seqs finalcontig_pharokka/mmseqs/results_mmseqs finalcontig_pharokka/mmseqs_results.tsv --full-header --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.048\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createtsv pharokka_v1.4.0_databases/phrogs_profile_db finalcontig_pharokka/target_dir/target_seqs finalcontig_pharokka/mmseqs/results_mmseqs finalcontig_pharokka/mmseqs_results.tsv --full-header --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.049\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_mmseqs\u001b[0m:\u001b[36m675\u001b[0m - \u001b[1mRunning MMseqs2 on CARD Database.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.051\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/CARD_target_dir/target_seqs ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.063\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/CARD_target_dir/target_seqs\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:12.064\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs search --min-seq-id 0.8 -c 0.4 pharokka_v1.4.0_databases/CARD finalcontig_pharokka/CARD_target_dir/target_seqs finalcontig_pharokka/CARD/results_mmseqs finalcontig_pharokka/CARD_dir/ -s 8.5 --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.838\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs search --min-seq-id 0.8 -c 0.4 pharokka_v1.4.0_databases/CARD finalcontig_pharokka/CARD_target_dir/target_seqs finalcontig_pharokka/CARD/results_mmseqs finalcontig_pharokka/CARD_dir/ -s 8.5 --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.840\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createtsv pharokka_v1.4.0_databases/CARD finalcontig_pharokka/CARD_target_dir/target_seqs finalcontig_pharokka/CARD/results_mmseqs finalcontig_pharokka/CARD_results.tsv --full-header --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.861\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createtsv pharokka_v1.4.0_databases/CARD finalcontig_pharokka/CARD_target_dir/target_seqs finalcontig_pharokka/CARD/results_mmseqs finalcontig_pharokka/CARD_results.tsv --full-header --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.863\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mprocesses\u001b[0m:\u001b[36mrun_mmseqs\u001b[0m:\u001b[36m675\u001b[0m - \u001b[1mRunning MMseqs2 on VFDB Database.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.864\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/VFDB_target_dir/target_seqs ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.877\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createdb finalcontig_pharokka/prodigal_aas_tmp.fasta finalcontig_pharokka/VFDB_target_dir/target_seqs\u001b[0m\n",
      "\u001b[32m2024-07-19 18:42:53.878\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs search --min-seq-id 0.8 -c 0.4 pharokka_v1.4.0_databases/vfdb finalcontig_pharokka/VFDB_target_dir/target_seqs finalcontig_pharokka/VFDB/results_mmseqs finalcontig_pharokka/VFDB_dir/ -s 8.5 --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:10.967\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs search --min-seq-id 0.8 -c 0.4 pharokka_v1.4.0_databases/vfdb finalcontig_pharokka/VFDB_target_dir/target_seqs finalcontig_pharokka/VFDB/results_mmseqs finalcontig_pharokka/VFDB_dir/ -s 8.5 --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:10.969\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mmseqs createtsv pharokka_v1.4.0_databases/vfdb finalcontig_pharokka/VFDB_target_dir/target_seqs finalcontig_pharokka/VFDB/results_mmseqs finalcontig_pharokka/vfdb_results.tsv --full-header --threads 7 ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:10.990\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mmseqs createtsv pharokka_v1.4.0_databases/vfdb finalcontig_pharokka/VFDB_target_dir/target_seqs finalcontig_pharokka/VFDB/results_mmseqs finalcontig_pharokka/vfdb_results.tsv --full-header --threads 7\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:10.991\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m348\u001b[0m - \u001b[1mRunning PyHMMER on PHROGs.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.247\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m364\u001b[0m - \u001b[1mPost Processing Output.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.271\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mcreate_mmseqs_tophits\u001b[0m:\u001b[36m2146\u001b[0m - \u001b[1mProcessing MMseqs2 outputs.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.272\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mcreate_mmseqs_tophits\u001b[0m:\u001b[36m2147\u001b[0m - \u001b[1mProcessing PHROGs output.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.550\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mprocess_vfdb_results\u001b[0m:\u001b[36m2351\u001b[0m - \u001b[1mProcessing VFDB output.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.559\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mprocess_vfdb_results\u001b[0m:\u001b[36m2437\u001b[0m - \u001b[1m0 VFDB virulence factors identified.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.561\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mprocess_card_results\u001b[0m:\u001b[36m2455\u001b[0m - \u001b[1mProcessing CARD output.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mprocess_card_results\u001b[0m:\u001b[36m2578\u001b[0m - \u001b[1m0 CARD AMR genes identified.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:32.980\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m435\u001b[0m - \u001b[1mConverting gff to genbank.\u001b[0m\n",
      "/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/Bio/SeqIO/InsdcIO.py:727: BiopythonWarning: Increasing length of locus line to allow long name. This will result in fields that are not in usual positions.\n",
      "  warnings.warn(\n",
      "\u001b[32m2024-07-19 18:44:33.050\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mpost_processing\u001b[0m:\u001b[36mextract_terl\u001b[0m:\u001b[36m1982\u001b[0m - \u001b[1mTerminase large subunit found.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:33.061\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m453\u001b[0m - \u001b[1mFinding the closest match for each contig in INPHARED using mash.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:33.063\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m50\u001b[0m - \u001b[1mStarted running mash sketch finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta -o finalcontig_pharokka/input_mash_sketch.msh -i ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:33.100\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun\u001b[0m:\u001b[36m52\u001b[0m - \u001b[1mDone running mash sketch finalcontig_pharokka/dnaapler/dnaapler_reoriented.fasta -o finalcontig_pharokka/input_mash_sketch.msh -i\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:33.101\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun_to_stdout\u001b[0m:\u001b[36m59\u001b[0m - \u001b[1mStarted running mash dist finalcontig_pharokka/input_mash_sketch.msh pharokka_v1.4.0_databases/1Aug2023_genomes.fa.msh -d 0.2 -i ...\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.293\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36mexternal_tools\u001b[0m:\u001b[36mrun_to_stdout\u001b[0m:\u001b[36m61\u001b[0m - \u001b[1mDone running mash dist finalcontig_pharokka/input_mash_sketch.msh pharokka_v1.4.0_databases/1Aug2023_genomes.fa.msh -d 0.2 -i\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m473\u001b[0m - \u001b[1mPharokka has finished.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m474\u001b[0m - \u001b[1mElapsed time: 460.09 seconds\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m476\u001b[0m - \u001b[1mIf you use Pharokka in your research, please cite:\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m477\u001b[0m - \u001b[1mGeorge Bouras, Roshan Nepal, Ghais Houtak, Alkis James Psaltis, Peter-John Wormald, Sarah Vreugde.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.569\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m480\u001b[0m - \u001b[1mPharokka: a fast scalable bacteriophage annotation tool.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.569\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m481\u001b[0m - \u001b[1mBioinformatics, Volume 39, Issue 1, January 2023, btac776.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.569\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m482\u001b[0m - \u001b[1mhttps://doi.org/10.1093/bioinformatics/btac776.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:44:40.569\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m483\u001b[0m - \u001b[1mYou should also cite the full list of tools Pharokka uses, which can be found at https://github.com/gbouras13/pharokka#citation.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pharokka.py -i newspades_results/contigs.fasta -o finalcontig_pharokka -d pharokka_v1.4.0_databases -g prodigal --dnaapler -t 7 -f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "8e1222b1-ab52-49e3-b55c-7e81db008f00",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m2024-07-19 18:47:01.747\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m116\u001b[0m - \u001b[1mStarting Pharokka v1.7.2\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m117\u001b[0m - \u001b[1mRunning pharokka_plotter.py to plot your phage.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m118\u001b[0m - \u001b[1mCommand executed: Namespace(infile='newspades_results/contigs.fasta', plot_name='SPSP9_plot', outdir='', gff='finalcontig_pharokka/pharokka.gff', genbank='finalcontig_pharokka/pharokka.gbk', prefix='pharokka', plot_title='Phage', force=False, label_hypotheticals=False, remove_other_features_labels=False, title_size='20', label_size='8', interval='5000', truncate='20', dpi='600', annotations='1', label_ids='')\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m119\u001b[0m - \u001b[1mRepository homepage is https://github.com/gbouras13/pharokka\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m120\u001b[0m - \u001b[1mWritten by George Bouras: george.bouras@adelaide.edu.au\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m121\u001b[0m - \u001b[1mChecking your inputs.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.748\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m221\u001b[0m - \u001b[1mYou have specified both Pharokka genbank and gff files.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.751\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m230\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.753\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m232\u001b[0m - \u001b[1mInput newspades_results/contigs.fasta is in FASTA format.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.753\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m237\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta for duplicate FASTA headers.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.754\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m239\u001b[0m - \u001b[1mAll headers in newspades_results/contigs.fasta are unique.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.754\u001b[0m | \u001b[33m\u001b[1mWARNING \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m229\u001b[0m - \u001b[33m\u001b[1mMore than one contig detected in the input file. Only the first phage contig will be plotted.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.754\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m290\u001b[0m - \u001b[1mAll files checked.\u001b[0m\n",
      "\u001b[32m2024-07-19 18:47:01.754\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36m<module>\u001b[0m:\u001b[36m291\u001b[0m - \u001b[1mPlotting the phage.\u001b[0m\n",
      "/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pycirclize/parser/genbank.py:66: UserWarning: min_range & max_range is no longer used in Genbank parser.\n",
      "  warnings.warn(\"min_range & max_range is no longer used in Genbank parser.\")\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/pharokka_plotter.py\", line 293, in <module>\n",
      "    create_plot(\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/plot.py\", line 140, in create_plot\n",
      "    cds_track.genomic_features(\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pycirclize/track.py\", line 1276, in genomic_features\n",
      "    self.arrow(start, end, r_lim=r_lim, **kwargs)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pycirclize/track.py\", line 330, in arrow\n",
      "    rad_arrow_start = self.x_to_rad(start)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pycirclize/track.py\", line 181, in x_to_rad\n",
      "    return self.parent_sector.x_to_rad(x, ignore_range_error)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pycirclize/sector.py\", line 220, in x_to_rad\n",
      "    raise ValueError(err_msg)\n",
      "ValueError: x=2064 is invalid range of 'pharokka' sector.\n",
      "# Sector = 'pharokka'\n",
      "# Size = 624 (0 - 624)\n",
      "# Degree Size = 360.00 (0.00 - 360.00)\n",
      "# Track List = ['Track01']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!pharokka_plotter.py -i newspades_results/contigs.fasta -n SPSP9_plot --gff finalcontig_pharokka/pharokka.gff --genbank finalcontig_pharokka/pharokka.gbk\n",
    "#pharokka_plotter.py -i input.fasta -n pharokka_plot --gff pharokka.gff --genbank pharokka.gbk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50539ad1-eb04-4857-bba8-05a3c6212e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#The ouput of the above 2 commands might be incorrect as the databse used by the pharokka common dabove does not contain the Salmonella Phage PRF-SP7 hence the annotation wouldn't be complete. and the other following steps would not be too. \n",
    "#But we can adapt the pharokka_protein command to see the proteins present in the organism."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "cbdd06fb-f4d6-4d88-a084-57c60fa00c06",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m2024-07-19 19:08:47.176\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36minstantiate_dirs\u001b[0m:\u001b[36m183\u001b[0m - \u001b[1mRemoving output directory Pharokka_proteins as -f or --force was specified.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.190\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m64\u001b[0m - \u001b[1mStarting Pharokka v1.7.2\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.190\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m65\u001b[0m - \u001b[1mRunning pharokka_proteins.py to annotate proteins.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.190\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m66\u001b[0m - \u001b[1mCommand executed: Namespace(infile='newspades_results/contigs.fasta', outdir='Pharokka_proteins', database='pharokka_v1.4.0_databases', threads='6', force=True, prefix='Default', evalue='1E-05', hmm_only=True, mmseqs2_only=False, citation=False)\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.190\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m67\u001b[0m - \u001b[1mRepository homepage is https://github.com/gbouras13/pharokka\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.191\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m68\u001b[0m - \u001b[1mWritten by George Bouras: george.bouras@adelaide.edu.au\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.191\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m71\u001b[0m - \u001b[1mChecking database installation.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.191\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m75\u001b[0m - \u001b[1mAll databases have been successfully checked.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.191\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m82\u001b[0m - \u001b[1mChecking dependencies.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.386\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m383\u001b[0m - \u001b[1mPhanotate version found is v1.5.1\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.386\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m392\u001b[0m - \u001b[1mPhanotate version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.390\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m414\u001b[0m - \u001b[1mMMseqs2 version found is v13.45111\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.390\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m423\u001b[0m - \u001b[1mMMseqs2 version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.491\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m447\u001b[0m - \u001b[1mtRNAscan-SE version found is v2.0.12\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.492\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m458\u001b[0m - \u001b[1mtRNAscan-SE version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m482\u001b[0m - \u001b[1mMinCED version found is v0.4.2\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.568\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m493\u001b[0m - \u001b[1mMinCED version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.571\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m519\u001b[0m - \u001b[1mARAGORN version found is v1.2.41\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.571\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m530\u001b[0m - \u001b[1mARAGORN version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.580\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m547\u001b[0m - \u001b[1mmash version found is v2.3\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:47.580\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m554\u001b[0m - \u001b[1mmash version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.532\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m573\u001b[0m - \u001b[1mDnaapler version found is v0.7.0\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.533\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m580\u001b[0m - \u001b[1mDnaapler version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.533\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m592\u001b[0m - \u001b[1mPyrodigal version is v3.4.1\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.534\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m593\u001b[0m - \u001b[1mPyrodigal version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.534\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m605\u001b[0m - \u001b[1mPyrodigal_gv version is v0.3.1\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.535\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mcheck_dependencies\u001b[0m:\u001b[36m606\u001b[0m - \u001b[1mPyrodigal_gv version is ok.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.540\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m230\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.541\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m232\u001b[0m - \u001b[1mInput newspades_results/contigs.fasta is in FASTA format.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.541\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m237\u001b[0m - \u001b[1mChecking input newspades_results/contigs.fasta for duplicate FASTA headers.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.542\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36minput_commands\u001b[0m:\u001b[36mvalidate_fasta\u001b[0m:\u001b[36m239\u001b[0m - \u001b[1mAll headers in newspades_results/contigs.fasta are unique.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.543\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m107\u001b[0m - \u001b[1mYou have specified --hmm_only. MMseqs2 will not be run.\u001b[0m\n",
      "\u001b[32m2024-07-19 19:08:48.543\u001b[0m | \u001b[1mINFO    \u001b[0m | \u001b[36m__main__\u001b[0m:\u001b[36mmain\u001b[0m:\u001b[36m146\u001b[0m - \u001b[1mRunning PyHMMER.\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/pharokka_proteins.py\", line 213, in <module>\n",
      "    main()\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/pharokka_proteins.py\", line 147, in main\n",
      "    best_results_pyhmmer = run_pyhmmer_proteins(\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/bin/proteins.py\", line 224, in run_pyhmmer_proteins\n",
      "    for hits in pyhmmer.hmmer.hmmscan(\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/hmmer.py\", line 520, in _multi_threaded\n",
      "    yield results[0].get()\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/hmmer.py\", line 122, in get\n",
      "    raise self.exception\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/hmmer.py\", line 215, in run\n",
      "    hits = self.process(chore.query)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/hmmer.py\", line 231, in process\n",
      "    hits = self.query(query)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/utils.py\", line 70, in _method\n",
      "    return method.__get__(obj, cls)(*args, **kwargs)\n",
      "  File \"/home/mghalib/miniconda3/envs/phageinformatics_UCPH_June_2024/lib/python3.10/site-packages/pyhmmer/hmmer.py\", line 415, in _\n",
      "    return self.pipeline.scan_seq(query, self.targets)\n",
      "  File \"pyhmmer/plan7.pyx\", line 6068, in pyhmmer.plan7.Pipeline.__pyx_fuse_1scan_seq\n",
      "  File \"pyhmmer/plan7.pyx\", line 6116, in pyhmmer.plan7.Pipeline.scan_seq\n",
      "pyhmmer.errors.AlphabetMismatch: Expected amino alphabet, found DNA alphabet\n"
     ]
    }
   ],
   "source": [
    "!pharokka_proteins.py -i newspades_results/contigs.fasta -d pharokka_v1.4.0_databases -o Pharokka_proteins -t 6 --hmm_only -f\n",
    "#pharokka_proteins.py -i <inpu protein FASTA file> -d <database directory> -o <output folder> -t <threads> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a0b5eed-1d1f-4bb0-a68d-ee54ca51fb20",
   "metadata": {},
   "outputs": [],
   "source": [
    "#This the final step of the analysis where we generate a circular genome graph of the phage and acquire \n",
    "#annotated segments of the genome with the orientation of the genome. I was unsuccesful at forming this\n",
    "#graph."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f3945dc-a422-4930-a482-9587f902f216",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ff931ec-2031-4068-b2f2-27121b0bb567",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "02e332ad-b4eb-4d0a-acbc-ba9318acde0f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edb01534-faa3-4983-befd-712c9e46e8b6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "372667c2-5c63-4cce-adea-6c467bd162b5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "phageinformatics_UCPH_June_2024",
   "language": "python",
   "name": "phageinformatics_ucph_june_2024"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
